--- a/net/minecraft/core/dispenser/ShearsDispenseItemBehavior.java
+++ b/net/minecraft/core/dispenser/ShearsDispenseItemBehavior.java
@@ -24,7 +_,7 @@
         ServerLevel serverlevel = p_302443_.level();
         if (!serverlevel.isClientSide()) {
             BlockPos blockpos = p_302443_.pos().relative(p_302443_.state().getValue(DispenserBlock.FACING));
-            this.setSuccess(tryShearBeehive(serverlevel, p_123581_, blockpos) || tryShearEntity(serverlevel, blockpos, p_123581_));
+            this.setSuccess(net.neoforged.neoforge.common.CommonHooks.tryDispenseShearsHarvestBlock(p_302443_, p_123581_, serverlevel, blockpos) || tryShearBeehive(serverlevel, blockpos) || tryShearEntity(serverlevel, blockpos, p_123581_));
             if (this.isSuccess()) {
                 p_123581_.hurtAndBreak(1, serverlevel, null, p_348118_ -> {});
             }
@@ -33,13 +_,13 @@
         return p_123581_;
     }
 
-    private static boolean tryShearBeehive(ServerLevel p_123577_, ItemStack p_432828_, BlockPos p_123578_) {
+    private static boolean tryShearBeehive(ServerLevel p_123577_, BlockPos p_123578_) {
         BlockState blockstate = p_123577_.getBlockState(p_123578_);
         if (blockstate.is(BlockTags.BEEHIVES, p_202454_ -> p_202454_.hasProperty(BeehiveBlock.HONEY_LEVEL) && p_202454_.getBlock() instanceof BeehiveBlock)) {
             int i = blockstate.getValue(BeehiveBlock.HONEY_LEVEL);
             if (i >= 5) {
                 p_123577_.playSound(null, p_123578_, SoundEvents.BEEHIVE_SHEAR, SoundSource.BLOCKS, 1.0F, 1.0F);
-                BeehiveBlock.dropHoneycomb(p_123577_, p_432828_, blockstate, p_123577_.getBlockEntity(p_123578_), null, p_123578_);
+                BeehiveBlock.dropHoneycomb(p_123577_, p_123578_);
                 ((BeehiveBlock)blockstate.getBlock())
                     .releaseBeesAndResetHoneyLevel(p_123577_, blockstate, p_123578_, null, BeehiveBlockEntity.BeeReleaseStatus.BEE_RELEASED);
                 p_123577_.gameEvent(null, GameEvent.SHEAR, p_123578_);
@@ -56,8 +_,9 @@
                 return true;
             }
 
-            if (entity instanceof Shearable shearable && shearable.readyForShearing()) {
-                shearable.shear(p_418032_, SoundSource.BLOCKS, p_418413_);
+            if (entity instanceof net.neoforged.neoforge.common.IShearable shearable && shearable.isShearable(null, p_418413_, p_418032_, p_418277_)) {
+                shearable.onSheared(null, p_418413_, p_418032_, p_418277_)
+                        .forEach(drop -> shearable.spawnShearedDrop(p_418032_, p_418277_, drop));
                 p_418032_.gameEvent(null, GameEvent.SHEAR, p_418277_);
                 return true;
             }
