--- a/net/minecraft/server/Services.java
+++ b/net/minecraft/server/Services.java
@@ -7,26 +_,19 @@
 import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
 import java.io.File;
 import javax.annotation.Nullable;
-import net.minecraft.server.players.CachedUserNameToIdResolver;
-import net.minecraft.server.players.ProfileResolver;
-import net.minecraft.server.players.UserNameToIdResolver;
+import net.minecraft.server.players.GameProfileCache;
 import net.minecraft.util.SignatureValidator;
 
 public record Services(
-    MinecraftSessionService sessionService,
-    ServicesKeySet servicesKeySet,
-    GameProfileRepository profileRepository,
-    UserNameToIdResolver nameToIdCache,
-    ProfileResolver profileResolver
+    MinecraftSessionService sessionService, ServicesKeySet servicesKeySet, GameProfileRepository profileRepository, GameProfileCache profileCache
 ) {
     private static final String USERID_CACHE_FILE = "usercache.json";
 
     public static Services create(YggdrasilAuthenticationService p_214345_, File p_214346_) {
         MinecraftSessionService minecraftsessionservice = p_214345_.createMinecraftSessionService();
         GameProfileRepository gameprofilerepository = p_214345_.createProfileRepository();
-        UserNameToIdResolver usernametoidresolver = new CachedUserNameToIdResolver(gameprofilerepository, new File(p_214346_, "usercache.json"));
-        ProfileResolver profileresolver = new ProfileResolver.Cached(minecraftsessionservice, usernametoidresolver);
-        return new Services(minecraftsessionservice, p_214345_.getServicesKeySet(), gameprofilerepository, usernametoidresolver, profileresolver);
+        GameProfileCache gameprofilecache = new GameProfileCache(gameprofilerepository, new File(p_214346_, "usercache.json"));
+        return new Services(minecraftsessionservice, p_214345_.getServicesKeySet(), gameprofilerepository, gameprofilecache);
     }
 
     @Nullable
