--- a/net/minecraft/client/renderer/texture/atlas/sources/PalettedPermutations.java
+++ b/net/minecraft/client/renderer/texture/atlas/sources/PalettedPermutations.java
@@ -26,6 +_,7 @@
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.packs.resources.Resource;
 import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.server.packs.resources.ResourceMetadata;
 import net.minecraft.util.ARGB;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
@@ -141,7 +_,9 @@
             Object object;
             try {
                 NativeImage nativeimage = this.baseImage.get().mappedCopy(this.palette.get());
-                return new SpriteContents(this.permutationLocation, new FrameSize(nativeimage.getWidth(), nativeimage.getHeight()), nativeimage);
+                return new SpriteContents(
+                    this.permutationLocation, new FrameSize(nativeimage.getWidth(), nativeimage.getHeight()), nativeimage, ResourceMetadata.EMPTY
+                );
             } catch (IllegalArgumentException | IOException ioexception) {
                 PalettedPermutations.LOGGER.error("unable to apply palette to {}", this.permutationLocation, ioexception);
                 object = null;
