--- a/net/minecraft/data/loot/packs/VanillaGiftLoot.java
+++ b/net/minecraft/data/loot/packs/VanillaGiftLoot.java
@@ -335,9 +_,5 @@
             BuiltInLootTables.ARMADILLO_SHED,
             LootTable.lootTable().withPool(LootPool.lootPool().setRolls(ConstantValue.exactly(1.0F)).add(LootItem.lootTableItem(Items.ARMADILLO_SCUTE)))
         );
-        p_250831_.accept(
-            BuiltInLootTables.TURTLE_GROW,
-            LootTable.lootTable().withPool(LootPool.lootPool().setRolls(ConstantValue.exactly(1.0F)).add(LootItem.lootTableItem(Items.TURTLE_SCUTE)))
-        );
     }
 }
