--- a/net/minecraft/client/gui/screens/packs/PackSelectionScreen.java
+++ b/net/minecraft/client/gui/screens/packs/PackSelectionScreen.java
@@ -27,6 +_,7 @@
 import net.minecraft.ChatFormatting;
 import net.minecraft.Util;
 import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.ComponentPath;
 import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.StringWidget;
@@ -115,18 +_,20 @@
                 .build()
         );
         this.doneButton = linearlayout1.addChild(Button.builder(CommonComponents.GUI_DONE, p_100036_ -> this.onClose()).build());
+        this.reload();
         this.layout.visitWidgets(p_329736_ -> {
             AbstractWidget abstractwidget = this.addRenderableWidget(p_329736_);
         });
         this.repositionElements();
-        this.reload();
     }
 
     @Override
     protected void repositionElements() {
         this.layout.arrangeElements();
-        this.availablePackList.updateSizeAndPosition(200, this.layout.getContentHeight(), this.width / 2 - 15 - 200, this.layout.getHeaderHeight());
-        this.selectedPackList.updateSizeAndPosition(200, this.layout.getContentHeight(), this.width / 2 + 15, this.layout.getHeaderHeight());
+        this.availablePackList.updateSize(200, this.layout);
+        this.availablePackList.setX(this.width / 2 - 15 - 200);
+        this.selectedPackList.updateSize(200, this.layout);
+        this.selectedPackList.setX(this.width / 2 + 15);
     }
 
     @Override
@@ -147,12 +_,35 @@
         }
     }
 
-    private void populateLists(@Nullable PackSelectionModel.EntryBase p_440233_) {
-        this.selectedPackList.updateList(this.model.getSelected(), p_440233_);
-        this.availablePackList.updateList(this.model.getUnselected(), p_440233_);
+    private void populateLists() {
+        this.updateList(this.selectedPackList, this.model.getSelected());
+        this.updateList(this.availablePackList, this.model.getUnselected());
         this.doneButton.active = !this.selectedPackList.children().isEmpty();
     }
 
+    private void updateList(TransferableSelectionList p_100014_, Stream<PackSelectionModel.Entry> p_100015_) {
+        p_100014_.children().clear();
+        TransferableSelectionList.PackEntry transferableselectionlist$packentry = p_100014_.getSelected();
+        String s = transferableselectionlist$packentry == null ? "" : transferableselectionlist$packentry.getPackId();
+        p_100014_.setSelected(null);
+        p_100015_.forEach(
+            p_425149_ -> {
+                TransferableSelectionList.PackEntry transferableselectionlist$packentry1 = new TransferableSelectionList.PackEntry(
+                    this.minecraft, p_100014_, p_425149_
+                );
+                p_100014_.children().add(transferableselectionlist$packentry1);
+                if (p_425149_.getId().equals(s)) {
+                    p_100014_.setSelected(transferableselectionlist$packentry1);
+                }
+            }
+        );
+    }
+
+    public void updateFocus(TransferableSelectionList p_265419_) {
+        TransferableSelectionList transferableselectionlist = this.selectedPackList == p_265419_ ? this.availablePackList : this.selectedPackList;
+        this.changeFocus(ComponentPath.path(transferableselectionlist.getFirstElement(), transferableselectionlist, this));
+    }
+
     public void clearSelected() {
         this.selectedPackList.setSelected(null);
         this.availablePackList.setSelected(null);
@@ -160,7 +_,7 @@
 
     private void reload() {
         this.model.findNewPacks();
-        this.populateLists(null);
+        this.populateLists();
         this.ticksToReload = 0L;
         this.packIcons.clear();
     }
