--- a/net/minecraft/server/packs/repository/PackCompatibility.java
+++ b/net/minecraft/server/packs/repository/PackCompatibility.java
@@ -2,16 +_,13 @@
 
 import net.minecraft.ChatFormatting;
 import net.minecraft.network.chat.Component;
-import net.minecraft.server.packs.metadata.pack.PackFormat;
 import net.minecraft.util.InclusiveRange;
 
 public enum PackCompatibility {
     TOO_OLD("old"),
     TOO_NEW("new"),
-    UNKNOWN("unknown"),
     COMPATIBLE("compatible");
 
-    public static final int UNKNOWN_VERSION = Integer.MAX_VALUE;
     private final Component description;
     private final Component confirmation;
 
@@ -24,13 +_,11 @@
         return this == COMPATIBLE;
     }
 
-    public static PackCompatibility forVersion(InclusiveRange<PackFormat> p_294404_, PackFormat p_433662_) {
-        if (p_294404_.minInclusive().major() == Integer.MAX_VALUE) {
-            return UNKNOWN;
-        } else if (p_294404_.maxInclusive().compareTo(p_433662_) < 0) {
+    public static PackCompatibility forVersion(InclusiveRange<Integer> p_294404_, int p_294597_) {
+        if (p_294404_.maxInclusive() < p_294597_) {
             return TOO_OLD;
         } else {
-            return p_433662_.compareTo(p_294404_.minInclusive()) < 0 ? TOO_NEW : COMPATIBLE;
+            return p_294597_ < p_294404_.minInclusive() ? TOO_NEW : COMPATIBLE;
         }
     }
 
