--- a/com/mojang/realmsclient/gui/screens/configuration/RealmsBackupInfoScreen.java
+++ b/com/mojang/realmsclient/gui/screens/configuration/RealmsBackupInfoScreen.java
@@ -44,7 +_,7 @@
 
     @Override
     protected void repositionElements() {
-        this.backupInfoList.updateSize(this.width, this.layout);
+        this.backupInfoList.setSize(this.width, this.layout.getContentHeight());
         this.layout.arrangeElements();
     }
 
@@ -118,14 +_,21 @@
         }
 
         @Override
-        public void renderContent(GuiGraphics p_439863_, int p_438898_, int p_440008_, boolean p_439966_, float p_439342_) {
-            p_439863_.drawString(RealmsBackupInfoScreen.this.font, this.translateKey(this.key), this.getContentX(), this.getContentY(), -6250336);
-            p_439863_.drawString(
-                RealmsBackupInfoScreen.this.font,
-                RealmsBackupInfoScreen.this.checkForSpecificMetadata(this.key, this.value),
-                this.getContentX(),
-                this.getContentY() + 12,
-                -1
+        public void render(
+            GuiGraphics p_419842_,
+            int p_420042_,
+            int p_419515_,
+            int p_419833_,
+            int p_419652_,
+            int p_419975_,
+            int p_419697_,
+            int p_419569_,
+            boolean p_419605_,
+            float p_419972_
+        ) {
+            p_419842_.drawString(RealmsBackupInfoScreen.this.font, this.translateKey(this.key), p_419833_, p_419515_, -6250336);
+            p_419842_.drawString(
+                RealmsBackupInfoScreen.this.font, RealmsBackupInfoScreen.this.checkForSpecificMetadata(this.key, this.value), p_419833_, p_419515_ + 12, -1
             );
         }
 
