--- a/net/minecraft/world/entity/animal/Bee.java
+++ b/net/minecraft/world/entity/animal/Bee.java
@@ -350,7 +_,7 @@
     }
 
     public static boolean isNightOrRaining(Level p_368571_) {
-        return p_368571_.dimensionType().hasSkyLight() && p_368571_.dimension() != Level.END && (p_368571_.isDarkOutside() || p_368571_.isRaining());
+        return p_368571_.dimensionType().hasSkyLight() && (p_368571_.isDarkOutside() || p_368571_.isRaining());
     }
 
     public void setStayOutOfHiveCountdown(int p_27916_) {
@@ -474,7 +_,7 @@
     @Override
     public void aiStep() {
         super.aiStep();
-        if (!this.level().isClientSide()) {
+        if (!this.level().isClientSide) {
             if (this.stayOutOfHiveCountdown > 0) {
                 this.stayOutOfHiveCountdown--;
             }
@@ -500,7 +_,7 @@
         if (this.hivePos == null) {
             return null;
         } else {
-            return this.isTooFarAway(this.hivePos) ? null : this.level().getBlockEntity(this.hivePos, BlockEntityType.BEEHIVE).orElse(null);
+            return this.isTooFarAway(this.hivePos) ? null : this.level().getBlockEntity(this.hivePos) instanceof BeehiveBlockEntity hive ? hive : null;
         }
     }
 
@@ -588,7 +_,7 @@
             MobEffectInstance mobeffectinstance = flowerblock.getBeeInteractionEffect();
             if (mobeffectinstance != null) {
                 this.usePlayerItem(p_383058_, p_383220_, itemstack);
-                if (!this.level().isClientSide()) {
+                if (!this.level().isClientSide) {
                     this.addEffect(mobeffectinstance);
                 }
 
@@ -663,8 +_,18 @@
     }
 
     @Override
+    @Deprecated // FORGE: use jumpInFluid instead
     protected void jumpInLiquid(TagKey<Fluid> p_204061_) {
+        this.jumpInLiquidInternal();
+    }
+
+    private void jumpInLiquidInternal() {
         this.setDeltaMovement(this.getDeltaMovement().add(0.0, 0.01, 0.0));
+    }
+
+    @Override
+    public void jumpInFluid(net.neoforged.neoforge.fluids.FluidType type) {
+        this.jumpInLiquidInternal();
     }
 
     @Override
