--- a/net/minecraft/world/entity/animal/Pufferfish.java
+++ b/net/minecraft/world/entity/animal/Pufferfish.java
@@ -96,7 +_,7 @@
 
     @Override
     public void tick() {
-        if (!this.level().isClientSide() && this.isAlive() && this.isEffectiveAi()) {
+        if (!this.level().isClientSide && this.isAlive() && this.isEffectiveAi()) {
             if (this.inflateCounter > 0) {
                 if (this.getPuffState() == 0) {
                     this.makeSound(SoundEvents.PUFFER_FISH_BLOW_UP);
