--- a/net/minecraft/world/entity/animal/Squid.java
+++ b/net/minecraft/world/entity/animal/Squid.java
@@ -114,7 +_,7 @@
         this.oldTentacleAngle = this.tentacleAngle;
         this.tentacleMovement = this.tentacleMovement + this.tentacleSpeed;
         if (this.tentacleMovement > Math.PI * 2) {
-            if (this.level().isClientSide()) {
+            if (this.level().isClientSide) {
                 this.tentacleMovement = (float) (Math.PI * 2);
             } else {
                 this.tentacleMovement -= (float) (Math.PI * 2);
@@ -156,7 +_,7 @@
             this.xBodyRot = this.xBodyRot + (-((float)Mth.atan2(d0, vec3.y)) * (180.0F / (float)Math.PI) - this.xBodyRot) * 0.1F;
         } else {
             this.tentacleAngle = Mth.abs(Mth.sin(this.tentacleMovement)) * (float) Math.PI * 0.25F;
-            if (!this.level().isClientSide()) {
+            if (!this.level().isClientSide) {
                 double d1 = this.getDeltaMovement().y;
                 if (this.hasEffect(MobEffects.LEVITATION)) {
                     d1 = 0.05 * (this.getEffect(MobEffects.LEVITATION).getAmplifier() + 1);
