--- a/net/minecraft/util/datafix/DataFixers.java
+++ b/net/minecraft/util/datafix/DataFixers.java
@@ -85,7 +_,6 @@
 import net.minecraft.util.datafix.fixes.ChunkToProtochunkFix;
 import net.minecraft.util.datafix.fixes.ColorlessShulkerEntityFix;
 import net.minecraft.util.datafix.fixes.ContainerBlockEntityLockPredicateFix;
-import net.minecraft.util.datafix.fixes.CopperGolemWeatherStateFix;
 import net.minecraft.util.datafix.fixes.CriteriaRenameFix;
 import net.minecraft.util.datafix.fixes.CustomModelDataExpandFix;
 import net.minecraft.util.datafix.fixes.DecoratedPotFieldRenameFix;
@@ -366,9 +_,6 @@
 import net.minecraft.util.datafix.schemas.V4312;
 import net.minecraft.util.datafix.schemas.V4420;
 import net.minecraft.util.datafix.schemas.V4421;
-import net.minecraft.util.datafix.schemas.V4531;
-import net.minecraft.util.datafix.schemas.V4532;
-import net.minecraft.util.datafix.schemas.V4533;
 import net.minecraft.util.datafix.schemas.V501;
 import net.minecraft.util.datafix.schemas.V700;
 import net.minecraft.util.datafix.schemas.V701;
@@ -1357,10 +_,35 @@
         Schema schema202 = p_14514_.addSchema(3800, SAME_NAMESPACED);
         UnaryOperator<String> unaryoperator2 = createRenamer(Map.of("minecraft:scute", "minecraft:turtle_scute"));
         p_14514_.addFixer(ItemRenameFix.create(schema202, "Rename scute item to turtle_scute", unaryoperator2));
+
+        // Neo: Remove step height attribute, which cannot be sanely updated
+        // Happens in 23w51a
+        Schema neoSchema3801 = p_14514_.addSchema(3801, SAME_NAMESPACED);
+        p_14514_.addFixer(new net.neoforged.neoforge.common.data.fixes.NeoForgeEntityLegacyAttributesFix(
+                "(Neo) Remove step height attribute",
+                neoSchema3801,
+                java.util.List.of(
+                        "neoforge:step_height"
+                )
+        ));
+
         Schema schema203 = p_14514_.addSchema(3803, SAME_NAMESPACED);
         p_14514_.addFixer(
             new RenameEnchantmentsFix(schema203, "Rename sweeping enchant to sweeping_edge", Map.of("minecraft:sweeping", "minecraft:sweeping_edge"))
         );
+
+        // Neo: rename neo attributes to new MC attributes
+        // Happens in 24w03a
+        Schema neoSchema3804 = p_14514_.addSchema(3804, SAME_NAMESPACED);
+        p_14514_.addFixer(new AttributesRenameLegacy(
+                neoSchema3804,
+                "(Neo) Rename reach attributes to vanilla",
+                createRenamer(ImmutableMap.of(
+                        "neoforge:entity_reach", "minecraft:player.entity_interaction_range",
+                        "neoforge:block_reach", "minecraft:player.block_interaction_range"
+                ))
+        ));
+
         Schema schema204 = p_14514_.addSchema(3807, V3807::new);
         p_14514_.addFixer(new AddNewChoices(schema204, "Added Vault", References.BLOCK_ENTITY));
         Schema schema205 = p_14514_.addSchema(3807, 1, SAME_NAMESPACED);
@@ -1383,6 +_,18 @@
                 schema212, "Rename jump strength attribute", createRenamer("minecraft:horse.jump_strength", "minecraft:generic.jump_strength")
             )
         );
+
+        // Neo: rename neo attributes to new MC attributes
+        // Happens in 24w06a
+        Schema neoSchema3815 = p_14514_.addSchema(3815, SAME_NAMESPACED);
+        p_14514_.addFixer(new AttributesRenameLegacy(
+                neoSchema3815,
+                "(Neo) Rename gravity attribute to vanilla",
+                createRenamer(
+                        "neoforge:entity_gravity", "minecraft:generic.gravity"
+                )
+        ));
+
         Schema schema213 = p_14514_.addSchema(3816, V3816::new);
         p_14514_.addFixer(new AddNewChoices(schema213, "Added Bogged", References.ENTITY));
         Schema schema214 = p_14514_.addSchema(3818, V3818::new);
@@ -1559,16 +_,6 @@
         p_14514_.addFixer(new AddNewChoices(schema267, "Added Happy Ghast", References.ENTITY));
         Schema schema268 = p_14514_.addSchema(4424, SAME_NAMESPACED);
         p_14514_.addFixer(new FeatureFlagRemoveFix(schema268, "Remove Locator Bar experimental feature flag", Set.of("minecraft:locator_bar")));
-        Schema schema269 = p_14514_.addSchema(4531, V4531::new);
-        p_14514_.addFixer(new AddNewChoices(schema269, "Added Copper Golem", References.ENTITY));
-        Schema schema270 = p_14514_.addSchema(4532, V4532::new);
-        p_14514_.addFixer(new AddNewChoices(schema270, "Added Copper Golem Statue Block Entity", References.BLOCK_ENTITY));
-        Schema schema271 = p_14514_.addSchema(4533, V4533::new);
-        p_14514_.addFixer(new AddNewChoices(schema271, "Added Shelf", References.BLOCK_ENTITY));
-        Schema schema272 = p_14514_.addSchema(4535, SAME_NAMESPACED);
-        p_14514_.addFixer(new CopperGolemWeatherStateFix(schema272));
-        Schema schema273 = p_14514_.addSchema(4537, SAME_NAMESPACED);
-        p_14514_.addFixer(new ChunkDeleteLightFix(schema273));
     }
 
     private static UnaryOperator<String> createRenamerNoNamespace(Map<String, String> p_326818_) {
