--- a/net/minecraft/client/renderer/entity/player/PlayerRenderer.java
+++ b/net/minecraft/client/renderer/entity/player/PlayerRenderer.java
@@ -3,14 +_,14 @@
 import com.mojang.blaze3d.vertex.PoseStack;
 import com.mojang.math.Axis;
 import javax.annotation.Nullable;
+import net.minecraft.client.model.HumanoidArmorModel;
 import net.minecraft.client.model.HumanoidModel;
 import net.minecraft.client.model.PlayerModel;
 import net.minecraft.client.model.geom.ModelLayers;
 import net.minecraft.client.model.geom.ModelPart;
 import net.minecraft.client.player.AbstractClientPlayer;
+import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.RenderType;
-import net.minecraft.client.renderer.SubmitNodeCollector;
-import net.minecraft.client.renderer.entity.ArmorModelSet;
 import net.minecraft.client.renderer.entity.EntityRendererProvider;
 import net.minecraft.client.renderer.entity.HumanoidMobRenderer;
 import net.minecraft.client.renderer.entity.LivingEntityRenderer;
@@ -58,11 +_,8 @@
         this.addLayer(
             new HumanoidArmorLayer<>(
                 this,
-                ArmorModelSet.bake(
-                    p_174558_ ? ModelLayers.PLAYER_SLIM_ARMOR : ModelLayers.PLAYER_ARMOR,
-                    p_174557_.getModelSet(),
-                    p_435816_ -> new PlayerModel(p_435816_, p_174558_)
-                ),
+                new HumanoidArmorModel(p_174557_.bakeLayer(p_174558_ ? ModelLayers.PLAYER_SLIM_INNER_ARMOR : ModelLayers.PLAYER_INNER_ARMOR)),
+                new HumanoidArmorModel(p_174557_.bakeLayer(p_174558_ ? ModelLayers.PLAYER_SLIM_OUTER_ARMOR : ModelLayers.PLAYER_OUTER_ARMOR)),
                 p_174557_.getEquipmentRenderer()
             )
         );
@@ -70,7 +_,7 @@
         this.addLayer(new ArrowLayer<>(this, p_174557_));
         this.addLayer(new Deadmau5EarsLayer(this, p_174557_.getModelSet()));
         this.addLayer(new CapeLayer(this, p_174557_.getModelSet(), p_174557_.getEquipmentAssets()));
-        this.addLayer(new CustomHeadLayer<>(this, p_174557_.getModelSet(), p_174557_.getPlayerSkinRenderCache()));
+        this.addLayer(new CustomHeadLayer<>(this, p_174557_.getModelSet()));
         this.addLayer(new WingsLayer<>(this, p_174557_.getModelSet(), p_174557_.getEquipmentRenderer()));
         this.addLayer(new ParrotOnShoulderLayer(this, p_174557_.getModelSet()));
         this.addLayer(new SpinAttackEffectLayer(this, p_174557_.getModelSet()));
@@ -99,6 +_,11 @@
     }
 
     private static HumanoidModel.ArmPose getArmPose(Player p_386775_, ItemStack p_388403_, InteractionHand p_117796_) {
+        var extensions = net.neoforged.neoforge.client.extensions.common.IClientItemExtensions.of(p_388403_);
+        var armPose = extensions.getArmPose(p_386775_, p_117796_, p_388403_);
+        if (armPose != null) {
+            return armPose;
+        }
         if (p_388403_.isEmpty()) {
             return HumanoidModel.ArmPose.EMPTY;
         } else if (!p_386775_.swinging && p_388403_.is(Items.CROSSBOW) && CrossbowItem.isCharged(p_388403_)) {
@@ -148,22 +_,22 @@
         p_117799_.scale(0.9375F, 0.9375F, 0.9375F);
     }
 
-    protected void submitNameTag(PlayerRenderState p_434091_, PoseStack p_434864_, SubmitNodeCollector p_433425_) {
-        p_434864_.pushPose();
-        if (p_434091_.scoreText != null) {
-            p_433425_.submitNameTag(
-                p_434864_, p_434091_.nameTagAttachment, p_434091_.scoreText, !p_434091_.isDiscrete, p_434091_.lightCoords, p_434091_.distanceToCameraSq
-            );
-            p_434864_.translate(0.0F, 9.0F * 1.15F * 0.025F, 0.0F);
-        }
-
-        if (p_434091_.nameTag != null) {
-            p_433425_.submitNameTag(
-                p_434864_, p_434091_.nameTagAttachment, p_434091_.nameTag, !p_434091_.isDiscrete, p_434091_.lightCoords, p_434091_.distanceToCameraSq
-            );
-        }
-
-        p_434864_.popPose();
+    @Override
+    public void render(PlayerRenderState p_361886_, PoseStack p_115311_, MultiBufferSource p_115312_, int p_115313_) {
+        if (net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderPlayerEvent.Pre(p_361886_, this, p_361886_.partialTick, p_115311_, p_115312_, p_115313_)).isCanceled()) return;
+        super.render(p_361886_, p_115311_, p_115312_, p_115313_);
+        net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderPlayerEvent.Post(p_361886_, this, p_361886_.partialTick, p_115311_, p_115312_, p_115313_));
+    }
+
+    protected void renderNameTag(PlayerRenderState p_363185_, Component p_117809_, PoseStack p_117810_, MultiBufferSource p_117811_, int p_117812_) {
+        p_117810_.pushPose();
+        if (p_363185_.scoreText != null) {
+            super.renderNameTag(p_363185_, p_363185_.scoreText, p_117810_, p_117811_, p_117812_);
+            p_117810_.translate(0.0F, 9.0F * 1.15F * 0.025F, 0.0F);
+        }
+
+        super.renderNameTag(p_363185_, p_117809_, p_117810_, p_117811_, p_117812_);
+        p_117810_.popPose();
     }
 
     public PlayerRenderState createRenderState() {
@@ -178,6 +_,8 @@
         p_360583_.skin = p_361478_.getSkin();
         p_360583_.arrowCount = p_361478_.getArrowCount();
         p_360583_.stingerCount = p_361478_.getStingerCount();
+        p_360583_.useItemRemainingTicks = p_361478_.getUseItemRemainingTicks();
+        p_360583_.swinging = p_361478_.swinging;
         p_360583_.isSpectator = p_361478_.isSpectator();
         p_360583_.showHat = p_361478_.isModelPartShown(PlayerModelPart.HAT);
         p_360583_.showJacket = p_361478_.isModelPartShown(PlayerModelPart.JACKET);
@@ -260,17 +_,33 @@
         }
     }
 
-    public void renderRightHand(PoseStack p_117771_, SubmitNodeCollector p_439755_, int p_117773_, ResourceLocation p_363694_, boolean p_366898_) {
-        this.renderHand(p_117771_, p_439755_, p_117773_, p_363694_, this.model.rightArm, p_366898_);
-    }
-
-    public void renderLeftHand(PoseStack p_117814_, SubmitNodeCollector p_439002_, int p_117816_, ResourceLocation p_361745_, boolean p_366730_) {
-        this.renderHand(p_117814_, p_439002_, p_117816_, p_361745_, this.model.leftArm, p_366730_);
-    }
-
-    private void renderHand(
-        PoseStack p_117776_, SubmitNodeCollector p_440196_, int p_117778_, ResourceLocation p_360319_, ModelPart p_117780_, boolean p_366655_
-    ) {
+    /**
+     * @deprecated Neo: use {@link #renderRightHand(PoseStack, MultiBufferSource, int, ResourceLocation, boolean, AbstractClientPlayer)} instead
+     */
+    @Deprecated
+    public void renderRightHand(PoseStack p_117771_, MultiBufferSource p_117772_, int p_117773_, ResourceLocation p_363694_, boolean p_366898_) {
+        this.renderRightHand(p_117771_, p_117772_, p_117773_, p_363694_, p_366898_, net.minecraft.client.Minecraft.getInstance().player);
+    }
+
+    public void renderRightHand(PoseStack p_117771_, MultiBufferSource p_117772_, int p_117773_, ResourceLocation p_363694_, boolean p_366898_, AbstractClientPlayer player) {
+        if(!net.neoforged.neoforge.client.ClientHooks.renderSpecificFirstPersonArm(p_117771_, p_117772_, p_117773_, player, HumanoidArm.RIGHT))
+        this.renderHand(p_117771_, p_117772_, p_117773_, p_363694_, this.model.rightArm, p_366898_);
+    }
+
+    /**
+     * @deprecated Neo: use {@link #renderLeftHand(PoseStack, MultiBufferSource, int, ResourceLocation, boolean, AbstractClientPlayer)} instead
+     */
+    @Deprecated
+    public void renderLeftHand(PoseStack p_117814_, MultiBufferSource p_117815_, int p_117816_, ResourceLocation p_361745_, boolean p_366730_) {
+        this.renderLeftHand(p_117814_, p_117815_, p_117816_, p_361745_, p_366730_, net.minecraft.client.Minecraft.getInstance().player);
+    }
+
+    public void renderLeftHand(PoseStack p_117814_, MultiBufferSource p_117815_, int p_117816_, ResourceLocation p_361745_, boolean p_366730_, AbstractClientPlayer player) {
+        if(!net.neoforged.neoforge.client.ClientHooks.renderSpecificFirstPersonArm(p_117814_, p_117815_, p_117816_, player, HumanoidArm.LEFT))
+        this.renderHand(p_117814_, p_117815_, p_117816_, p_361745_, this.model.leftArm, p_366730_);
+    }
+
+    private void renderHand(PoseStack p_117776_, MultiBufferSource p_117777_, int p_117778_, ResourceLocation p_360319_, ModelPart p_117780_, boolean p_366655_) {
         PlayerModel playermodel = this.getModel();
         p_117780_.resetPose();
         p_117780_.visible = true;
@@ -278,7 +_,7 @@
         playermodel.rightSleeve.visible = p_366655_;
         playermodel.leftArm.zRot = -0.1F;
         playermodel.rightArm.zRot = 0.1F;
-        p_440196_.submitModelPart(p_117780_, p_117776_, RenderType.entityTranslucent(p_360319_), p_117778_, OverlayTexture.NO_OVERLAY, null);
+        p_117780_.render(p_117776_, p_117777_.getBuffer(RenderType.entityTranslucent(p_360319_)), p_117778_, OverlayTexture.NO_OVERLAY);
     }
 
     protected void setupRotations(PlayerRenderState p_363355_, PoseStack p_117803_, float p_117804_, float p_117805_) {
