--- a/net/minecraft/client/gui/components/AbstractButton.java
+++ b/net/minecraft/client/gui/components/AbstractButton.java
@@ -38,7 +_,7 @@
             this.getHeight(),
             ARGB.white(this.alpha)
         );
-        int i = ARGB.color(this.alpha, this.active ? -1 : -6250336);
+        int i = ARGB.color(this.alpha, getFGColor());
         this.renderString(p_281670_, minecraft.font, i);
     }
 
@@ -47,13 +_,13 @@
     }
 
     @Override
-    public void onClick(double p_93371_, double p_93372_, boolean p_434377_) {
+    public void onClick(double p_93371_, double p_93372_) {
         this.onPress();
     }
 
     @Override
     public boolean keyPressed(int p_93374_, int p_93375_, int p_93376_) {
-        if (!this.isActive()) {
+        if (!this.active || !this.visible) {
             return false;
         } else if (CommonInputs.selected(p_93374_)) {
             this.playDownSound(Minecraft.getInstance().getSoundManager());
