--- a/net/minecraft/client/gui/components/FocusableTextWidget.java
+++ b/net/minecraft/client/gui/components/FocusableTextWidget.java
@@ -14,7 +_,7 @@
 public class FocusableTextWidget extends MultiLineTextWidget {
     public static final int DEFAULT_PADDING = 4;
     private final boolean alwaysShowBorder;
-    private final FocusableTextWidget.BackgroundFill backgroundFill;
+    private final boolean fillBackground;
     private final int padding;
 
     public FocusableTextWidget(int p_295441_, Component p_296440_, Font p_296307_) {
@@ -22,18 +_,16 @@
     }
 
     public FocusableTextWidget(int p_295671_, Component p_295867_, Font p_294548_, int p_330770_) {
-        this(p_295671_, p_295867_, p_294548_, true, FocusableTextWidget.BackgroundFill.ALWAYS, p_330770_);
+        this(p_295671_, p_295867_, p_294548_, true, true, p_330770_);
     }
 
-    public FocusableTextWidget(
-        int p_331308_, Component p_330271_, Font p_330898_, boolean p_332016_, FocusableTextWidget.BackgroundFill p_440536_, int p_331228_
-    ) {
+    public FocusableTextWidget(int p_331308_, Component p_330271_, Font p_330898_, boolean p_332016_, boolean p_419882_, int p_331228_) {
         super(p_330271_, p_330898_);
         this.setMaxWidth(p_331308_);
         this.setCentered(true);
         this.active = true;
         this.alwaysShowBorder = p_332016_;
-        this.backgroundFill = p_440536_;
+        this.fillBackground = p_419882_;
         this.padding = p_331228_;
     }
 
@@ -53,19 +_,12 @@
         int k = this.getWidth() + this.padding * 2;
         int l = this.getHeight() + this.padding * 2;
         int i1 = ARGB.color(this.alpha, this.alwaysShowBorder ? (this.isFocused() ? -1 : -6250336) : -1);
-        switch (this.backgroundFill) {
-            case ALWAYS:
-                p_296375_.fill(i + 1, j, i + k, j + l, ARGB.color(this.alpha, -16777216));
-                break;
-            case ON_FOCUS:
-                if (this.isFocused()) {
-                    p_296375_.fill(i + 1, j, i + k, j + l, ARGB.color(this.alpha, -16777216));
-                }
-            case NEVER:
+        if (this.fillBackground) {
+            p_296375_.fill(i + 1, j, i + k, j + l, ARGB.color(this.alpha, -16777216));
         }
 
         if (this.isFocused() || this.alwaysShowBorder) {
-            p_296375_.submitOutline(i, j, k, l, i1);
+            p_296375_.renderOutline(i, j, k, l, i1);
         }
 
         super.renderWidget(p_296375_, p_295686_, p_295354_, p_295563_);
@@ -73,12 +_,5 @@
 
     @Override
     public void playDownSound(SoundManager p_295576_) {
-    }
-
-    @OnlyIn(Dist.CLIENT)
-    public static enum BackgroundFill {
-        ALWAYS,
-        ON_FOCUS,
-        NEVER;
     }
 }
