--- a/net/minecraft/client/gui/components/MultiLineTextWidget.java
+++ b/net/minecraft/client/gui/components/MultiLineTextWidget.java
@@ -80,10 +_,9 @@
         int k = 9;
         int l = this.getColor();
         if (this.centered) {
-            int i1 = i + this.getWidth() / 2;
-            multilinelabel.render(p_282535_, MultiLineLabel.Align.CENTER, i1, j, k, true, l);
+            multilinelabel.renderCentered(p_282535_, i + this.getWidth() / 2, j, k, l);
         } else {
-            multilinelabel.render(p_282535_, MultiLineLabel.Align.LEFT, i, j, k, true, l);
+            multilinelabel.renderLeftAligned(p_282535_, i, j, k, l);
         }
 
         if (this.allowHoverComponents) {
@@ -100,16 +_,13 @@
         int i = this.getX();
         int j = this.getY();
         int k = 9;
-        if (this.centered) {
-            int l = i + this.getWidth() / 2;
-            return multilinelabel.getStyle(MultiLineLabel.Align.CENTER, l, j, k, p_428536_, p_428335_);
-        } else {
-            return multilinelabel.getStyle(MultiLineLabel.Align.LEFT, i, j, k, p_428536_, p_428335_);
-        }
+        return this.centered
+            ? multilinelabel.getStyleAtCentered(i + this.getWidth() / 2, j, k, p_428536_, p_428335_)
+            : multilinelabel.getStyleAtLeftAligned(i, j, k, p_428536_, p_428335_);
     }
 
     @Override
-    public void onClick(double p_428318_, double p_428462_, boolean p_434512_) {
+    public void onClick(double p_428318_, double p_428462_) {
         if (this.componentClickHandler != null) {
             Style style = this.getComponentStyleAt(p_428318_, p_428462_);
             if (style != null) {
@@ -118,7 +_,7 @@
             }
         }
 
-        super.onClick(p_428318_, p_428462_, p_434512_);
+        super.onClick(p_428318_, p_428462_);
     }
 
     private MultiLineTextWidget.CacheKey getFreshCacheKey() {
