--- a/net/minecraft/client/gui/components/SpriteIconButton.java
+++ b/net/minecraft/client/gui/components/SpriteIconButton.java
@@ -11,7 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public abstract class SpriteIconButton extends Button {
-    protected final WidgetSprites sprite;
+    protected final ResourceLocation sprite;
     protected final int spriteWidth;
     protected final int spriteHeight;
 
@@ -21,19 +_,14 @@
         Component p_294760_,
         int p_296342_,
         int p_296291_,
-        WidgetSprites p_440029_,
+        ResourceLocation p_294131_,
         Button.OnPress p_295667_,
-        @Nullable Component p_440489_,
         @Nullable Button.CreateNarration p_332100_
     ) {
         super(0, 0, p_294218_, p_294461_, p_294760_, p_295667_, p_332100_ == null ? DEFAULT_NARRATION : p_332100_);
-        if (p_440489_ != null) {
-            this.setTooltip(Tooltip.create(p_440489_));
-        }
-
         this.spriteWidth = p_296342_;
         this.spriteHeight = p_296291_;
-        this.sprite = p_440029_;
+        this.sprite = p_294131_;
     }
 
     public static SpriteIconButton.Builder builder(Component p_294639_, Button.OnPress p_295155_, boolean p_295622_) {
@@ -48,13 +_,11 @@
         private int width = 150;
         private int height = 20;
         @Nullable
-        private WidgetSprites sprite;
+        private ResourceLocation sprite;
         private int spriteWidth;
         private int spriteHeight;
         @Nullable
-        private Component tooltip;
-        @Nullable
-        private Button.CreateNarration narration;
+        Button.CreateNarration narration;
 
         public Builder(Component p_294568_, Button.OnPress p_294520_, boolean p_294559_) {
             this.message = p_294568_;
@@ -74,24 +_,12 @@
         }
 
         public SpriteIconButton.Builder sprite(ResourceLocation p_295718_, int p_296046_, int p_295188_) {
-            this.sprite = new WidgetSprites(p_295718_);
+            this.sprite = p_295718_;
             this.spriteWidth = p_296046_;
             this.spriteHeight = p_295188_;
             return this;
         }
 
-        public SpriteIconButton.Builder sprite(WidgetSprites p_439332_, int p_440349_, int p_440074_) {
-            this.sprite = p_439332_;
-            this.spriteWidth = p_440349_;
-            this.spriteHeight = p_440074_;
-            return this;
-        }
-
-        public SpriteIconButton.Builder withTootip() {
-            this.tooltip = this.message;
-            return this;
-        }
-
         public SpriteIconButton.Builder narration(Button.CreateNarration p_331295_) {
             this.narration = p_331295_;
             return this;
@@ -103,10 +_,10 @@
             } else {
                 return (SpriteIconButton)(this.iconOnly
                     ? new SpriteIconButton.CenteredIcon(
-                        this.width, this.height, this.message, this.spriteWidth, this.spriteHeight, this.sprite, this.onPress, this.tooltip, this.narration
+                        this.width, this.height, this.message, this.spriteWidth, this.spriteHeight, this.sprite, this.onPress, this.narration
                     )
                     : new SpriteIconButton.TextAndIcon(
-                        this.width, this.height, this.message, this.spriteWidth, this.spriteHeight, this.sprite, this.onPress, this.tooltip, this.narration
+                        this.width, this.height, this.message, this.spriteWidth, this.spriteHeight, this.sprite, this.onPress, this.narration
                     ));
             }
         }
@@ -120,12 +_,11 @@
             Component p_295609_,
             int p_294922_,
             int p_296462_,
-            WidgetSprites p_440019_,
+            ResourceLocation p_295554_,
             Button.OnPress p_294427_,
-            @Nullable Component p_440407_,
             @Nullable Button.CreateNarration p_330653_
         ) {
-            super(p_295914_, p_294852_, p_295609_, p_294922_, p_296462_, p_440019_, p_294427_, p_440407_, p_330653_);
+            super(p_295914_, p_294852_, p_295609_, p_294922_, p_296462_, p_295554_, p_294427_, p_330653_);
         }
 
         @Override
@@ -133,15 +_,7 @@
             super.renderWidget(p_295402_, p_295733_, p_294839_, p_296191_);
             int i = this.getX() + this.getWidth() / 2 - this.spriteWidth / 2;
             int j = this.getY() + this.getHeight() / 2 - this.spriteHeight / 2;
-            p_295402_.blitSprite(
-                RenderPipelines.GUI_TEXTURED,
-                this.sprite.get(this.isActive(), this.isHoveredOrFocused()),
-                i,
-                j,
-                this.spriteWidth,
-                this.spriteHeight,
-                this.alpha
-            );
+            p_295402_.blitSprite(RenderPipelines.GUI_TEXTURED, this.sprite, i, j, this.spriteWidth, this.spriteHeight, this.alpha);
         }
 
         @Override
@@ -157,12 +_,11 @@
             Component p_296265_,
             int p_294900_,
             int p_295900_,
-            WidgetSprites p_439719_,
+            ResourceLocation p_296097_,
             Button.OnPress p_295566_,
-            @Nullable Component p_439158_,
             @Nullable Button.CreateNarration p_330735_
         ) {
-            super(p_296442_, p_294340_, p_296265_, p_294900_, p_295900_, p_439719_, p_295566_, p_439158_, p_330735_);
+            super(p_296442_, p_294340_, p_296265_, p_294900_, p_295900_, p_296097_, p_295566_, p_330735_);
         }
 
         @Override
@@ -170,15 +_,7 @@
             super.renderWidget(p_294156_, p_295818_, p_294994_, p_296436_);
             int i = this.getX() + this.getWidth() - this.spriteWidth - 2;
             int j = this.getY() + this.getHeight() / 2 - this.spriteHeight / 2;
-            p_294156_.blitSprite(
-                RenderPipelines.GUI_TEXTURED,
-                this.sprite.get(this.isActive(), this.isHoveredOrFocused()),
-                i,
-                j,
-                this.spriteWidth,
-                this.spriteHeight,
-                this.alpha
-            );
+            p_294156_.blitSprite(RenderPipelines.GUI_TEXTURED, this.sprite, i, j, this.spriteWidth, this.spriteHeight, this.alpha);
         }
 
         @Override
