--- a/net/minecraft/client/multiplayer/chat/ChatListener.java
+++ b/net/minecraft/client/multiplayer/chat/ChatListener.java
@@ -35,21 +_,12 @@
     }
 
     public void tick() {
-        if (this.minecraft.isPaused()) {
-            if (this.messageDelay > 0L) {
-                this.previousMessageTime += 50L;
-            }
-        } else {
-            if (this.messageDelay == 0L) {
-                if (!this.delayedMessageQueue.isEmpty()) {
-                    this.flushQueue();
-                }
-            } else {
-                ChatListener.Message chatlistener$message;
-                if (Util.getMillis() >= this.previousMessageTime + this.messageDelay) {
-                    do {
-                        chatlistener$message = this.delayedMessageQueue.poll();
-                    } while (chatlistener$message != null && !chatlistener$message.accept());
+        if (this.messageDelay != 0L) {
+            if (Util.getMillis() >= this.previousMessageTime + this.messageDelay) {
+                ChatListener.Message chatlistener$message = this.delayedMessageQueue.poll();
+
+                while (chatlistener$message != null && !chatlistener$message.accept()) {
+                    chatlistener$message = this.delayedMessageQueue.poll();
                 }
             }
         }
@@ -57,8 +_,9 @@
 
     public void setMessageDelay(double p_240785_) {
         long i = (long)(p_240785_ * 1000.0);
-        if (i == 0L && this.messageDelay > 0L && !this.minecraft.isPaused()) {
-            this.flushQueue();
+        if (i == 0L && this.messageDelay > 0L) {
+            this.delayedMessageQueue.forEach(ChatListener.Message::accept);
+            this.delayedMessageQueue.clear();
         }
 
         this.messageDelay = i;
@@ -72,10 +_,9 @@
         return this.delayedMessageQueue.size();
     }
 
-    public void flushQueue() {
+    public void clearQueue() {
         this.delayedMessageQueue.forEach(ChatListener.Message::accept);
         this.delayedMessageQueue.clear();
-        this.previousMessageTime = 0L;
     }
 
     public boolean removeFromDelayedMessageQueue(MessageSignature p_241445_) {
@@ -133,7 +_,9 @@
         Instant instant = Instant.now();
         this.handleMessage(null, () -> {
             Component component = p_251256_.decorate(p_250375_);
-            this.minecraft.gui.getChat().addMessage(component);
+            Component forgeComponent = net.neoforged.neoforge.client.ClientHooks.onClientChat(p_251256_, component, Util.NIL_UUID);
+            if (forgeComponent == null) return false;
+            this.minecraft.gui.getChat().addMessage(forgeComponent);
             this.narrateChatMessage(p_251256_, p_250375_);
             this.logSystemMessage(component, instant);
             this.previousMessageTime = Util.getMillis();
@@ -152,17 +_,21 @@
             MessageSignature messagesignature = p_249430_.signature();
             FilterMask filtermask = p_249430_.filterMask();
             if (filtermask.isEmpty()) {
-                this.minecraft.gui.getChat().addMessage(p_249231_, messagesignature, guimessagetag);
+                Component forgeComponent = net.neoforged.neoforge.client.ClientHooks.onClientPlayerChat(p_251766_, p_249231_, p_249430_, p_249430_.sender());
+                if (forgeComponent == null) return false;
+                this.minecraft.gui.getChat().addMessage(forgeComponent, messagesignature, guimessagetag);
                 this.narrateChatMessage(p_251766_, p_249430_.decoratedContent());
             } else {
                 Component component = filtermask.applyWithFormatting(p_249430_.signedContent());
                 if (component != null) {
-                    this.minecraft.gui.getChat().addMessage(p_251766_.decorate(component), messagesignature, guimessagetag);
+                    Component forgeComponent = net.neoforged.neoforge.client.ClientHooks.onClientPlayerChat(p_251766_, p_251766_.decorate(component), p_249430_, p_249430_.sender());
+                    if (forgeComponent == null) return false;
+                    this.minecraft.gui.getChat().addMessage(forgeComponent, messagesignature, guimessagetag);
                     this.narrateChatMessage(p_251766_, component);
                 }
             }
 
-            this.logPlayerMessage(p_249430_, p_249177_, chattrustlevel);
+            this.logPlayerMessage(p_249430_, p_251766_, p_249177_, chattrustlevel);
             this.previousMessageTime = Util.getMillis();
             return true;
         } else {
@@ -178,7 +_,7 @@
         return this.isSenderLocalPlayer(p_251246_.sender()) ? ChatTrustLevel.SECURE : ChatTrustLevel.evaluate(p_251246_, p_250576_, p_249995_);
     }
 
-    private void logPlayerMessage(PlayerChatMessage p_252155_, GameProfile p_248589_, ChatTrustLevel p_248881_) {
+    private void logPlayerMessage(PlayerChatMessage p_252155_, ChatType.Bound p_249730_, GameProfile p_248589_, ChatTrustLevel p_248881_) {
         ChatLog chatlog = this.minecraft.getReportingContext().chatLog();
         chatlog.push(LoggedChatMessage.player(p_248589_, p_252155_, p_248881_));
     }
@@ -190,6 +_,8 @@
 
     public void handleSystemMessage(Component p_240522_, boolean p_240642_) {
         if (!this.minecraft.options.hideMatchedNames().get() || !this.minecraft.isBlocked(this.guessChatUUID(p_240522_))) {
+            p_240522_ = net.neoforged.neoforge.client.ClientHooks.onClientSystemChat(p_240522_, p_240642_);
+            if (p_240522_ == null) return;
             if (p_240642_) {
                 this.minecraft.gui.setOverlayMessage(p_240522_, false);
                 this.minecraft.getNarrator().saySystemQueued(p_240522_);
