--- a/net/minecraft/world/entity/EntityType.java
+++ b/net/minecraft/world/entity/EntityType.java
@@ -20,10 +_,7 @@
 import net.minecraft.core.registries.BuiltInRegistries;
 import net.minecraft.core.registries.Registries;
 import net.minecraft.nbt.CompoundTag;
-import net.minecraft.network.RegistryFriendlyByteBuf;
 import net.minecraft.network.chat.Component;
-import net.minecraft.network.codec.ByteBufCodecs;
-import net.minecraft.network.codec.StreamCodec;
 import net.minecraft.resources.DependantName;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.resources.ResourceLocation;
@@ -60,7 +_,6 @@
 import net.minecraft.world.entity.animal.armadillo.Armadillo;
 import net.minecraft.world.entity.animal.axolotl.Axolotl;
 import net.minecraft.world.entity.animal.camel.Camel;
-import net.minecraft.world.entity.animal.coppergolem.CopperGolem;
 import net.minecraft.world.entity.animal.frog.Frog;
 import net.minecraft.world.entity.animal.frog.Tadpole;
 import net.minecraft.world.entity.animal.goat.Goat;
@@ -166,7 +_,7 @@
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.Items;
-import net.minecraft.world.item.component.TypedEntityData;
+import net.minecraft.world.item.component.CustomData;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.LevelReader;
 import net.minecraft.world.level.block.Block;
@@ -187,7 +_,6 @@
     private static final Logger LOGGER = LogUtils.getLogger();
     private final Holder.Reference<EntityType<?>> builtInRegistryHolder = BuiltInRegistries.ENTITY_TYPE.createIntrusiveHolder(this);
     public static final Codec<EntityType<?>> CODEC = BuiltInRegistries.ENTITY_TYPE.byNameCodec();
-    public static final StreamCodec<RegistryFriendlyByteBuf, EntityType<?>> STREAM_CODEC = ByteBufCodecs.registry(Registries.ENTITY_TYPE);
     private static final float MAGIC_HORSE_WIDTH = 1.3964844F;
     private static final int DISPLAY_TRACKING_RANGE = 10;
     public static final EntityType<Boat> ACACIA_BOAT = register(
@@ -276,18 +_,17 @@
             .clientTrackingRange(10)
     );
     public static final EntityType<Blaze> BLAZE = register(
-        "blaze", EntityType.Builder.of(Blaze::new, MobCategory.MONSTER).fireImmune().sized(0.6F, 1.8F).clientTrackingRange(8).notInPeaceful()
+        "blaze", EntityType.Builder.of(Blaze::new, MobCategory.MONSTER).fireImmune().sized(0.6F, 1.8F).clientTrackingRange(8)
     );
     public static final EntityType<Display.BlockDisplay> BLOCK_DISPLAY = register(
         "block_display",
         EntityType.Builder.of(Display.BlockDisplay::new, MobCategory.MISC).noLootTable().sized(0.0F, 0.0F).clientTrackingRange(10).updateInterval(1)
     );
     public static final EntityType<Bogged> BOGGED = register(
-        "bogged",
-        EntityType.Builder.of(Bogged::new, MobCategory.MONSTER).sized(0.6F, 1.99F).eyeHeight(1.74F).ridingOffset(-0.7F).clientTrackingRange(8).notInPeaceful()
+        "bogged", EntityType.Builder.of(Bogged::new, MobCategory.MONSTER).sized(0.6F, 1.99F).eyeHeight(1.74F).ridingOffset(-0.7F).clientTrackingRange(8)
     );
     public static final EntityType<Breeze> BREEZE = register(
-        "breeze", EntityType.Builder.of(Breeze::new, MobCategory.MONSTER).sized(0.6F, 1.77F).eyeHeight(1.3452F).clientTrackingRange(10).notInPeaceful()
+        "breeze", EntityType.Builder.of(Breeze::new, MobCategory.MONSTER).sized(0.6F, 1.77F).eyeHeight(1.3452F).clientTrackingRange(10)
     );
     public static final EntityType<BreezeWindCharge> BREEZE_WIND_CHARGE = register(
         "breeze_wind_charge",
@@ -305,7 +_,7 @@
         "cat", EntityType.Builder.of(Cat::new, MobCategory.CREATURE).sized(0.6F, 0.7F).eyeHeight(0.35F).passengerAttachments(0.5125F).clientTrackingRange(8)
     );
     public static final EntityType<CaveSpider> CAVE_SPIDER = register(
-        "cave_spider", EntityType.Builder.of(CaveSpider::new, MobCategory.MONSTER).sized(0.7F, 0.5F).eyeHeight(0.45F).clientTrackingRange(8).notInPeaceful()
+        "cave_spider", EntityType.Builder.of(CaveSpider::new, MobCategory.MONSTER).sized(0.7F, 0.5F).eyeHeight(0.45F).clientTrackingRange(8)
     );
     public static final EntityType<Boat> CHERRY_BOAT = register(
         "cherry_boat",
@@ -338,9 +_,6 @@
     public static final EntityType<Cod> COD = register(
         "cod", EntityType.Builder.of(Cod::new, MobCategory.WATER_AMBIENT).sized(0.5F, 0.3F).eyeHeight(0.195F).clientTrackingRange(4)
     );
-    public static final EntityType<CopperGolem> COPPER_GOLEM = register(
-        "copper_golem", EntityType.Builder.of(CopperGolem::new, MobCategory.MISC).sized(0.6F, 1.01F).eyeHeight(0.8125F).clientTrackingRange(10)
-    );
     public static final EntityType<MinecartCommandBlock> COMMAND_BLOCK_MINECART = register(
         "command_block_minecart",
         EntityType.Builder.of(MinecartCommandBlock::new, MobCategory.MISC)
@@ -353,10 +_,10 @@
         "cow", EntityType.Builder.of(Cow::new, MobCategory.CREATURE).sized(0.9F, 1.4F).eyeHeight(1.3F).passengerAttachments(1.36875F).clientTrackingRange(10)
     );
     public static final EntityType<Creaking> CREAKING = register(
-        "creaking", EntityType.Builder.of(Creaking::new, MobCategory.MONSTER).sized(0.9F, 2.7F).eyeHeight(2.3F).clientTrackingRange(8).notInPeaceful()
+        "creaking", EntityType.Builder.of(Creaking::new, MobCategory.MONSTER).sized(0.9F, 2.7F).eyeHeight(2.3F).clientTrackingRange(8)
     );
     public static final EntityType<Creeper> CREEPER = register(
-        "creeper", EntityType.Builder.of(Creeper::new, MobCategory.MONSTER).sized(0.6F, 1.7F).clientTrackingRange(8).notInPeaceful()
+        "creeper", EntityType.Builder.of(Creeper::new, MobCategory.MONSTER).sized(0.6F, 1.7F).clientTrackingRange(8)
     );
     public static final EntityType<Boat> DARK_OAK_BOAT = register(
         "dark_oak_boat",
@@ -397,7 +_,6 @@
             .passengerAttachments(2.0125F)
             .ridingOffset(-0.7F)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<ThrownEgg> EGG = register(
         "egg", EntityType.Builder.<ThrownEgg>of(ThrownEgg::new, MobCategory.MISC).noLootTable().sized(0.25F, 0.25F).clientTrackingRange(4).updateInterval(10)
@@ -409,25 +_,14 @@
             .eyeHeight(0.99875F)
             .passengerAttachments(2.350625F)
             .clientTrackingRange(10)
-            .notInPeaceful()
     );
     public static final EntityType<EnderMan> ENDERMAN = register(
         "enderman",
-        EntityType.Builder.of(EnderMan::new, MobCategory.MONSTER)
-            .sized(0.6F, 2.9F)
-            .eyeHeight(2.55F)
-            .passengerAttachments(2.80625F)
-            .clientTrackingRange(8)
-            .notInPeaceful()
+        EntityType.Builder.of(EnderMan::new, MobCategory.MONSTER).sized(0.6F, 2.9F).eyeHeight(2.55F).passengerAttachments(2.80625F).clientTrackingRange(8)
     );
     public static final EntityType<Endermite> ENDERMITE = register(
         "endermite",
-        EntityType.Builder.of(Endermite::new, MobCategory.MONSTER)
-            .sized(0.4F, 0.3F)
-            .eyeHeight(0.13F)
-            .passengerAttachments(0.2375F)
-            .clientTrackingRange(8)
-            .notInPeaceful()
+        EntityType.Builder.of(Endermite::new, MobCategory.MONSTER).sized(0.4F, 0.3F).eyeHeight(0.13F).passengerAttachments(0.2375F).clientTrackingRange(8)
     );
     public static final EntityType<EnderDragon> ENDER_DRAGON = register(
         "ender_dragon",
@@ -520,7 +_,6 @@
             .passengerAttachments(4.0625F)
             .ridingOffset(0.5F)
             .clientTrackingRange(10)
-            .notInPeaceful()
     );
     public static final EntityType<HappyGhast> HAPPY_GHAST = register(
         "happy_ghast",
@@ -532,13 +_,7 @@
             .clientTrackingRange(10)
     );
     public static final EntityType<Giant> GIANT = register(
-        "giant",
-        EntityType.Builder.of(Giant::new, MobCategory.MONSTER)
-            .sized(3.6F, 12.0F)
-            .eyeHeight(10.44F)
-            .ridingOffset(-3.75F)
-            .clientTrackingRange(10)
-            .notInPeaceful()
+        "giant", EntityType.Builder.of(Giant::new, MobCategory.MONSTER).sized(3.6F, 12.0F).eyeHeight(10.44F).ridingOffset(-3.75F).clientTrackingRange(10)
     );
     public static final EntityType<GlowItemFrame> GLOW_ITEM_FRAME = register(
         "glow_item_frame",
@@ -557,12 +_,7 @@
     );
     public static final EntityType<Guardian> GUARDIAN = register(
         "guardian",
-        EntityType.Builder.of(Guardian::new, MobCategory.MONSTER)
-            .sized(0.85F, 0.85F)
-            .eyeHeight(0.425F)
-            .passengerAttachments(0.975F)
-            .clientTrackingRange(8)
-            .notInPeaceful()
+        EntityType.Builder.of(Guardian::new, MobCategory.MONSTER).sized(0.85F, 0.85F).eyeHeight(0.425F).passengerAttachments(0.975F).clientTrackingRange(8)
     );
     public static final EntityType<Hoglin> HOGLIN = register(
         "hoglin", EntityType.Builder.of(Hoglin::new, MobCategory.MONSTER).sized(1.3964844F, 1.4F).passengerAttachments(1.49375F).clientTrackingRange(8)
@@ -583,16 +_,10 @@
             .passengerAttachments(2.075F)
             .ridingOffset(-0.7F)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<Illusioner> ILLUSIONER = register(
         "illusioner",
-        EntityType.Builder.of(Illusioner::new, MobCategory.MONSTER)
-            .sized(0.6F, 1.95F)
-            .passengerAttachments(2.0F)
-            .ridingOffset(-0.6F)
-            .clientTrackingRange(8)
-            .notInPeaceful()
+        EntityType.Builder.of(Illusioner::new, MobCategory.MONSTER).sized(0.6F, 1.95F).passengerAttachments(2.0F).ridingOffset(-0.6F).clientTrackingRange(8)
     );
     public static final EntityType<Interaction> INTERACTION = register(
         "interaction", EntityType.Builder.of(Interaction::new, MobCategory.MISC).noLootTable().sized(0.0F, 0.0F).clientTrackingRange(10)
@@ -677,7 +_,6 @@
             .eyeHeight(0.325F)
             .spawnDimensionsScale(4.0F)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<Boat> MANGROVE_BOAT = register(
         "mangrove_boat",
@@ -771,7 +_,6 @@
             .passengerAttachments(0.3375F)
             .ridingOffset(-0.125F)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<Pig> PIG = register(
         "pig", EntityType.Builder.of(Pig::new, MobCategory.CREATURE).sized(0.9F, 0.9F).passengerAttachments(0.86875F).clientTrackingRange(10)
@@ -793,7 +_,6 @@
             .passengerAttachments(2.0125F)
             .ridingOffset(-0.7F)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<Pillager> PILLAGER = register(
         "pillager",
@@ -803,7 +_,6 @@
             .passengerAttachments(2.0F)
             .ridingOffset(-0.6F)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<PolarBear> POLAR_BEAR = register(
         "polar_bear", EntityType.Builder.of(PolarBear::new, MobCategory.CREATURE).immuneTo(Blocks.POWDER_SNOW).sized(1.4F, 1.4F).clientTrackingRange(10)
@@ -836,7 +_,6 @@
             .sized(1.95F, 2.2F)
             .passengerAttachments(new Vec3(0.0, 2.2625, -0.0625))
             .clientTrackingRange(10)
-            .notInPeaceful()
     );
     public static final EntityType<Salmon> SALMON = register(
         "salmon", EntityType.Builder.of(Salmon::new, MobCategory.WATER_AMBIENT).sized(0.7F, 0.4F).eyeHeight(0.26F).clientTrackingRange(4)
@@ -855,21 +_,10 @@
     );
     public static final EntityType<Silverfish> SILVERFISH = register(
         "silverfish",
-        EntityType.Builder.of(Silverfish::new, MobCategory.MONSTER)
-            .sized(0.4F, 0.3F)
-            .eyeHeight(0.13F)
-            .passengerAttachments(0.2375F)
-            .clientTrackingRange(8)
-            .notInPeaceful()
+        EntityType.Builder.of(Silverfish::new, MobCategory.MONSTER).sized(0.4F, 0.3F).eyeHeight(0.13F).passengerAttachments(0.2375F).clientTrackingRange(8)
     );
     public static final EntityType<Skeleton> SKELETON = register(
-        "skeleton",
-        EntityType.Builder.of(Skeleton::new, MobCategory.MONSTER)
-            .sized(0.6F, 1.99F)
-            .eyeHeight(1.74F)
-            .ridingOffset(-0.7F)
-            .clientTrackingRange(8)
-            .notInPeaceful()
+        "skeleton", EntityType.Builder.of(Skeleton::new, MobCategory.MONSTER).sized(0.6F, 1.99F).eyeHeight(1.74F).ridingOffset(-0.7F).clientTrackingRange(8)
     );
     public static final EntityType<SkeletonHorse> SKELETON_HORSE = register(
         "skeleton_horse",
@@ -881,12 +_,7 @@
     );
     public static final EntityType<Slime> SLIME = register(
         "slime",
-        EntityType.Builder.of(Slime::new, MobCategory.MONSTER)
-            .sized(0.52F, 0.52F)
-            .eyeHeight(0.325F)
-            .spawnDimensionsScale(4.0F)
-            .clientTrackingRange(10)
-            .notInPeaceful()
+        EntityType.Builder.of(Slime::new, MobCategory.MONSTER).sized(0.52F, 0.52F).eyeHeight(0.325F).spawnDimensionsScale(4.0F).clientTrackingRange(10)
     );
     public static final EntityType<SmallFireball> SMALL_FIREBALL = register(
         "small_fireball",
@@ -928,12 +_,7 @@
     );
     public static final EntityType<Spider> SPIDER = register(
         "spider",
-        EntityType.Builder.of(Spider::new, MobCategory.MONSTER)
-            .sized(1.4F, 0.9F)
-            .eyeHeight(0.65F)
-            .passengerAttachments(0.765F)
-            .clientTrackingRange(8)
-            .notInPeaceful()
+        EntityType.Builder.of(Spider::new, MobCategory.MONSTER).sized(1.4F, 0.9F).eyeHeight(0.65F).passengerAttachments(0.765F).clientTrackingRange(8)
     );
     public static final EntityType<Boat> SPRUCE_BOAT = register(
         "spruce_boat",
@@ -962,7 +_,6 @@
             .ridingOffset(-0.7F)
             .immuneTo(Blocks.POWDER_SNOW)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<Strider> STRIDER = register(
         "strider", EntityType.Builder.of(Strider::new, MobCategory.CREATURE).fireImmune().sized(0.9F, 1.7F).clientTrackingRange(10)
@@ -1021,19 +_,13 @@
             .passengerAttachments(0.7375F)
             .ridingOffset(0.04F)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<Villager> VILLAGER = register(
         "villager", EntityType.Builder.<Villager>of(Villager::new, MobCategory.MISC).sized(0.6F, 1.95F).eyeHeight(1.62F).clientTrackingRange(10)
     );
     public static final EntityType<Vindicator> VINDICATOR = register(
         "vindicator",
-        EntityType.Builder.of(Vindicator::new, MobCategory.MONSTER)
-            .sized(0.6F, 1.95F)
-            .passengerAttachments(2.0F)
-            .ridingOffset(-0.6F)
-            .clientTrackingRange(8)
-            .notInPeaceful()
+        EntityType.Builder.of(Vindicator::new, MobCategory.MONSTER).sized(0.6F, 1.95F).passengerAttachments(2.0F).ridingOffset(-0.6F).clientTrackingRange(8)
     );
     public static final EntityType<WanderingTrader> WANDERING_TRADER = register(
         "wandering_trader", EntityType.Builder.of(WanderingTrader::new, MobCategory.CREATURE).sized(0.6F, 1.95F).eyeHeight(1.62F).clientTrackingRange(10)
@@ -1046,7 +_,6 @@
             .attach(EntityAttachment.WARDEN_CHEST, 0.0F, 1.6F, 0.0F)
             .clientTrackingRange(16)
             .fireImmune()
-            .notInPeaceful()
     );
     public static final EntityType<WindCharge> WIND_CHARGE = register(
         "wind_charge",
@@ -1059,21 +_,11 @@
     );
     public static final EntityType<Witch> WITCH = register(
         "witch",
-        EntityType.Builder.of(Witch::new, MobCategory.MONSTER)
-            .sized(0.6F, 1.95F)
-            .eyeHeight(1.62F)
-            .passengerAttachments(2.2625F)
-            .clientTrackingRange(8)
-            .notInPeaceful()
+        EntityType.Builder.of(Witch::new, MobCategory.MONSTER).sized(0.6F, 1.95F).eyeHeight(1.62F).passengerAttachments(2.2625F).clientTrackingRange(8)
     );
     public static final EntityType<WitherBoss> WITHER = register(
         "wither",
-        EntityType.Builder.of(WitherBoss::new, MobCategory.MONSTER)
-            .fireImmune()
-            .immuneTo(Blocks.WITHER_ROSE)
-            .sized(0.9F, 3.5F)
-            .clientTrackingRange(10)
-            .notInPeaceful()
+        EntityType.Builder.of(WitherBoss::new, MobCategory.MONSTER).fireImmune().immuneTo(Blocks.WITHER_ROSE).sized(0.9F, 3.5F).clientTrackingRange(10)
     );
     public static final EntityType<WitherSkeleton> WITHER_SKELETON = register(
         "wither_skeleton",
@@ -1084,7 +_,6 @@
             .eyeHeight(2.1F)
             .ridingOffset(-0.875F)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<WitherSkull> WITHER_SKULL = register(
         "wither_skull",
@@ -1100,12 +_,7 @@
     );
     public static final EntityType<Zoglin> ZOGLIN = register(
         "zoglin",
-        EntityType.Builder.of(Zoglin::new, MobCategory.MONSTER)
-            .fireImmune()
-            .sized(1.3964844F, 1.4F)
-            .passengerAttachments(1.49375F)
-            .clientTrackingRange(8)
-            .notInPeaceful()
+        EntityType.Builder.of(Zoglin::new, MobCategory.MONSTER).fireImmune().sized(1.3964844F, 1.4F).passengerAttachments(1.49375F).clientTrackingRange(8)
     );
     public static final EntityType<Zombie> ZOMBIE = register(
         "zombie",
@@ -1115,7 +_,6 @@
             .passengerAttachments(2.0125F)
             .ridingOffset(-0.7F)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<ZombieHorse> ZOMBIE_HORSE = register(
         "zombie_horse",
@@ -1133,7 +_,6 @@
             .ridingOffset(-0.7F)
             .eyeHeight(1.74F)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<ZombifiedPiglin> ZOMBIFIED_PIGLIN = register(
         "zombified_piglin",
@@ -1144,7 +_,6 @@
             .passengerAttachments(2.0F)
             .ridingOffset(-0.7F)
             .clientTrackingRange(8)
-            .notInPeaceful()
     );
     public static final EntityType<Player> PLAYER = register(
         "player",
@@ -1184,7 +_,11 @@
     private final EntityDimensions dimensions;
     private final float spawnDimensionsScale;
     private final FeatureFlagSet requiredFeatures;
-    private final boolean allowedInPeaceful;
+
+    private final java.util.function.Predicate<EntityType<?>> trackDeltasSupplier;
+    private final java.util.function.ToIntFunction<EntityType<?>> trackingRangeSupplier;
+    private final java.util.function.ToIntFunction<EntityType<?>> updateIntervalSupplier;
+    private final boolean onlyOpCanSetNbt;
 
     private static <T extends Entity> EntityType<T> register(ResourceKey<EntityType<?>> p_368669_, EntityType.Builder<T> p_368714_) {
         return Registry.register(BuiltInRegistries.ENTITY_TYPE, p_368669_, p_368714_.build(p_368669_));
@@ -1220,8 +_,30 @@
         int p_273451_,
         String p_368582_,
         Optional<ResourceKey<LootTable>> p_368696_,
+        FeatureFlagSet p_273518_
+    ) {
+        this(p_273268_, p_272918_, p_273417_, p_273389_, p_273556_, p_272654_, p_273631_, p_272946_, p_338404_, p_272895_, p_273451_, p_368582_, p_368696_, p_273518_, EntityType::defaultTrackDeltasSupplier, EntityType::defaultTrackingRangeSupplier, EntityType::defaultUpdateIntervalSupplier, false);
+    }
+
+    public EntityType(
+        EntityType.EntityFactory<T> p_273268_,
+        MobCategory p_272918_,
+        boolean p_273417_,
+        boolean p_273389_,
+        boolean p_273556_,
+        boolean p_272654_,
+        ImmutableSet<Block> p_273631_,
+        EntityDimensions p_272946_,
+        float p_338404_,
+        int p_272895_,
+        int p_273451_,
+        String p_368582_,
+        Optional<ResourceKey<LootTable>> p_368696_,
         FeatureFlagSet p_273518_,
-        boolean p_434970_
+        final java.util.function.Predicate<EntityType<?>> trackDeltasSupplier,
+        final java.util.function.ToIntFunction<EntityType<?>> trackingRangeSupplier,
+        final java.util.function.ToIntFunction<EntityType<?>> updateIntervalSupplier,
+        boolean onlyOpCanSetNbt
     ) {
         this.factory = p_273268_;
         this.category = p_272918_;
@@ -1237,7 +_,10 @@
         this.descriptionId = p_368582_;
         this.lootTable = p_368696_;
         this.requiredFeatures = p_273518_;
-        this.allowedInPeaceful = p_434970_;
+        this.trackDeltasSupplier = trackDeltasSupplier;
+        this.trackingRangeSupplier = trackingRangeSupplier;
+        this.updateIntervalSupplier = updateIntervalSupplier;
+        this.onlyOpCanSetNbt = onlyOpCanSetNbt;
     }
 
     @Nullable
@@ -1277,8 +_,8 @@
     public static <T extends Entity> Consumer<T> appendCustomEntityStackConfig(
         Consumer<T> p_263579_, Level p_366735_, ItemStack p_263582_, @Nullable LivingEntity p_394266_
     ) {
-        TypedEntityData<EntityType<?>> typedentitydata = p_263582_.get(DataComponents.ENTITY_DATA);
-        return typedentitydata != null ? p_263579_.andThen(p_432524_ -> updateCustomEntityTag(p_366735_, p_394266_, p_432524_, typedentitydata)) : p_263579_;
+        CustomData customdata = p_263582_.getOrDefault(DataComponents.ENTITY_DATA, CustomData.EMPTY);
+        return !customdata.isEmpty() ? p_263579_.andThen(p_392975_ -> updateCustomEntityTag(p_366735_, p_394266_, p_392975_, customdata)) : p_263579_;
     }
 
     @Nullable
@@ -1322,6 +_,13 @@
                 mob.yHeadRot = mob.getYRot();
                 mob.yBodyRot = mob.getYRot();
                 mob.finalizeSpawn(p_262637_, p_262637_.getCurrentDifficultyAt(mob.blockPosition()), p_360546_, null);
+                if (mob.isSpawnCancelled()) {
+                    // Neo: Discard mob, spawn was cancelled
+                    mob.discard();
+                    // return null, mob was killed, context should be lost
+                    // fixes llamas for wandering trader spawning if wandering trader was cancelled
+                    return null;
+                }
             }
 
             if (p_262629_ != null) {
@@ -1342,16 +_,15 @@
         return 1.0 + Shapes.collide(Direction.Axis.Y, p_20629_, iterable, p_20628_ ? -2.0 : -1.0);
     }
 
-    public static void updateCustomEntityTag(
-        Level p_20621_, @Nullable LivingEntity p_394374_, @Nullable Entity p_20623_, TypedEntityData<EntityType<?>> p_432820_
-    ) {
+    public static void updateCustomEntityTag(Level p_20621_, @Nullable LivingEntity p_394374_, @Nullable Entity p_20623_, CustomData p_331752_) {
         MinecraftServer minecraftserver = p_20621_.getServer();
         if (minecraftserver != null && p_20623_ != null) {
-            if (p_20623_.getType() == p_432820_.type()) {
-                if (p_20621_.isClientSide()
+            EntityType<?> entitytype = p_331752_.parseEntityType(minecraftserver.registryAccess(), Registries.ENTITY_TYPE);
+            if (p_20623_.getType() == entitytype) {
+                if (p_20621_.isClientSide
                     || !p_20623_.getType().onlyOpCanSetNbt()
-                    || p_394374_ instanceof Player player && minecraftserver.getPlayerList().isOp(player.nameAndId())) {
-                    p_432820_.loadInto(p_20623_);
+                    || p_394374_ instanceof Player player && minecraftserver.getPlayerList().isOp(player.getGameProfile())) {
+                    p_331752_.loadInto(p_20623_);
                 }
             }
         }
@@ -1429,12 +_,6 @@
         );
     }
 
-    public static Optional<Entity> create(EntityType<?> p_435889_, ValueInput p_434459_, Level p_433549_, EntitySpawnReason p_434270_) {
-        Optional<Entity> optional = Optional.ofNullable(p_435889_.create(p_433549_, p_434270_));
-        optional.ifPresent(p_432510_ -> p_432510_.load(p_434459_));
-        return optional;
-    }
-
     public AABB getSpawnAABB(double p_338403_, double p_338190_, double p_338227_) {
         float f = this.spawnDimensionsScale * this.getWidth() / 2.0F;
         float f1 = this.spawnDimensionsScale * this.getHeight();
@@ -1472,48 +_,17 @@
     }
 
     @Nullable
-    public static Entity loadEntityRecursive(
-        EntityType<?> p_432899_, CompoundTag p_435591_, Level p_434379_, EntitySpawnReason p_432886_, Function<Entity, Entity> p_434465_
-    ) {
-        Entity entity;
-        try (ProblemReporter.ScopedCollector problemreporter$scopedcollector = new ProblemReporter.ScopedCollector(LOGGER)) {
-            entity = loadEntityRecursive(
-                p_432899_, TagValueInput.create(problemreporter$scopedcollector, p_434379_.registryAccess(), p_435591_), p_434379_, p_432886_, p_434465_
-            );
-        }
-
-        return entity;
-    }
-
-    @Nullable
     public static Entity loadEntityRecursive(ValueInput p_421842_, Level p_422387_, EntitySpawnReason p_422720_, Function<Entity, Entity> p_422153_) {
-        return loadStaticEntity(p_421842_, p_422387_, p_422720_)
-            .map(p_422153_)
-            .map(p_432515_ -> loadPassengersRecursive(p_432515_, p_421842_, p_422387_, p_422720_, p_422153_))
-            .orElse(null);
-    }
-
-    @Nullable
-    public static Entity loadEntityRecursive(
-        EntityType<?> p_433647_, ValueInput p_434069_, Level p_432763_, EntitySpawnReason p_433534_, Function<Entity, Entity> p_433418_
-    ) {
-        return loadStaticEntity(p_433647_, p_434069_, p_432763_, p_433534_)
-            .map(p_433418_)
-            .map(p_432520_ -> loadPassengersRecursive(p_432520_, p_434069_, p_432763_, p_433534_, p_433418_))
-            .orElse(null);
-    }
-
-    private static Entity loadPassengersRecursive(
-        Entity p_433340_, ValueInput p_433716_, Level p_435035_, EntitySpawnReason p_435495_, Function<Entity, Entity> p_435968_
-    ) {
-        for (ValueInput valueinput : p_433716_.childrenListOrEmpty("Passengers")) {
-            Entity entity = loadEntityRecursive(valueinput, p_435035_, p_435495_, p_435968_);
-            if (entity != null) {
-                entity.startRiding(p_433340_, true, false);
+        return loadStaticEntity(p_421842_, p_422387_, p_422720_).map(p_422153_).map(p_421358_ -> {
+            for (ValueInput valueinput : p_421842_.childrenListOrEmpty("Passengers")) {
+                Entity entity = loadEntityRecursive(valueinput, p_422387_, p_422720_, p_422153_);
+                if (entity != null) {
+                    entity.startRiding(p_421358_, true);
+                }
             }
-        }
 
-        return p_433340_;
+            return (Entity)p_421358_;
+        }).orElse(null);
     }
 
     public static Stream<Entity> loadEntitiesRecursive(ValueInput.ValueInputList p_422664_, Level p_147047_, EntitySpawnReason p_360400_) {
@@ -1532,24 +_,24 @@
         }
     }
 
-    private static Optional<Entity> loadStaticEntity(EntityType<?> p_434366_, ValueInput p_435029_, Level p_434374_, EntitySpawnReason p_435895_) {
-        try {
-            return create(p_434366_, p_435029_, p_434374_, p_435895_);
-        } catch (RuntimeException runtimeexception) {
-            LOGGER.warn("Exception loading entity: ", (Throwable)runtimeexception);
-            return Optional.empty();
-        }
-    }
-
     public int clientTrackingRange() {
+        return trackingRangeSupplier.applyAsInt(this);
+    }
+    private int defaultTrackingRangeSupplier() {
         return this.clientTrackingRange;
     }
 
     public int updateInterval() {
+        return updateIntervalSupplier.applyAsInt(this);
+    }
+    private int defaultUpdateIntervalSupplier() {
         return this.updateInterval;
     }
 
     public boolean trackDeltas() {
+        return this.trackDeltasSupplier.test(this);
+    }
+    private boolean defaultTrackDeltasSupplier() {
         return this != PLAYER
             && this != LLAMA_SPIT
             && this != WITHER
@@ -1585,10 +_,6 @@
         return this.builtInRegistryHolder;
     }
 
-    public boolean isAllowedInPeaceful() {
-        return this.allowedInPeaceful;
-    }
-
     private static EntityType.EntityFactory<Boat> boatFactory(Supplier<Item> p_376580_) {
         return (p_375558_, p_375559_) -> new Boat(p_375558_, p_375559_, p_376580_);
     }
@@ -1606,9 +_,12 @@
     }
 
     public boolean onlyOpCanSetNbt() {
+        if (onlyOpCanSetNbt) return true;
         return OP_ONLY_CUSTOM_DATA.contains(this);
     }
 
+    public Stream<TagKey<EntityType<?>>> getTags() {return this.builtInRegistryHolder().tags();}
+
     public static class Builder<T extends Entity> {
         private final EntityType.EntityFactory<T> factory;
         private final MobCategory category;
@@ -1627,7 +_,10 @@
             ResourceKey.create(Registries.LOOT_TABLE, p_367917_.location().withPrefix("entities/"))
         );
         private final DependantName<EntityType<?>, String> descriptionId = p_367918_ -> Util.makeDescriptionId("entity", p_367918_.location());
-        private boolean allowedInPeaceful = true;
+        private java.util.function.Predicate<EntityType<?>> velocityUpdateSupplier = EntityType::defaultTrackDeltasSupplier;
+        private java.util.function.ToIntFunction<EntityType<?>> trackingRangeSupplier = EntityType::defaultTrackingRangeSupplier;
+        private java.util.function.ToIntFunction<EntityType<?>> updateIntervalSupplier = EntityType::defaultUpdateIntervalSupplier;
+        private boolean onlyOpCanSetNbt = false;
 
         private Builder(EntityType.EntityFactory<T> p_20696_, MobCategory p_20697_) {
             this.factory = p_20696_;
@@ -1741,8 +_,23 @@
             return this;
         }
 
-        public EntityType.Builder<T> notInPeaceful() {
-            this.allowedInPeaceful = false;
+        public EntityType.Builder<T> setUpdateInterval(int interval) {
+            this.updateIntervalSupplier = t->interval;
+            return this;
+        }
+
+        public EntityType.Builder<T> setTrackingRange(int range) {
+            this.trackingRangeSupplier = t->range;
+            return this;
+        }
+
+        public EntityType.Builder<T> setShouldReceiveVelocityUpdates(boolean value) {
+            this.velocityUpdateSupplier = t->value;
+            return this;
+        }
+
+        public EntityType.Builder<T> setOnlyOpCanSetNbt(boolean onlyOpCanSetNbt) {
+            this.onlyOpCanSetNbt = onlyOpCanSetNbt;
             return this;
         }
 
@@ -1766,7 +_,10 @@
                 this.descriptionId.get(p_368626_),
                 this.lootTable.get(p_368626_),
                 this.requiredFeatures,
-                this.allowedInPeaceful
+                this.velocityUpdateSupplier,
+                this.trackingRangeSupplier,
+                this.updateIntervalSupplier,
+                this.onlyOpCanSetNbt
             );
         }
     }
