--- a/com/mojang/blaze3d/font/GlyphInfo.java
+++ b/com/mojang/blaze3d/font/GlyphInfo.java
@@ -1,5 +_,8 @@
 package com.mojang.blaze3d.font;
 
+import java.util.function.Function;
+import net.minecraft.client.gui.font.glyphs.BakedGlyph;
+import net.minecraft.client.gui.font.glyphs.EmptyGlyph;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
@@ -19,7 +_,13 @@
         return 1.0F;
     }
 
-    static GlyphInfo simple(float p_439899_) {
-        return () -> p_439899_;
+    BakedGlyph bake(Function<SheetGlyphInfo, BakedGlyph> p_231088_);
+
+    @OnlyIn(Dist.CLIENT)
+    public interface SpaceGlyphInfo extends GlyphInfo {
+        @Override
+        default BakedGlyph bake(Function<SheetGlyphInfo, BakedGlyph> p_231090_) {
+            return EmptyGlyph.INSTANCE;
+        }
     }
 }
