--- a/net/minecraft/client/renderer/item/properties/numeric/CompassAngleState.java
+++ b/net/minecraft/client/renderer/item/properties/numeric/CompassAngleState.java
@@ -12,7 +_,7 @@
 import net.minecraft.util.Mth;
 import net.minecraft.util.RandomSource;
 import net.minecraft.util.StringRepresentable;
-import net.minecraft.world.entity.ItemOwner;
+import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.component.LodestoneTracker;
@@ -42,12 +_,12 @@
     }
 
     @Override
-    protected float calculate(ItemStack p_388108_, ClientLevel p_387750_, int p_388073_, @Nullable ItemOwner p_435973_) {
-        GlobalPos globalpos = this.compassTarget.get(p_387750_, p_388108_, p_435973_);
+    protected float calculate(ItemStack p_388108_, ClientLevel p_387750_, int p_388073_, Entity p_388489_) {
+        GlobalPos globalpos = this.compassTarget.get(p_387750_, p_388108_, p_388489_);
         long i = p_387750_.getGameTime();
-        return !isValidCompassTargetPos(p_435973_, globalpos)
+        return !isValidCompassTargetPos(p_388489_, globalpos)
             ? this.getRandomlySpinningRotation(p_388073_, i)
-            : this.getRotationTowardsCompassTarget(p_435973_, i, globalpos.pos());
+            : this.getRotationTowardsCompassTarget(p_388489_, i, globalpos.pos());
     }
 
     private float getRandomlySpinningRotation(int p_388932_, long p_387198_) {
@@ -59,11 +_,11 @@
         return Mth.positiveModulo(f, 1.0F);
     }
 
-    private float getRotationTowardsCompassTarget(ItemOwner p_432821_, long p_387654_, BlockPos p_388263_) {
-        float f = (float)getAngleFromEntityToPos(p_432821_, p_388263_);
-        float f1 = getWrappedVisualRotationY(p_432821_);
+    private float getRotationTowardsCompassTarget(Entity p_387599_, long p_387654_, BlockPos p_388263_) {
+        float f = (float)getAngleFromEntityToPos(p_387599_, p_388263_);
+        float f1 = getWrappedVisualRotationY(p_387599_);
         float f2;
-        if (p_432821_.asLivingEntity() instanceof Player player && player.isLocalPlayer() && player.level().tickRateManager().runsNormally()) {
+        if (p_387599_ instanceof Player player && player.isLocalPlayer() && player.level().tickRateManager().runsNormally()) {
             if (this.wobbler.shouldUpdate(p_387654_)) {
                 this.wobbler.update(p_387654_, 0.5F - (f1 - 0.25F));
             }
@@ -76,21 +_,19 @@
         return Mth.positiveModulo(f2, 1.0F);
     }
 
-    private static boolean isValidCompassTargetPos(@Nullable ItemOwner p_434341_, @Nullable GlobalPos p_387891_) {
+    private static boolean isValidCompassTargetPos(Entity p_386563_, @Nullable GlobalPos p_387891_) {
         return p_387891_ != null
-            && p_434341_ != null
-            && p_387891_.dimension() == p_434341_.level().dimension()
-            && !(p_387891_.pos().distToCenterSqr(p_434341_.position()) < 1.0E-5F);
+            && p_387891_.dimension() == p_386563_.level().dimension()
+            && !(p_387891_.pos().distToCenterSqr(p_386563_.position()) < 1.0E-5F);
     }
 
-    private static double getAngleFromEntityToPos(ItemOwner p_432840_, BlockPos p_387426_) {
+    private static double getAngleFromEntityToPos(Entity p_388327_, BlockPos p_387426_) {
         Vec3 vec3 = Vec3.atCenterOf(p_387426_);
-        Vec3 vec31 = p_432840_.position();
-        return Math.atan2(vec3.z() - vec31.z(), vec3.x() - vec31.x()) / (float) (Math.PI * 2);
+        return Math.atan2(vec3.z() - p_388327_.getZ(), vec3.x() - p_388327_.getX()) / (float) (Math.PI * 2);
     }
 
-    private static float getWrappedVisualRotationY(ItemOwner p_432979_) {
-        return Mth.positiveModulo(p_432979_.getVisualRotationYInDegrees() / 360.0F, 1.0F);
+    private static float getWrappedVisualRotationY(Entity p_386969_) {
+        return Mth.positiveModulo(p_386969_.getVisualRotationYInDegrees() / 360.0F, 1.0F);
     }
 
     private static int hash(int p_387430_) {
@@ -106,30 +_,29 @@
         NONE("none") {
             @Nullable
             @Override
-            public GlobalPos get(ClientLevel p_388090_, ItemStack p_388593_, @Nullable ItemOwner p_434343_) {
+            public GlobalPos get(ClientLevel p_388090_, ItemStack p_388593_, Entity p_388853_) {
                 return null;
             }
         },
         LODESTONE("lodestone") {
             @Nullable
             @Override
-            public GlobalPos get(ClientLevel p_388590_, ItemStack p_387822_, @Nullable ItemOwner p_436068_) {
+            public GlobalPos get(ClientLevel p_388590_, ItemStack p_387822_, Entity p_386963_) {
                 LodestoneTracker lodestonetracker = p_387822_.get(DataComponents.LODESTONE_TRACKER);
                 return lodestonetracker != null ? lodestonetracker.target().orElse(null) : null;
             }
         },
         SPAWN("spawn") {
-            @Nullable
             @Override
-            public GlobalPos get(ClientLevel p_386460_, ItemStack p_387401_, @Nullable ItemOwner p_434467_) {
+            public GlobalPos get(ClientLevel p_386460_, ItemStack p_387401_, Entity p_387630_) {
                 return GlobalPos.of(p_386460_.dimension(), p_386460_.getSharedSpawnPos());
             }
         },
         RECOVERY("recovery") {
             @Nullable
             @Override
-            public GlobalPos get(ClientLevel p_390456_, ItemStack p_390419_, @Nullable ItemOwner p_435538_) {
-                return (p_435538_ == null ? null : p_435538_.asLivingEntity()) instanceof Player player ? player.getLastDeathLocation().orElse(null) : null;
+            public GlobalPos get(ClientLevel p_390456_, ItemStack p_390419_, Entity p_390489_) {
+                return p_390489_ instanceof Player player ? player.getLastDeathLocation().orElse(null) : null;
             }
         };
 
@@ -146,6 +_,6 @@
         }
 
         @Nullable
-        abstract GlobalPos get(ClientLevel p_387000_, ItemStack p_387505_, @Nullable ItemOwner p_434685_);
+        abstract GlobalPos get(ClientLevel p_387000_, ItemStack p_387505_, Entity p_387524_);
     }
 }
