--- a/net/minecraft/client/renderer/item/properties/numeric/CompassAngle.java
+++ b/net/minecraft/client/renderer/item/properties/numeric/CompassAngle.java
@@ -3,7 +_,7 @@
 import com.mojang.serialization.MapCodec;
 import javax.annotation.Nullable;
 import net.minecraft.client.multiplayer.ClientLevel;
-import net.minecraft.world.entity.ItemOwner;
+import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.item.ItemStack;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
@@ -22,8 +_,8 @@
     }
 
     @Override
-    public float get(ItemStack p_387228_, @Nullable ClientLevel p_386952_, @Nullable ItemOwner p_434216_, int p_387210_) {
-        return this.state.get(p_387228_, p_386952_, p_434216_, p_387210_);
+    public float get(ItemStack p_387228_, @Nullable ClientLevel p_386952_, @Nullable LivingEntity p_386971_, int p_387210_) {
+        return this.state.get(p_387228_, p_386952_, p_386971_, p_387210_);
     }
 
     @Override
