--- a/net/minecraft/client/renderer/item/properties/numeric/CustomModelDataProperty.java
+++ b/net/minecraft/client/renderer/item/properties/numeric/CustomModelDataProperty.java
@@ -7,7 +_,7 @@
 import net.minecraft.client.multiplayer.ClientLevel;
 import net.minecraft.core.component.DataComponents;
 import net.minecraft.util.ExtraCodecs;
-import net.minecraft.world.entity.ItemOwner;
+import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.item.ItemStack;
 import net.minecraft.world.item.component.CustomModelData;
 import net.neoforged.api.distmarker.Dist;
@@ -21,7 +_,7 @@
     );
 
     @Override
-    public float get(ItemStack p_386883_, @Nullable ClientLevel p_387867_, @Nullable ItemOwner p_434671_, int p_388040_) {
+    public float get(ItemStack p_386883_, @Nullable ClientLevel p_387867_, @Nullable LivingEntity p_388117_, int p_388040_) {
         CustomModelData custommodeldata = p_386883_.get(DataComponents.CUSTOM_MODEL_DATA);
         if (custommodeldata != null) {
             Float f = custommodeldata.getFloat(this.index);
