--- a/net/minecraft/client/renderer/item/properties/numeric/NeedleDirectionHelper.java
+++ b/net/minecraft/client/renderer/item/properties/numeric/NeedleDirectionHelper.java
@@ -3,7 +_,8 @@
 import javax.annotation.Nullable;
 import net.minecraft.client.multiplayer.ClientLevel;
 import net.minecraft.util.Mth;
-import net.minecraft.world.entity.ItemOwner;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.item.ItemStack;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
@@ -16,19 +_,20 @@
         this.wobble = p_388777_;
     }
 
-    public float get(ItemStack p_386635_, @Nullable ClientLevel p_387256_, @Nullable ItemOwner p_436053_, int p_387873_) {
-        if (p_436053_ == null) {
-            p_436053_ = p_386635_.getEntityRepresentation();
-        }
-
-        if (p_387256_ == null && p_436053_ != null && p_436053_.level() instanceof ClientLevel clientlevel) {
-            p_387256_ = clientlevel;
-        }
-
-        return p_387256_ == null ? 0.0F : this.calculate(p_386635_, p_387256_, p_387873_, p_436053_);
+    public float get(ItemStack p_386635_, @Nullable ClientLevel p_387256_, @Nullable LivingEntity p_388745_, int p_387873_) {
+        Entity entity = (Entity)(p_388745_ != null ? p_388745_ : p_386635_.getEntityRepresentation());
+        if (entity == null) {
+            return 0.0F;
+        } else {
+            if (p_387256_ == null && entity.level() instanceof ClientLevel clientlevel) {
+                p_387256_ = clientlevel;
+            }
+
+            return p_387256_ == null ? 0.0F : this.calculate(p_386635_, p_387256_, p_387873_, entity);
+        }
     }
 
-    protected abstract float calculate(ItemStack p_386601_, ClientLevel p_387798_, int p_387568_, @Nullable ItemOwner p_434879_);
+    protected abstract float calculate(ItemStack p_386601_, ClientLevel p_387798_, int p_387568_, Entity p_388687_);
 
     protected boolean wobble() {
         return this.wobble;
