--- a/net/minecraft/network/protocol/configuration/ClientConfigurationPacketListener.java
+++ b/net/minecraft/network/protocol/configuration/ClientConfigurationPacketListener.java
@@ -9,8 +_,6 @@
         return ConnectionProtocol.CONFIGURATION;
     }
 
-    void handleCodeOfConduct(ClientboundCodeOfConductPacket p_439391_);
-
     void handleConfigurationFinished(ClientboundFinishConfigurationPacket p_295371_);
 
     void handleRegistryData(ClientboundRegistryDataPacket p_295674_);
