--- a/net/minecraft/network/protocol/configuration/ServerConfigurationPacketListener.java
+++ b/net/minecraft/network/protocol/configuration/ServerConfigurationPacketListener.java
@@ -12,6 +_,4 @@
     void handleConfigurationFinished(ServerboundFinishConfigurationPacket p_295218_);
 
     void handleSelectKnownPacks(ServerboundSelectKnownPacks p_326510_);
-
-    void handleAcceptCodeOfConduct(ServerboundAcceptCodeOfConductPacket p_440002_);
 }
