--- a/net/minecraft/world/entity/projectile/SpectralArrow.java
+++ b/net/minecraft/world/entity/projectile/SpectralArrow.java
@@ -31,7 +_,7 @@
     @Override
     public void tick() {
         super.tick();
-        if (this.level().isClientSide() && !this.isInGround()) {
+        if (this.level().isClientSide && !this.isInGround()) {
             this.level().addParticle(ParticleTypes.INSTANT_EFFECT, this.getX(), this.getY(), this.getZ(), 0.0, 0.0, 0.0);
         }
     }
