--- a/net/minecraft/world/entity/projectile/ThrownEgg.java
+++ b/net/minecraft/world/entity/projectile/ThrownEgg.java
@@ -61,7 +_,7 @@
     @Override
     protected void onHit(HitResult p_37488_) {
         super.onHit(p_37488_);
-        if (!this.level().isClientSide()) {
+        if (!this.level().isClientSide) {
             if (this.random.nextInt(8) == 0) {
                 int i = 1;
                 if (this.random.nextInt(32) == 0) {
