--- a/net/minecraft/network/codec/ByteBufCodecs.java
+++ b/net/minecraft/network/codec/ByteBufCodecs.java
@@ -19,12 +_,12 @@
 import java.util.Map;
 import java.util.Optional;
 import java.util.OptionalInt;
+import java.util.UUID;
 import java.util.function.BiFunction;
 import java.util.function.Function;
 import java.util.function.IntFunction;
 import java.util.function.Supplier;
 import java.util.function.ToIntFunction;
-import net.minecraft.Util;
 import net.minecraft.core.Holder;
 import net.minecraft.core.HolderSet;
 import net.minecraft.core.IdMap;
@@ -236,25 +_,30 @@
             }
         }
     };
-    StreamCodec<ByteBuf, String> PLAYER_NAME = stringUtf8(16);
-    StreamCodec<ByteBuf, GameProfile> GAME_PROFILE = StreamCodec.composite(
-        UUIDUtil.STREAM_CODEC,
-        GameProfile::getId,
-        PLAYER_NAME,
-        GameProfile::getName,
-        GAME_PROFILE_PROPERTIES,
-        GameProfile::getProperties,
-        Util::createGameProfile
-    );
+    StreamCodec<ByteBuf, GameProfile> GAME_PROFILE = new StreamCodec<ByteBuf, GameProfile>() {
+        public GameProfile decode(ByteBuf p_422404_) {
+            UUID uuid = UUIDUtil.STREAM_CODEC.decode(p_422404_);
+            String s = Utf8String.read(p_422404_, 16);
+            GameProfile gameprofile = new GameProfile(uuid, s);
+            gameprofile.getProperties().putAll(ByteBufCodecs.GAME_PROFILE_PROPERTIES.decode(p_422404_));
+            return gameprofile;
+        }
+
+        public void encode(ByteBuf p_422062_, GameProfile p_428503_) {
+            UUIDUtil.STREAM_CODEC.encode(p_422062_, p_428503_.getId());
+            Utf8String.write(p_422062_, p_428503_.getName(), 16);
+            ByteBufCodecs.GAME_PROFILE_PROPERTIES.encode(p_422062_, p_428503_.getProperties());
+        }
+    };
     StreamCodec<ByteBuf, Integer> RGB_COLOR = new StreamCodec<ByteBuf, Integer>() {
-        public Integer decode(ByteBuf p_422404_) {
-            return ARGB.color(p_422404_.readByte() & 0xFF, p_422404_.readByte() & 0xFF, p_422404_.readByte() & 0xFF);
+        public Integer decode(ByteBuf p_422595_) {
+            return ARGB.color(p_422595_.readByte() & 0xFF, p_422595_.readByte() & 0xFF, p_422595_.readByte() & 0xFF);
         }
 
-        public void encode(ByteBuf p_422062_, Integer p_440662_) {
-            p_422062_.writeByte(ARGB.red(p_440662_));
-            p_422062_.writeByte(ARGB.green(p_440662_));
-            p_422062_.writeByte(ARGB.blue(p_440662_));
+        public void encode(ByteBuf p_422247_, Integer p_428314_) {
+            p_422247_.writeByte(ARGB.red(p_428314_));
+            p_422247_.writeByte(ARGB.green(p_428314_));
+            p_422247_.writeByte(ARGB.blue(p_428314_));
         }
     };
 
@@ -572,7 +_,7 @@
     ) {
         return new StreamCodec<RegistryFriendlyByteBuf, R>() {
             private IdMap<R> getRegistryOrThrow(RegistryFriendlyByteBuf p_421864_) {
-                return p_320353_.apply(p_421864_.registryAccess().lookupOrThrow(p_319942_));
+                return p_320353_.apply(getSyncableRegistryOrThrow(p_421864_, p_319942_));
             }
 
             public R decode(RegistryFriendlyByteBuf p_412249_) {
@@ -602,7 +_,7 @@
             private static final int DIRECT_HOLDER_ID = 0;
 
             private IdMap<Holder<T>> getRegistryOrThrow(RegistryFriendlyByteBuf p_428365_) {
-                return p_428365_.registryAccess().lookupOrThrow(p_320391_).asHolderIdMap();
+                return getSyncableRegistryOrThrow(p_428365_, p_320391_).asHolderIdMap();
             }
 
             public Holder<T> decode(RegistryFriendlyByteBuf p_421902_) {
@@ -624,13 +_,38 @@
         };
     }
 
+    private static <T> Registry<T> getSyncableRegistryOrThrow(RegistryFriendlyByteBuf buffer, ResourceKey<? extends Registry<T>> registryKey) {
+        var registry = buffer.registryAccess().lookupOrThrow(registryKey);
+        if (net.neoforged.neoforge.registries.RegistryManager.isNonSyncedBuiltInRegistry(registry)) {
+            throw new io.netty.handler.codec.CodecException("Cannot use ID syncing for non-synced built-in registry: " + registry.key());
+        }
+        return registry;
+    }
+
     static <T> StreamCodec<RegistryFriendlyByteBuf, HolderSet<T>> holderSet(final ResourceKey<? extends Registry<T>> p_332137_) {
         return new StreamCodec<RegistryFriendlyByteBuf, HolderSet<T>>() {
             private static final int NAMED_SET = -1;
             private final StreamCodec<RegistryFriendlyByteBuf, Holder<T>> holderCodec = ByteBufCodecs.holderRegistry(p_332137_);
 
+            private final Map<net.neoforged.neoforge.registries.holdersets.HolderSetType, StreamCodec<RegistryFriendlyByteBuf, ? extends net.neoforged.neoforge.registries.holdersets.ICustomHolderSet<T>>> holderSetCodecs = new java.util.concurrent.ConcurrentHashMap<>();
+
+            private StreamCodec<RegistryFriendlyByteBuf, ? extends net.neoforged.neoforge.registries.holdersets.ICustomHolderSet<T>> holderSetCodec(net.neoforged.neoforge.registries.holdersets.HolderSetType type) {
+                return this.holderSetCodecs.computeIfAbsent(type, key -> key.makeStreamCodec(p_332137_));
+            }
+
+            private <H extends net.neoforged.neoforge.registries.holdersets.ICustomHolderSet<T>> H cast(net.neoforged.neoforge.registries.holdersets.ICustomHolderSet<T> holderSet) {
+                return (H) holderSet;
+            }
+
             public HolderSet<T> decode(RegistryFriendlyByteBuf p_428283_) {
                 int i = VarInt.read(p_428283_) - 1;
+                // Neo: Co-opt negative VarInt values within the HolderSet codec as an HolderSetType id.
+                // Vanilla uses 0 for tag and [1, Integer.MAX_VALUE] for list size [0, Integer.MAX_VALUE - 1].
+                // So we may encode the registry id for custom holder set types in [Integer.MIN_VALUE + 1, -1] (local variable i must not be underflow).
+                // The registry id for custom holder set types is (-1 - network id), while local variable i is (network id - 1), so the registry id would be (-2 - i).
+                if (i < -1) {
+                    return this.holderSetCodec(net.neoforged.neoforge.registries.NeoForgeRegistries.HOLDER_SET_TYPES.byIdOrThrow(-2 - i)).decode(p_428283_);
+                }
                 if (i == -1) {
                     Registry<T> registry = p_428283_.registryAccess().lookupOrThrow(p_332137_);
                     return registry.get(TagKey.create(p_332137_, ResourceLocation.STREAM_CODEC.decode(p_428283_))).orElseThrow();
@@ -646,6 +_,15 @@
             }
 
             public void encode(RegistryFriendlyByteBuf p_428187_, HolderSet<T> p_428305_) {
+                // Neo: Co-opt negative VarInt values within the HolderSet codec as an HolderSetType id.
+                // Vanilla uses 0 for tag and [1, Integer.MAX_VALUE] for list size [0, Integer.MAX_VALUE - 1] (local variable i in decode() must not be underflow).
+                // So we may encode the registry id for custom holder set types in [Integer.MIN_VALUE + 1, -1].
+                // The network id for custom holder set types is (-1 - registry id)
+                if (p_428187_.getConnectionType().isNeoForge() && p_428305_ instanceof net.neoforged.neoforge.registries.holdersets.ICustomHolderSet<T> customHolderSet) {
+                    VarInt.write(p_428187_, -1 - net.neoforged.neoforge.registries.NeoForgeRegistries.HOLDER_SET_TYPES.getId(customHolderSet.type()));
+                    this.holderSetCodec(customHolderSet.type()).encode(p_428187_, cast(customHolderSet));
+                    return;
+                }
                 Optional<TagKey<T>> optional = p_428305_.unwrapKey();
                 if (optional.isPresent()) {
                     VarInt.write(p_428187_, 0);
@@ -665,8 +_,8 @@
         return new StreamCodec<ByteBuf, JsonElement>() {
             private static final Gson GSON = new GsonBuilder().disableHtmlEscaping().create();
 
-            public JsonElement decode(ByteBuf p_422595_) {
-                String s = Utf8String.read(p_422595_, p_422673_);
+            public JsonElement decode(ByteBuf p_428344_) {
+                String s = Utf8String.read(p_428344_, p_422673_);
 
                 try {
                     return LenientJsonParser.parse(s);
@@ -675,9 +_,9 @@
                 }
             }
 
-            public void encode(ByteBuf p_422247_, JsonElement p_439988_) {
-                String s = GSON.toJson(p_439988_);
-                Utf8String.write(p_422247_, s, p_422673_);
+            public void encode(ByteBuf p_428191_, JsonElement p_428484_) {
+                String s = GSON.toJson(p_428484_);
+                Utf8String.write(p_428191_, s, p_422673_);
             }
         };
     }
