--- a/net/minecraft/world/inventory/CrafterMenu.java
+++ b/net/minecraft/world/inventory/CrafterMenu.java
@@ -109,7 +_,7 @@
             ServerLevel serverlevel = serverplayer.level();
             CraftingInput craftinginput = this.container.asCraftInput();
             ItemStack itemstack = CrafterBlock.getPotentialResults(serverlevel, craftinginput)
-                .map(p_432562_ -> p_432562_.value().assemble(craftinginput, serverlevel.registryAccess()))
+                .map(p_425439_ -> p_425439_.value().assemble(craftinginput, serverlevel.registryAccess()))
                 .orElse(ItemStack.EMPTY);
             this.resultContainer.setItem(0, itemstack);
         }
