--- a/net/minecraft/world/inventory/InventoryMenu.java
+++ b/net/minecraft/world/inventory/InventoryMenu.java
@@ -88,7 +_,7 @@
     public void removed(Player p_39721_) {
         super.removed(p_39721_);
         this.resultSlots.clearContent();
-        if (!p_39721_.level().isClientSide()) {
+        if (!p_39721_.level().isClientSide) {
             this.clearContainer(p_39721_, this.craftSlots);
         }
     }
