--- a/net/minecraft/client/gui/font/FontManager.java
+++ b/net/minecraft/client/gui/font/FontManager.java
@@ -35,13 +_,8 @@
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.Options;
 import net.minecraft.client.gui.Font;
-import net.minecraft.client.gui.GlyphSource;
-import net.minecraft.client.gui.font.glyphs.EffectGlyph;
 import net.minecraft.client.gui.font.providers.GlyphProviderDefinition;
-import net.minecraft.client.renderer.texture.TextureAtlas;
 import net.minecraft.client.renderer.texture.TextureManager;
-import net.minecraft.client.resources.model.AtlasManager;
-import net.minecraft.network.chat.FontDescription;
 import net.minecraft.resources.FileToIdConverter;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.packs.resources.PreparableReloadListener;
@@ -61,26 +_,16 @@
     public static final ResourceLocation MISSING_FONT = ResourceLocation.withDefaultNamespace("missing");
     private static final FileToIdConverter FONT_DEFINITIONS = FileToIdConverter.json("font");
     private static final Gson GSON = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
-    final FontSet missingFontSet;
+    private final FontSet missingFontSet;
     private final List<GlyphProvider> providersToClose = new ArrayList<>();
     private final Map<ResourceLocation, FontSet> fontSets = new HashMap<>();
     private final TextureManager textureManager;
-    private final FontManager.CachedFontProvider anyGlyphs = new FontManager.CachedFontProvider(false);
-    private final FontManager.CachedFontProvider nonFishyGlyphs = new FontManager.CachedFontProvider(true);
-    private final AtlasManager atlasManager;
-    private final Map<ResourceLocation, AtlasGlyphProvider> atlasProviders = new HashMap<>();
+    @Nullable
+    private volatile FontSet lastFontSetCache;
 
-    public FontManager(TextureManager p_95005_, AtlasManager p_436601_) {
+    public FontManager(TextureManager p_95005_) {
         this.textureManager = p_95005_;
-        this.atlasManager = p_436601_;
-        this.missingFontSet = this.createFontSet(MISSING_FONT, List.of(createFallbackProvider()), Set.of());
-    }
-
-    private FontSet createFontSet(ResourceLocation p_435475_, List<GlyphProvider.Conditional> p_433289_, Set<FontOption> p_434588_) {
-        GlyphStitcher glyphstitcher = new GlyphStitcher(this.textureManager, p_435475_);
-        FontSet fontset = new FontSet(glyphstitcher);
-        fontset.reload(p_433289_, p_434588_);
-        return fontset;
+        this.missingFontSet = Util.make(new FontSet(p_95005_, MISSING_FONT), p_325488_ -> p_325488_.reload(List.of(createFallbackProvider()), Set.of()));
     }
 
     private static GlyphProvider.Conditional createFallbackProvider() {
@@ -89,11 +_,9 @@
 
     @Override
     public CompletableFuture<Void> reload(
-        PreparableReloadListener.SharedState p_435059_, Executor p_284975_, PreparableReloadListener.PreparationBarrier p_285160_, Executor p_285218_
+        PreparableReloadListener.PreparationBarrier p_285160_, ResourceManager p_285231_, Executor p_284975_, Executor p_285218_
     ) {
-        return this.prepare(p_435059_.resourceManager(), p_284975_)
-            .thenCompose(p_285160_::wait)
-            .thenAcceptAsync(p_372503_ -> this.apply(p_372503_, Profiler.get()), p_285218_);
+        return this.prepare(p_285231_, p_284975_).thenCompose(p_285160_::wait).thenAcceptAsync(p_372503_ -> this.apply(p_372503_, Profiler.get()), p_285218_);
     }
 
     private CompletableFuture<FontManager.Preparation> prepare(ResourceManager p_285252_, Executor p_284969_) {
@@ -179,10 +_,10 @@
             intset.addAll(glyphprovider$conditional.provider().getSupportedGlyphs());
         }
 
-        intset.forEach(p_438726_ -> {
-            if (p_438726_ != 32) {
+        intset.forEach(p_325466_ -> {
+            if (p_325466_ != 32) {
                 for (GlyphProvider.Conditional glyphprovider$conditional1 : Lists.reverse(p_285520_)) {
-                    if (glyphprovider$conditional1.provider().getGlyph(p_438726_) != null) {
+                    if (glyphprovider$conditional1.provider().getGlyph(p_325466_) != null) {
                         break;
                     }
                 }
@@ -205,26 +_,22 @@
 
     private void apply(FontManager.Preparation p_284939_, ProfilerFiller p_285407_) {
         p_285407_.push("closing");
-        this.anyGlyphs.invalidate();
-        this.nonFishyGlyphs.invalidate();
+        this.lastFontSetCache = null;
         this.fontSets.values().forEach(FontSet::close);
         this.fontSets.clear();
         this.providersToClose.forEach(GlyphProvider::close);
         this.providersToClose.clear();
         Set<FontOption> set = getFontOptions(Minecraft.getInstance().options);
         p_285407_.popPush("reloading");
-        p_284939_.fontSets()
-            .forEach(
-                (p_432219_, p_432220_) -> this.fontSets
-                    .put(p_432219_, this.createFontSet(p_432219_, Lists.reverse((List<GlyphProvider.Conditional>)p_432220_), set))
-            );
+        p_284939_.fontSets().forEach((p_325478_, p_325479_) -> {
+            FontSet fontset = new FontSet(this.textureManager, p_325478_);
+            fontset.reload(Lists.reverse((List<GlyphProvider.Conditional>)p_325479_), set);
+            this.fontSets.put(p_325478_, fontset);
+        });
         this.providersToClose.addAll(p_284939_.allProviders);
         p_285407_.pop();
         if (!this.fontSets.containsKey(Minecraft.DEFAULT_FONT)) {
             throw new IllegalStateException("Default font failed to load");
-        } else {
-            this.atlasProviders.clear();
-            this.atlasManager.forEach((p_436485_, p_436486_) -> this.atlasProviders.put(p_436485_, new AtlasGlyphProvider(p_436486_)));
         }
     }
 
@@ -262,26 +_,30 @@
     }
 
     public Font createFont() {
-        return new Font(this.anyGlyphs);
+        return new Font(this::getFontSetCached, false);
     }
 
     public Font createFontFilterFishy() {
-        return new Font(this.nonFishyGlyphs);
+        return new Font(this::getFontSetCached, true);
     }
 
-    FontSet getFontSetRaw(ResourceLocation p_325954_) {
+    private FontSet getFontSetRaw(ResourceLocation p_325954_) {
         return this.fontSets.getOrDefault(p_325954_, this.missingFontSet);
     }
 
-    GlyphSource getSpriteFont(FontDescription.AtlasSprite p_436579_) {
-        AtlasGlyphProvider atlasglyphprovider = this.atlasProviders.get(p_436579_.atlasId());
-        return atlasglyphprovider == null ? this.missingFontSet.source(false) : atlasglyphprovider.sourceForSprite(p_436579_.spriteId());
+    private FontSet getFontSetCached(ResourceLocation p_326503_) {
+        FontSet fontset = this.lastFontSetCache;
+        if (fontset != null && p_326503_.equals(fontset.name())) {
+            return fontset;
+        } else {
+            FontSet fontset1 = this.getFontSetRaw(p_326503_);
+            this.lastFontSetCache = fontset1;
+            return fontset1;
+        }
     }
 
     @Override
     public void close() {
-        this.anyGlyphs.close();
-        this.nonFishyGlyphs.close();
         this.fontSets.values().forEach(FontSet::close);
         this.providersToClose.forEach(GlyphProvider::close);
         this.missingFontSet.close();
@@ -320,65 +_,6 @@
 
         private GlyphProvider.Conditional mergeFilters(GlyphProvider.Conditional p_326502_) {
             return new GlyphProvider.Conditional(p_326502_.provider(), this.filter.merge(p_326502_.filter()));
-        }
-    }
-
-    @OnlyIn(Dist.CLIENT)
-    class CachedFontProvider implements Font.Provider, AutoCloseable {
-        private final boolean nonFishyOnly;
-        @Nullable
-        private volatile FontManager.CachedFontProvider.CachedEntry lastEntry;
-        @Nullable
-        private volatile EffectGlyph whiteGlyph;
-
-        CachedFontProvider(boolean p_435857_) {
-            this.nonFishyOnly = p_435857_;
-        }
-
-        public void invalidate() {
-            this.lastEntry = null;
-            this.whiteGlyph = null;
-        }
-
-        @Override
-        public void close() {
-            this.invalidate();
-        }
-
-        private GlyphSource getGlyphSource(FontDescription p_436666_) {
-            return switch (p_436666_) {
-                case FontDescription.Resource fontdescription$resource -> FontManager.this.getFontSetRaw(fontdescription$resource.id())
-                    .source(this.nonFishyOnly);
-                case FontDescription.AtlasSprite fontdescription$atlassprite -> FontManager.this.getSpriteFont(fontdescription$atlassprite);
-                default -> FontManager.this.missingFontSet.source(this.nonFishyOnly);
-            };
-        }
-
-        @Override
-        public GlyphSource glyphs(FontDescription p_436755_) {
-            FontManager.CachedFontProvider.CachedEntry fontmanager$cachedfontprovider$cachedentry = this.lastEntry;
-            if (fontmanager$cachedfontprovider$cachedentry != null && p_436755_.equals(fontmanager$cachedfontprovider$cachedentry.description)) {
-                return fontmanager$cachedfontprovider$cachedentry.source;
-            } else {
-                GlyphSource glyphsource = this.getGlyphSource(p_436755_);
-                this.lastEntry = new FontManager.CachedFontProvider.CachedEntry(p_436755_, glyphsource);
-                return glyphsource;
-            }
-        }
-
-        @Override
-        public EffectGlyph effect() {
-            EffectGlyph effectglyph = this.whiteGlyph;
-            if (effectglyph == null) {
-                effectglyph = FontManager.this.getFontSetRaw(FontDescription.DEFAULT.id()).whiteGlyph();
-                this.whiteGlyph = effectglyph;
-            }
-
-            return effectglyph;
-        }
-
-        @OnlyIn(Dist.CLIENT)
-        record CachedEntry(FontDescription description, GlyphSource source) {
         }
     }
 
