--- a/net/minecraft/client/renderer/item/RangeSelectItemModel.java
+++ b/net/minecraft/client/renderer/item/RangeSelectItemModel.java
@@ -14,7 +_,7 @@
 import net.minecraft.client.renderer.item.properties.numeric.RangeSelectItemModelProperties;
 import net.minecraft.client.renderer.item.properties.numeric.RangeSelectItemModelProperty;
 import net.minecraft.client.resources.model.ResolvableModel;
-import net.minecraft.world.entity.ItemOwner;
+import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.item.ItemDisplayContext;
 import net.minecraft.world.item.ItemStack;
 import net.neoforged.api.distmarker.Dist;
@@ -64,11 +_,11 @@
         ItemModelResolver p_388786_,
         ItemDisplayContext p_387570_,
         @Nullable ClientLevel p_388512_,
-        @Nullable ItemOwner p_435703_,
+        @Nullable LivingEntity p_388280_,
         int p_388577_
     ) {
         p_387732_.appendModelIdentityElement(this);
-        float f = this.property.get(p_386891_, p_388512_, p_435703_, p_388577_) * this.scale;
+        float f = this.property.get(p_386891_, p_388512_, p_388280_, p_388577_) * this.scale;
         ItemModel itemmodel;
         if (Float.isNaN(f)) {
             itemmodel = this.fallback;
@@ -77,7 +_,7 @@
             itemmodel = i == -1 ? this.fallback : this.models[i];
         }
 
-        itemmodel.update(p_387732_, p_386891_, p_388786_, p_387570_, p_388512_, p_435703_, p_388577_);
+        itemmodel.update(p_387732_, p_386891_, p_388786_, p_387570_, p_388512_, p_388280_, p_388577_);
     }
 
     @OnlyIn(Dist.CLIENT)
