--- a/net/minecraft/world/entity/vehicle/AbstractBoat.java
+++ b/net/minecraft/world/entity/vehicle/AbstractBoat.java
@@ -30,6 +_,8 @@
 import net.minecraft.world.entity.MoverType;
 import net.minecraft.world.entity.Pose;
 import net.minecraft.world.entity.animal.Animal;
+import net.minecraft.world.entity.animal.WaterAnimal;
+import net.minecraft.world.entity.monster.creaking.Creaking;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
@@ -215,7 +_,7 @@
             this.outOfControlTicks++;
         }
 
-        if (!this.level().isClientSide() && this.outOfControlTicks >= 60.0F) {
+        if (!this.level().isClientSide && this.outOfControlTicks >= 60.0F) {
             this.ejectPassengers();
         }
 
@@ -235,7 +_,7 @@
             }
 
             this.floatBoat();
-            if (this.level().isClientSide()) {
+            if (this.level().isClientSide) {
                 this.controlBoat();
                 this.level().sendPacketToServer(new ServerboundPaddleBoatPacket(this.getPaddleState(0), this.getPaddleState(1)));
             }
@@ -281,7 +_,7 @@
 
         List<Entity> list = this.level().getEntities(this, this.getBoundingBox().inflate(0.2F, -0.01F, 0.2F), EntitySelector.pushableBy(this));
         if (!list.isEmpty()) {
-            boolean flag = !this.level().isClientSide() && !(this.getControllingPassenger() instanceof Player);
+            boolean flag = !this.level().isClientSide && !(this.getControllingPassenger() instanceof Player);
 
             for (Entity entity : list) {
                 if (!entity.hasPassenger(this)) {
@@ -290,7 +_,9 @@
                         && !entity.isPassenger()
                         && this.hasEnoughSpaceFor(entity)
                         && entity instanceof LivingEntity
-                        && !entity.getType().is(EntityTypeTags.CANNOT_BE_PUSHED_ONTO_BOATS)) {
+                        && !(entity instanceof WaterAnimal)
+                        && !(entity instanceof Player)
+                        && !(entity instanceof Creaking)) {
                         entity.startRiding(this);
                     } else {
                         this.push(entity);
@@ -301,7 +_,7 @@
     }
 
     private void tickBubbleColumn() {
-        if (this.level().isClientSide()) {
+        if (this.level().isClientSide) {
             int i = this.getBubbleTime();
             if (i > 0) {
                 this.bubbleMultiplier += 0.05F;
@@ -419,7 +_,7 @@
                 for (int i2 = i1; i2 < j1; i2++) {
                     blockpos$mutableblockpos.set(l1, k1, i2);
                     FluidState fluidstate = this.level().getFluidState(blockpos$mutableblockpos);
-                    if (fluidstate.is(FluidTags.WATER)) {
+                    if (this.canBoatInFluid(fluidstate)) {
                         f = Math.max(f, fluidstate.getHeight(this.level(), blockpos$mutableblockpos));
                     }
 
@@ -465,7 +_,7 @@
                                     voxelshape,
                                     BooleanOp.AND
                                 )) {
-                                f += blockstate.getBlock().getFriction();
+                                f += blockstate.getFriction(this.level(), blockpos$mutableblockpos, this);
                                 k1++;
                             }
                         }
@@ -494,7 +_,7 @@
                 for (int i2 = i1; i2 < j1; i2++) {
                     blockpos$mutableblockpos.set(k1, l1, i2);
                     FluidState fluidstate = this.level().getFluidState(blockpos$mutableblockpos);
-                    if (fluidstate.is(FluidTags.WATER)) {
+                    if (this.canBoatInFluid(fluidstate)) {
                         float f = l1 + fluidstate.getHeight(this.level(), blockpos$mutableblockpos);
                         this.waterLevel = Math.max((double)f, this.waterLevel);
                         flag |= aabb.minY < f;
@@ -524,7 +_,7 @@
                 for (int i2 = i1; i2 < j1; i2++) {
                     blockpos$mutableblockpos.set(k1, l1, i2);
                     FluidState fluidstate = this.level().getFluidState(blockpos$mutableblockpos);
-                    if (fluidstate.is(FluidTags.WATER) && d0 < blockpos$mutableblockpos.getY() + fluidstate.getHeight(this.level(), blockpos$mutableblockpos)) {
+                    if (this.canBoatInFluid(fluidstate) && d0 < blockpos$mutableblockpos.getY() + fluidstate.getHeight(this.level(), blockpos$mutableblockpos)) {
                         if (!fluidstate.isSource()) {
                             return AbstractBoat.Status.UNDER_FLOWING_WATER;
                         }
@@ -705,7 +_,7 @@
         } else {
             return (InteractionResult)(p_376241_.isSecondaryUseActive()
                     || !(this.outOfControlTicks < 60.0F)
-                    || !this.level().isClientSide() && !p_376241_.startRiding(this)
+                    || !this.level().isClientSide && !p_376241_.startRiding(this)
                 ? InteractionResult.PASS
                 : InteractionResult.SUCCESS);
         }
@@ -713,7 +_,7 @@
 
     @Override
     public void remove(Entity.RemovalReason p_376469_) {
-        if (!this.level().isClientSide() && p_376469_.shouldDestroy() && this.isLeashed()) {
+        if (!this.level().isClientSide && p_376469_.shouldDestroy() && this.isLeashed()) {
             this.dropLeash();
         }
 
@@ -726,7 +_,7 @@
         if (!this.isPassenger()) {
             if (p_376924_) {
                 this.resetFallDistance();
-            } else if (!this.level().getFluidState(this.blockPosition().below()).is(FluidTags.WATER) && p_376661_ < 0.0) {
+            } else if (!this.canBoatInFluid(this.level().getFluidState(this.blockPosition().below())) && p_376661_ < 0.0) {
                 this.fallDistance -= (float)p_376661_;
             }
         }
@@ -750,7 +_,7 @@
 
     @Override
     protected boolean canAddPassenger(Entity p_376443_) {
-        return this.getPassengers().size() < this.getMaxPassengers() && !this.isEyeInFluid(FluidTags.WATER);
+        return this.getPassengers().size() < this.getMaxPassengers() && !this.canBoatInFluid(this.getEyeInFluidType());
     }
 
     protected int getMaxPassengers() {
