--- a/net/minecraft/world/entity/vehicle/AbstractChestBoat.java
+++ b/net/minecraft/world/entity/vehicle/AbstractChestBoat.java
@@ -9,7 +_,6 @@
 import net.minecraft.world.InteractionHand;
 import net.minecraft.world.InteractionResult;
 import net.minecraft.world.damagesource.DamageSource;
-import net.minecraft.world.entity.ContainerUser;
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.entity.HasCustomInventoryScreen;
@@ -68,7 +_,7 @@
 
     @Override
     public void remove(Entity.RemovalReason p_376764_) {
-        if (!this.level().isClientSide() && p_376764_.shouldDestroy()) {
+        if (!this.level().isClientSide && p_376764_.shouldDestroy()) {
             Containers.dropContents(this.level(), this, this);
         }
 
@@ -193,7 +_,7 @@
     }
 
     @Override
-    public void stopOpen(ContainerUser p_435779_) {
-        this.level().gameEvent(GameEvent.CONTAINER_CLOSE, this.position(), GameEvent.Context.of(p_435779_.getLivingEntity()));
+    public void stopOpen(Player p_376820_) {
+        this.level().gameEvent(GameEvent.CONTAINER_CLOSE, this.position(), GameEvent.Context.of(p_376820_));
     }
 }
