--- a/net/minecraft/world/entity/vehicle/AbstractMinecartContainer.java
+++ b/net/minecraft/world/entity/vehicle/AbstractMinecartContainer.java
@@ -73,7 +_,7 @@
 
     @Override
     public void remove(Entity.RemovalReason p_150255_) {
-        if (!this.level().isClientSide() && p_150255_.shouldDestroy()) {
+        if (!this.level().isClientSide && p_150255_.shouldDestroy()) {
             Containers.dropContents(this.level(), this, this);
         }
 
@@ -94,6 +_,8 @@
 
     @Override
     public InteractionResult interact(Player p_38232_, InteractionHand p_38233_) {
+        InteractionResult ret = super.interact(p_38232_, p_38233_);
+        if (ret.consumesAction()) return ret;
         return this.interactWithContainerVehicle(p_38232_);
     }
 
