--- a/net/minecraft/world/entity/vehicle/MinecartFurnace.java
+++ b/net/minecraft/world/entity/vehicle/MinecartFurnace.java
@@ -107,25 +_,19 @@
 
     @Override
     public InteractionResult interact(Player p_38562_, InteractionHand p_38563_) {
+        InteractionResult ret = super.interact(p_38562_, p_38563_);
+        if (ret.consumesAction()) return ret;
         ItemStack itemstack = p_38562_.getItemInHand(p_38563_);
-        if (this.addFuel(p_38562_.position(), itemstack)) {
+        if (itemstack.is(ItemTags.FURNACE_MINECART_FUEL) && this.fuel + 3600 <= 32000) {
             itemstack.consume(1, p_38562_);
-        }
-
-        return InteractionResult.SUCCESS;
-    }
-
-    public boolean addFuel(Vec3 p_437319_, ItemStack p_437362_) {
-        if (p_437362_.is(ItemTags.FURNACE_MINECART_FUEL) && this.fuel + 3600 <= 32000) {
             this.fuel += 3600;
-            if (this.fuel > 0) {
-                this.push = this.position().subtract(p_437319_).horizontal();
-            }
-
-            return true;
-        } else {
-            return false;
-        }
+        }
+
+        if (this.fuel > 0) {
+            this.push = this.position().subtract(p_38562_.position()).horizontal();
+        }
+
+        return InteractionResult.SUCCESS;
     }
 
     @Override
