--- a/net/minecraft/client/gui/components/debug/DebugScreenDisplayer.java
+++ /dev/null
@@ -1,17 +_,0 @@
-package net.minecraft.client.gui.components.debug;
-
-import java.util.Collection;
-import net.minecraft.resources.ResourceLocation;
-import net.neoforged.api.distmarker.Dist;
-import net.neoforged.api.distmarker.OnlyIn;
-
-@OnlyIn(Dist.CLIENT)
-public interface DebugScreenDisplayer {
-    void addPriorityLine(String p_435623_);
-
-    void addLine(String p_433177_);
-
-    void addToGroup(ResourceLocation p_435682_, Collection<String> p_434681_);
-
-    void addToGroup(ResourceLocation p_434612_, String p_433984_);
-}
