--- a/net/minecraft/client/gui/components/debug/package-info.java
+++ /dev/null
@@ -1,11 +_,0 @@
-@ParametersAreNonnullByDefault
-@MethodsReturnNonnullByDefault
-@FieldsAreNonnullByDefault
-@OnlyIn(Dist.CLIENT)
-package net.minecraft.client.gui.components.debug;
-
-import javax.annotation.ParametersAreNonnullByDefault;
-import net.minecraft.FieldsAreNonnullByDefault;
-import net.minecraft.MethodsReturnNonnullByDefault;
-import net.neoforged.api.distmarker.Dist;
-import net.neoforged.api.distmarker.OnlyIn;
