--- a/net/minecraft/data/recipes/packs/VanillaRecipeProvider.java
+++ b/net/minecraft/data/recipes/packs/VanillaRecipeProvider.java
@@ -113,18 +_,6 @@
         this.woodenBoat(Items.OAK_BOAT, Blocks.OAK_PLANKS);
         this.woodenBoat(Items.SPRUCE_BOAT, Blocks.SPRUCE_PLANKS);
         this.woodenBoat(Items.MANGROVE_BOAT, Blocks.MANGROVE_PLANKS);
-        this.shelf(Blocks.ACACIA_SHELF, Items.STRIPPED_ACACIA_LOG);
-        this.shelf(Blocks.BAMBOO_SHELF, Items.STRIPPED_BAMBOO_BLOCK);
-        this.shelf(Blocks.BIRCH_SHELF, Items.STRIPPED_BIRCH_LOG);
-        this.shelf(Blocks.CHERRY_SHELF, Items.STRIPPED_CHERRY_LOG);
-        this.shelf(Blocks.CRIMSON_SHELF, Items.STRIPPED_CRIMSON_STEM);
-        this.shelf(Blocks.DARK_OAK_SHELF, Items.STRIPPED_DARK_OAK_LOG);
-        this.shelf(Blocks.JUNGLE_SHELF, Items.STRIPPED_JUNGLE_LOG);
-        this.shelf(Blocks.MANGROVE_SHELF, Items.STRIPPED_MANGROVE_LOG);
-        this.shelf(Blocks.OAK_SHELF, Items.STRIPPED_OAK_LOG);
-        this.shelf(Blocks.PALE_OAK_SHELF, Items.STRIPPED_PALE_OAK_LOG);
-        this.shelf(Blocks.SPRUCE_SHELF, Items.STRIPPED_SPRUCE_LOG);
-        this.shelf(Blocks.WARPED_SHELF, Items.STRIPPED_WARPED_STEM);
         List<Item> list = List.of(
             Items.BLACK_DYE,
             Items.BLUE_DYE,
@@ -621,14 +_,6 @@
                     )
             )
             .save(this.output);
-        this.shaped(RecipeCategory.DECORATIONS, Blocks.COPPER_CHEST)
-            .define('#', Items.COPPER_INGOT)
-            .define('X', Items.CHEST)
-            .pattern("###")
-            .pattern("#X#")
-            .pattern("###")
-            .unlockedBy("has_copper_chest", this.has(Items.COPPER_CHEST))
-            .save(this.output);
         this.shapeless(RecipeCategory.TRANSPORTATION, Items.CHEST_MINECART)
             .requires(Blocks.CHEST)
             .requires(Items.MINECART)
@@ -1131,12 +_,6 @@
             .pattern("###")
             .unlockedBy("has_iron_ingot", this.has(Items.IRON_INGOT))
             .save(this.output);
-        this.shaped(RecipeCategory.DECORATIONS, Blocks.COPPER_BARS.unaffected(), 16)
-            .define('#', Items.COPPER_INGOT)
-            .pattern("###")
-            .pattern("###")
-            .unlockedBy("has_copper_ingot", this.has(Items.COPPER_INGOT))
-            .save(this.output);
         this.shaped(RecipeCategory.COMBAT, Items.IRON_BOOTS)
             .define('X', Items.IRON_INGOT)
             .pattern("X X")
@@ -1277,75 +_,6 @@
             .pattern("X#X")
             .unlockedBy("has_leather", this.has(Items.LEATHER))
             .save(this.output);
-        this.nineBlockStorageRecipesWithCustomPacking(
-            RecipeCategory.MISC, Items.COPPER_NUGGET, RecipeCategory.MISC, Items.COPPER_INGOT, "copper_ingot_from_nuggets", "copper_ingot"
-        );
-        this.shaped(RecipeCategory.TOOLS, Items.COPPER_AXE)
-            .define('#', Items.STICK)
-            .define('X', ItemTags.COPPER_TOOL_MATERIALS)
-            .pattern("XX")
-            .pattern("X#")
-            .pattern(" #")
-            .unlockedBy("has_copper_ingot", this.has(ItemTags.COPPER_TOOL_MATERIALS))
-            .save(this.output);
-        this.shaped(RecipeCategory.TOOLS, Items.COPPER_HOE)
-            .define('#', Items.STICK)
-            .define('X', ItemTags.COPPER_TOOL_MATERIALS)
-            .pattern("XX")
-            .pattern(" #")
-            .pattern(" #")
-            .unlockedBy("has_copper_ingot", this.has(ItemTags.COPPER_TOOL_MATERIALS))
-            .save(this.output);
-        this.shaped(RecipeCategory.TOOLS, Items.COPPER_PICKAXE)
-            .define('#', Items.STICK)
-            .define('X', ItemTags.COPPER_TOOL_MATERIALS)
-            .pattern("XXX")
-            .pattern(" # ")
-            .pattern(" # ")
-            .unlockedBy("has_copper_ingot", this.has(ItemTags.COPPER_TOOL_MATERIALS))
-            .save(this.output);
-        this.shaped(RecipeCategory.TOOLS, Items.COPPER_SHOVEL)
-            .define('#', Items.STICK)
-            .define('X', ItemTags.COPPER_TOOL_MATERIALS)
-            .pattern("X")
-            .pattern("#")
-            .pattern("#")
-            .unlockedBy("has_copper_ingot", this.has(ItemTags.COPPER_TOOL_MATERIALS))
-            .save(this.output);
-        this.shaped(RecipeCategory.COMBAT, Items.COPPER_SWORD)
-            .define('#', Items.STICK)
-            .define('X', ItemTags.COPPER_TOOL_MATERIALS)
-            .pattern("X")
-            .pattern("X")
-            .pattern("#")
-            .unlockedBy("has_copper_ingot", this.has(ItemTags.COPPER_TOOL_MATERIALS))
-            .save(this.output);
-        this.shaped(RecipeCategory.COMBAT, Items.COPPER_BOOTS)
-            .define('X', Items.COPPER_INGOT)
-            .pattern("X X")
-            .pattern("X X")
-            .unlockedBy("has_copper_ingot", this.has(Items.COPPER_INGOT))
-            .save(this.output);
-        this.shaped(RecipeCategory.COMBAT, Items.COPPER_CHESTPLATE)
-            .define('X', Items.COPPER_INGOT)
-            .pattern("X X")
-            .pattern("XXX")
-            .pattern("XXX")
-            .unlockedBy("has_copper_ingot", this.has(Items.COPPER_INGOT))
-            .save(this.output);
-        this.shaped(RecipeCategory.COMBAT, Items.COPPER_HELMET)
-            .define('X', Items.COPPER_INGOT)
-            .pattern("XXX")
-            .pattern("X X")
-            .unlockedBy("has_copper_ingot", this.has(Items.COPPER_INGOT))
-            .save(this.output);
-        this.shaped(RecipeCategory.COMBAT, Items.COPPER_LEGGINGS)
-            .define('X', Items.COPPER_INGOT)
-            .pattern("XXX")
-            .pattern("X X")
-            .pattern("X X")
-            .unlockedBy("has_copper_ingot", this.has(Items.COPPER_INGOT))
-            .save(this.output);
         this.shaped(RecipeCategory.REDSTONE, Blocks.LECTERN)
             .define('S', ItemTags.WOODEN_SLABS)
             .define('B', Blocks.BOOKSHELF)
@@ -1923,15 +_,6 @@
             .pattern("S")
             .unlockedBy("has_soul_sand", this.has(ItemTags.SOUL_FIRE_BASE_BLOCKS))
             .save(this.output);
-        this.shaped(RecipeCategory.DECORATIONS, Blocks.COPPER_TORCH, 4)
-            .define('X', Ingredient.of(Items.COAL, Items.CHARCOAL))
-            .define('#', Items.STICK)
-            .define('C', Items.COPPER_NUGGET)
-            .pattern("C")
-            .pattern("X")
-            .pattern("#")
-            .unlockedBy("has_copper_nugget", this.has(Items.COPPER_NUGGET))
-            .save(this.output);
         this.shaped(RecipeCategory.DECORATIONS, Blocks.LANTERN)
             .define('#', Items.TORCH)
             .define('X', Items.IRON_NUGGET)
@@ -1949,14 +_,6 @@
             .pattern("XXX")
             .unlockedBy("has_soul_torch", this.has(Items.SOUL_TORCH))
             .save(this.output);
-        this.shaped(RecipeCategory.DECORATIONS, Blocks.COPPER_LANTERN.unaffected())
-            .define('#', Items.COPPER_TORCH)
-            .define('X', Items.COPPER_NUGGET)
-            .pattern("XXX")
-            .pattern("X#X")
-            .pattern("XXX")
-            .unlockedBy("has_copper_torch", this.has(Items.COPPER_TORCH))
-            .save(this.output);
         this.shapeless(RecipeCategory.REDSTONE, Blocks.TRAPPED_CHEST)
             .requires(Blocks.CHEST)
             .requires(Blocks.TRIPWIRE_HOOK)
@@ -2181,7 +_,7 @@
             .pattern("OOO")
             .unlockedBy("has_obsidian", this.has(Blocks.CRYING_OBSIDIAN))
             .save(this.output);
-        this.shaped(RecipeCategory.DECORATIONS, Blocks.IRON_CHAIN)
+        this.shaped(RecipeCategory.DECORATIONS, Blocks.CHAIN)
             .define('I', Items.IRON_INGOT)
             .define('N', Items.IRON_NUGGET)
             .pattern("N")
@@ -2190,15 +_,6 @@
             .unlockedBy("has_iron_nugget", this.has(Items.IRON_NUGGET))
             .unlockedBy("has_iron_ingot", this.has(Items.IRON_INGOT))
             .save(this.output);
-        this.shaped(RecipeCategory.DECORATIONS, Blocks.COPPER_CHAIN.unaffected())
-            .define('I', Items.COPPER_INGOT)
-            .define('N', Items.COPPER_NUGGET)
-            .pattern("N")
-            .pattern("I")
-            .pattern("N")
-            .unlockedBy("has_copper_nugget", this.has(Items.COPPER_NUGGET))
-            .unlockedBy("has_copper_ingot", this.has(Items.COPPER_INGOT))
-            .save(this.output);
         this.shaped(RecipeCategory.BUILDING_BLOCKS, Blocks.TINTED_GLASS, 2)
             .define('G', Blocks.GLASS)
             .define('S', Items.AMETHYST_SHARD)
@@ -2322,35 +_,6 @@
             .save(this.output, getSmeltingRecipeName(Items.GOLD_NUGGET));
         SimpleCookingRecipeBuilder.smelting(
                 Ingredient.of(
-                    Items.COPPER_PICKAXE,
-                    Items.COPPER_SHOVEL,
-                    Items.COPPER_AXE,
-                    Items.COPPER_HOE,
-                    Items.COPPER_SWORD,
-                    Items.COPPER_HELMET,
-                    Items.COPPER_CHESTPLATE,
-                    Items.COPPER_LEGGINGS,
-                    Items.COPPER_BOOTS,
-                    Items.COPPER_HORSE_ARMOR
-                ),
-                RecipeCategory.MISC,
-                Items.COPPER_NUGGET,
-                0.1F,
-                200
-            )
-            .unlockedBy("has_copper_pickaxe", this.has(Items.COPPER_PICKAXE))
-            .unlockedBy("has_copper_shovel", this.has(Items.COPPER_SHOVEL))
-            .unlockedBy("has_copper_axe", this.has(Items.COPPER_AXE))
-            .unlockedBy("has_copper_hoe", this.has(Items.COPPER_HOE))
-            .unlockedBy("has_copper_sword", this.has(Items.COPPER_SWORD))
-            .unlockedBy("has_copper_helmet", this.has(Items.COPPER_HELMET))
-            .unlockedBy("has_copper_chestplate", this.has(Items.COPPER_CHESTPLATE))
-            .unlockedBy("has_copper_leggings", this.has(Items.COPPER_LEGGINGS))
-            .unlockedBy("has_copper_boots", this.has(Items.COPPER_BOOTS))
-            .unlockedBy("has_copper_horse_armor", this.has(Items.COPPER_HORSE_ARMOR))
-            .save(this.output, getSmeltingRecipeName(Items.COPPER_NUGGET));
-        SimpleCookingRecipeBuilder.smelting(
-                Ingredient.of(
                     Items.IRON_PICKAXE,
                     Items.IRON_SHOVEL,
                     Items.IRON_AXE,
@@ -2553,35 +_,6 @@
             .save(this.output, getBlastingRecipeName(Items.GOLD_NUGGET));
         SimpleCookingRecipeBuilder.blasting(
                 Ingredient.of(
-                    Items.COPPER_PICKAXE,
-                    Items.COPPER_SHOVEL,
-                    Items.COPPER_AXE,
-                    Items.COPPER_HOE,
-                    Items.COPPER_SWORD,
-                    Items.COPPER_HELMET,
-                    Items.COPPER_CHESTPLATE,
-                    Items.COPPER_LEGGINGS,
-                    Items.COPPER_BOOTS,
-                    Items.COPPER_HORSE_ARMOR
-                ),
-                RecipeCategory.MISC,
-                Items.COPPER_NUGGET,
-                0.1F,
-                100
-            )
-            .unlockedBy("has_copper_pickaxe", this.has(Items.COPPER_PICKAXE))
-            .unlockedBy("has_copper_shovel", this.has(Items.COPPER_SHOVEL))
-            .unlockedBy("has_copper_axe", this.has(Items.COPPER_AXE))
-            .unlockedBy("has_copper_hoe", this.has(Items.COPPER_HOE))
-            .unlockedBy("has_copper_sword", this.has(Items.COPPER_SWORD))
-            .unlockedBy("has_copper_helmet", this.has(Items.COPPER_HELMET))
-            .unlockedBy("has_copper_chestplate", this.has(Items.COPPER_CHESTPLATE))
-            .unlockedBy("has_copper_leggings", this.has(Items.COPPER_LEGGINGS))
-            .unlockedBy("has_copper_boots", this.has(Items.COPPER_BOOTS))
-            .unlockedBy("has_copper_horse_armor", this.has(Items.COPPER_HORSE_ARMOR))
-            .save(this.output, getBlastingRecipeName(Items.COPPER_NUGGET));
-        SimpleCookingRecipeBuilder.blasting(
-                Ingredient.of(
                     Items.IRON_PICKAXE,
                     Items.IRON_SHOVEL,
                     Items.IRON_AXE,
@@ -3020,7 +_,9 @@
         this.doorBuilder(Blocks.COPPER_DOOR, Ingredient.of(Items.COPPER_INGOT))
             .unlockedBy(getHasName(Items.COPPER_INGOT), this.has(Items.COPPER_INGOT))
             .save(this.output);
-        this.twoByTwoPacker(RecipeCategory.REDSTONE, Blocks.COPPER_TRAPDOOR, Items.COPPER_INGOT);
+        this.trapdoorBuilder(Blocks.COPPER_TRAPDOOR, Ingredient.of(Items.COPPER_INGOT))
+            .unlockedBy(getHasName(Items.COPPER_INGOT), this.has(Items.COPPER_INGOT))
+            .save(this.output);
         this.shaped(RecipeCategory.TOOLS, Items.BUNDLE)
             .define('-', Items.STRING)
             .define('#', Items.LEATHER)
