--- a/net/minecraft/client/gui/render/state/BlitRenderState.java
+++ b/net/minecraft/client/gui/render/state/BlitRenderState.java
@@ -60,11 +_,11 @@
     }
 
     @Override
-    public void buildVertices(VertexConsumer p_415779_) {
-        p_415779_.addVertexWith2DPose(this.pose(), this.x0(), this.y0()).setUv(this.u0(), this.v0()).setColor(this.color());
-        p_415779_.addVertexWith2DPose(this.pose(), this.x0(), this.y1()).setUv(this.u0(), this.v1()).setColor(this.color());
-        p_415779_.addVertexWith2DPose(this.pose(), this.x1(), this.y1()).setUv(this.u1(), this.v1()).setColor(this.color());
-        p_415779_.addVertexWith2DPose(this.pose(), this.x1(), this.y0()).setUv(this.u1(), this.v0()).setColor(this.color());
+    public void buildVertices(VertexConsumer p_415779_, float p_418375_) {
+        p_415779_.addVertexWith2DPose(this.pose(), this.x0(), this.y0(), p_418375_).setUv(this.u0(), this.v0()).setColor(this.color());
+        p_415779_.addVertexWith2DPose(this.pose(), this.x0(), this.y1(), p_418375_).setUv(this.u0(), this.v1()).setColor(this.color());
+        p_415779_.addVertexWith2DPose(this.pose(), this.x1(), this.y1(), p_418375_).setUv(this.u1(), this.v1()).setColor(this.color());
+        p_415779_.addVertexWith2DPose(this.pose(), this.x1(), this.y0(), p_418375_).setUv(this.u1(), this.v0()).setColor(this.color());
     }
 
     @Nullable
