--- a/net/minecraft/client/gui/render/state/ColoredRectangleRenderState.java
+++ b/net/minecraft/client/gui/render/state/ColoredRectangleRenderState.java
@@ -51,11 +_,11 @@
     }
 
     @Override
-    public void buildVertices(VertexConsumer p_415536_) {
-        p_415536_.addVertexWith2DPose(this.pose(), this.x0(), this.y0()).setColor(this.col1());
-        p_415536_.addVertexWith2DPose(this.pose(), this.x0(), this.y1()).setColor(this.col2());
-        p_415536_.addVertexWith2DPose(this.pose(), this.x1(), this.y1()).setColor(this.col2());
-        p_415536_.addVertexWith2DPose(this.pose(), this.x1(), this.y0()).setColor(this.col1());
+    public void buildVertices(VertexConsumer p_415536_, float p_418064_) {
+        p_415536_.addVertexWith2DPose(this.pose(), this.x0(), this.y0(), p_418064_).setColor(this.col1());
+        p_415536_.addVertexWith2DPose(this.pose(), this.x0(), this.y1(), p_418064_).setColor(this.col2());
+        p_415536_.addVertexWith2DPose(this.pose(), this.x1(), this.y1(), p_418064_).setColor(this.col2());
+        p_415536_.addVertexWith2DPose(this.pose(), this.x1(), this.y0(), p_418064_).setColor(this.col1());
     }
 
     @Nullable
