--- a/net/minecraft/client/gui/screens/BackupConfirmScreen.java
+++ b/net/minecraft/client/gui/screens/BackupConfirmScreen.java
@@ -66,7 +_,7 @@
     public void render(GuiGraphics p_282759_, int p_282356_, int p_282725_, float p_281518_) {
         super.render(p_282759_, p_282356_, p_282725_, p_281518_);
         p_282759_.drawCenteredString(this.font, this.title, this.width / 2, 50, -1);
-        this.message.render(p_282759_, MultiLineLabel.Align.CENTER, this.width / 2, 70, 9, true, -1);
+        this.message.renderCentered(p_282759_, this.width / 2, 70);
     }
 
     @Override
