--- a/net/minecraft/client/gui/screens/ChatScreen.java
+++ b/net/minecraft/client/gui/screens/ChatScreen.java
@@ -1,7 +_,6 @@
 package net.minecraft.client.gui.screens;
 
 import javax.annotation.Nullable;
-import net.minecraft.ChatFormatting;
 import net.minecraft.client.GuiMessageTag;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiGraphics;
@@ -13,7 +_,6 @@
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.MutableComponent;
 import net.minecraft.network.chat.Style;
-import net.minecraft.util.FormattedCharSequence;
 import net.minecraft.util.Mth;
 import net.minecraft.util.StringUtil;
 import net.neoforged.api.distmarker.Dist;
@@ -28,15 +_,12 @@
     private String historyBuffer = "";
     private int historyPos = -1;
     protected EditBox input;
-    protected String initial;
-    protected boolean isDraft;
-    protected ChatScreen.ExitReason exitReason = ChatScreen.ExitReason.INTERRUPTED;
+    private String initial;
     CommandSuggestions commandSuggestions;
 
-    public ChatScreen(String p_95579_, boolean p_437367_) {
+    public ChatScreen(String p_95579_) {
         super(Component.translatable("chat_screen.title"));
         this.initial = p_95579_;
-        this.isDraft = p_437367_;
     }
 
     @Override
@@ -52,12 +_,10 @@
         this.input.setBordered(false);
         this.input.setValue(this.initial);
         this.input.setResponder(this::onEdited);
-        this.input.addFormatter(this::formatChat);
         this.input.setCanLoseFocus(false);
         this.addRenderableWidget(this.input);
         this.commandSuggestions = new CommandSuggestions(this.minecraft, this, this.input, this.font, false, false, 1, 10, true, -805306368);
         this.commandSuggestions.setAllowHiding(false);
-        this.commandSuggestions.setAllowSuggestions(false);
         this.commandSuggestions.updateCommandInfo();
     }
 
@@ -68,73 +_,53 @@
 
     @Override
     public void resize(Minecraft p_95600_, int p_95601_, int p_95602_) {
-        this.initial = this.input.getValue();
+        String s = this.input.getValue();
         this.init(p_95600_, p_95601_, p_95602_);
-    }
-
-    @Override
-    public void onClose() {
-        this.exitReason = ChatScreen.ExitReason.INTENTIONAL;
-        super.onClose();
+        this.setChatLine(s);
+        this.commandSuggestions.updateCommandInfo();
     }
 
     @Override
     public void removed() {
         this.minecraft.gui.getChat().resetChatScroll();
-        this.initial = this.input.getValue();
-        if (this.shouldDiscardDraft() || StringUtils.isBlank(this.initial)) {
-            this.minecraft.gui.getChat().discardDraft();
-        } else if (!this.isDraft) {
-            this.minecraft.gui.getChat().saveAsDraft(this.initial);
-        }
-    }
-
-    protected boolean shouldDiscardDraft() {
-        return this.exitReason != ChatScreen.ExitReason.INTERRUPTED
-            && (this.exitReason != ChatScreen.ExitReason.INTENTIONAL || !this.minecraft.options.saveChatDrafts().get());
     }
 
     private void onEdited(String p_95611_) {
-        this.commandSuggestions.setAllowSuggestions(true);
+        String s = this.input.getValue();
+        this.commandSuggestions.setAllowSuggestions(!s.equals(this.initial));
         this.commandSuggestions.updateCommandInfo();
-        this.isDraft = false;
     }
 
     @Override
     public boolean keyPressed(int p_95591_, int p_95592_, int p_95593_) {
         if (this.commandSuggestions.keyPressed(p_95591_, p_95592_, p_95593_)) {
             return true;
-        } else if (this.isDraft && p_95591_ == 259) {
-            this.input.setValue("");
-            this.isDraft = false;
-            return true;
         } else if (super.keyPressed(p_95591_, p_95592_, p_95593_)) {
             return true;
-        } else {
-            switch (p_95591_) {
-                case 257:
-                case 335:
-                    this.handleChatInput(this.input.getValue(), true);
-                    this.exitReason = ChatScreen.ExitReason.DONE;
-                    this.minecraft.setScreen(null);
-                    break;
-                case 264:
-                    this.moveInHistory(1);
-                    break;
-                case 265:
-                    this.moveInHistory(-1);
-                    break;
-                case 266:
-                    this.minecraft.gui.getChat().scrollChat(this.minecraft.gui.getChat().getLinesPerPage() - 1);
-                    break;
-                case 267:
-                    this.minecraft.gui.getChat().scrollChat(-this.minecraft.gui.getChat().getLinesPerPage() + 1);
-                    break;
-                default:
-                    return false;
+        } else if (p_95591_ == 256) {
+            this.minecraft.setScreen(null);
+            return true;
+        } else if (p_95591_ == 257 || p_95591_ == 335) {
+            this.handleChatInput(this.input.getValue(), true);
+            // FORGE: Prevent closing the screen if another screen has been opened.
+            if (minecraft.screen == this) {
+                this.minecraft.setScreen(null);
             }
-
-            return true;
+            return true;
+        } else if (p_95591_ == 265) {
+            this.moveInHistory(-1);
+            return true;
+        } else if (p_95591_ == 264) {
+            this.moveInHistory(1);
+            return true;
+        } else if (p_95591_ == 266) {
+            this.minecraft.gui.getChat().scrollChat(this.minecraft.gui.getChat().getLinesPerPage() - 1);
+            return true;
+        } else if (p_95591_ == 267) {
+            this.minecraft.gui.getChat().scrollChat(-this.minecraft.gui.getChat().getLinesPerPage() + 1);
+            return true;
+        } else {
+            return false;
         }
     }
 
@@ -154,7 +_,7 @@
     }
 
     @Override
-    public boolean mouseClicked(double p_95585_, double p_95586_, int p_95587_, boolean p_434924_) {
+    public boolean mouseClicked(double p_95585_, double p_95586_, int p_95587_) {
         if (this.commandSuggestions.mouseClicked((int)p_95585_, (int)p_95586_, p_95587_)) {
             return true;
         } else {
@@ -171,12 +_,12 @@
                 }
             }
 
-            return this.input.mouseClicked(p_95585_, p_95586_, p_95587_, p_434924_) ? true : super.mouseClicked(p_95585_, p_95586_, p_95587_, p_434924_);
+            return this.input.mouseClicked(p_95585_, p_95586_, p_95587_) ? true : super.mouseClicked(p_95585_, p_95586_, p_95587_);
         }
     }
 
     @Override
-    public void insertText(String p_95606_, boolean p_95607_) {
+    protected void insertText(String p_95606_, boolean p_95607_) {
         if (p_95607_) {
             this.input.setValue(p_95606_);
         } else {
@@ -204,11 +_,6 @@
         }
     }
 
-    @Nullable
-    private FormattedCharSequence formatChat(String p_437302_, int p_437324_) {
-        return this.isDraft ? FormattedCharSequence.forward(p_437302_, Style.EMPTY.withColor(ChatFormatting.GRAY).withItalic(true)) : null;
-    }
-
     @Override
     public void render(GuiGraphics p_282470_, int p_282674_, int p_282014_, float p_283132_) {
         p_282470_.fill(2, this.height - 14, this.width - 2, this.height - 2, this.minecraft.options.getBackgroundColor(Integer.MIN_VALUE));
@@ -235,9 +_,8 @@
         return false;
     }
 
-    @Override
-    public boolean isAllowedInPortal() {
-        return true;
+    private void setChatLine(String p_95613_) {
+        this.input.setValue(p_95613_);
     }
 
     @Override
@@ -272,18 +_,5 @@
 
     public String normalizeChatMessage(String p_232707_) {
         return StringUtil.trimChatMessage(StringUtils.normalizeSpace(p_232707_.trim()));
-    }
-
-    @FunctionalInterface
-    @OnlyIn(Dist.CLIENT)
-    public interface ChatConstructor<T extends ChatScreen> {
-        T create(String p_437286_, boolean p_437298_);
-    }
-
-    @OnlyIn(Dist.CLIENT)
-    protected static enum ExitReason {
-        INTENTIONAL,
-        INTERRUPTED,
-        DONE;
     }
 }
