--- a/net/minecraft/client/gui/screens/ConfirmLinkScreen.java
+++ b/net/minecraft/client/gui/screens/ConfirmLinkScreen.java
@@ -15,6 +_,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class ConfirmLinkScreen extends ConfirmScreen {
+    private static final Component COPY_BUTTON_TEXT = Component.translatable("chat.copy");
     private static final Component WARNING_TEXT = Component.translatable("chat.link.warning").withColor(-13108);
     private static final int BUTTON_WIDTH = 100;
     private final String url;
@@ -71,7 +_,7 @@
     @Override
     protected void addButtons(LinearLayout p_428302_) {
         this.yesButton = p_428302_.addChild(Button.builder(this.yesButtonComponent, p_169249_ -> this.callback.accept(true)).width(100).build());
-        p_428302_.addChild(Button.builder(CommonComponents.GUI_COPY_TO_CLIPBOARD, p_169247_ -> {
+        p_428302_.addChild(Button.builder(COPY_BUTTON_TEXT, p_169247_ -> {
             this.copyToClipboard();
             this.callback.accept(false);
         }).width(100).build());
