--- a/net/minecraft/world/level/block/AmethystBlock.java
+++ b/net/minecraft/world/level/block/AmethystBlock.java
@@ -24,7 +_,7 @@
 
     @Override
     protected void onProjectileHit(Level p_152001_, BlockState p_152002_, BlockHitResult p_152003_, Projectile p_152004_) {
-        if (!p_152001_.isClientSide()) {
+        if (!p_152001_.isClientSide) {
             BlockPos blockpos = p_152003_.getBlockPos();
             p_152001_.playSound(null, blockpos, SoundEvents.AMETHYST_BLOCK_CHIME, SoundSource.BLOCKS, 1.0F, 0.5F + p_152001_.random.nextFloat() * 1.2F);
         }
