--- a/net/minecraft/world/level/block/BeaconBlock.java
+++ b/net/minecraft/world/level/block/BeaconBlock.java
@@ -46,7 +_,7 @@
 
     @Override
     protected InteractionResult useWithoutItem(BlockState p_49432_, Level p_49433_, BlockPos p_49434_, Player p_49435_, BlockHitResult p_49437_) {
-        if (!p_49433_.isClientSide() && p_49433_.getBlockEntity(p_49434_) instanceof BeaconBlockEntity beaconblockentity) {
+        if (!p_49433_.isClientSide && p_49433_.getBlockEntity(p_49434_) instanceof BeaconBlockEntity beaconblockentity) {
             p_49435_.openMenu(beaconblockentity);
             p_49435_.awardStat(Stats.INTERACT_WITH_BEACON);
         }
