--- a/net/minecraft/world/level/block/BigDripleafBlock.java
+++ b/net/minecraft/world/level/block/BigDripleafBlock.java
@@ -141,6 +_,8 @@
     protected boolean canSurvive(BlockState p_152289_, LevelReader p_152290_, BlockPos p_152291_) {
         BlockPos blockpos = p_152291_.below();
         BlockState blockstate = p_152290_.getBlockState(blockpos);
+        var soilDecision = blockstate.canSustainPlant(p_152290_, blockpos, Direction.UP, p_152289_);
+        if (!soilDecision.isDefault()) return soilDecision.isTrue();
         return blockstate.is(this) || blockstate.is(Blocks.BIG_DRIPLEAF_STEM) || blockstate.is(BlockTags.BIG_DRIPLEAF_PLACEABLE);
     }
 
@@ -192,7 +_,7 @@
 
     @Override
     protected void entityInside(BlockState p_152266_, Level p_152267_, BlockPos p_152268_, Entity p_152269_, InsideBlockEffectApplier p_405262_) {
-        if (!p_152267_.isClientSide()) {
+        if (!p_152267_.isClientSide) {
             if (p_152266_.getValue(TILT) == Tilt.NONE && canEntityTilt(p_152268_, p_152269_) && !p_152267_.hasNeighborSignal(p_152268_)) {
                 this.setTiltAndScheduleTick(p_152266_, p_152267_, p_152268_, Tilt.UNSTABLE, null);
             }
