--- a/net/minecraft/world/level/block/BlockTypes.java
+++ b/net/minecraft/world/level/block/BlockTypes.java
@@ -68,8 +_,6 @@
         Registry.register(p_304881_, "concrete_powder", ConcretePowderBlock.CODEC);
         Registry.register(p_304881_, "conduit", ConduitBlock.CODEC);
         Registry.register(p_304881_, "copper_bulb_block", CopperBulbBlock.CODEC);
-        Registry.register(p_304881_, "copper_chest", CopperChestBlock.CODEC);
-        Registry.register(p_304881_, "copper_golem_statue", CopperGolemStatueBlock.CODEC);
         Registry.register(p_304881_, "coral", CoralBlock.CODEC);
         Registry.register(p_304881_, "coral_fan", CoralFanBlock.CODEC);
         Registry.register(p_304881_, "coral_plant", CoralPlantBlock.CODEC);
@@ -103,6 +_,7 @@
         Registry.register(p_304881_, "fence_gate", FenceGateBlock.CODEC);
         Registry.register(p_304881_, "fire", FireBlock.CODEC);
         Registry.register(p_304881_, "firefly_bush", FireflyBushBlock.CODEC);
+        Registry.register(p_304881_, "fletching_table", FletchingTableBlock.CODEC);
         Registry.register(p_304881_, "flower", FlowerBlock.CODEC);
         Registry.register(p_304881_, "flower_pot", FlowerPotBlock.CODEC);
         Registry.register(p_304881_, "frogspawn", FrogspawnBlock.CODEC);
@@ -193,7 +_,6 @@
         Registry.register(p_304881_, "sculk_vein", SculkVeinBlock.CODEC);
         Registry.register(p_304881_, "seagrass", SeagrassBlock.CODEC);
         Registry.register(p_304881_, "sea_pickle", SeaPickleBlock.CODEC);
-        Registry.register(p_304881_, "shelf", ShelfBlock.CODEC);
         Registry.register(p_304881_, "short_dry_grass", ShortDryGrassBlock.CODEC);
         Registry.register(p_304881_, "shulker_box", ShulkerBoxBlock.CODEC);
         Registry.register(p_304881_, "skull", SkullBlock.CODEC);
@@ -253,19 +_,13 @@
         Registry.register(p_304881_, "wall", WallBlock.CODEC);
         Registry.register(p_304881_, "waterlily", WaterlilyBlock.CODEC);
         Registry.register(p_304881_, "waterlogged_transparent", WaterloggedTransparentBlock.CODEC);
-        Registry.register(p_304881_, "weathering_copper_bar", WeatheringCopperBarsBlock.CODEC);
         Registry.register(p_304881_, "weathering_copper_bulb", WeatheringCopperBulbBlock.CODEC);
-        Registry.register(p_304881_, "weathering_copper_chain", WeatheringCopperChainBlock.CODEC);
-        Registry.register(p_304881_, "weathering_copper_chest", WeatheringCopperChestBlock.CODEC);
         Registry.register(p_304881_, "weathering_copper_door", WeatheringCopperDoorBlock.CODEC);
         Registry.register(p_304881_, "weathering_copper_full", WeatheringCopperFullBlock.CODEC);
-        Registry.register(p_304881_, "weathering_copper_golem_statue", WeatheringCopperGolemStatueBlock.CODEC);
         Registry.register(p_304881_, "weathering_copper_grate", WeatheringCopperGrateBlock.CODEC);
         Registry.register(p_304881_, "weathering_copper_slab", WeatheringCopperSlabBlock.CODEC);
         Registry.register(p_304881_, "weathering_copper_stair", WeatheringCopperStairBlock.CODEC);
         Registry.register(p_304881_, "weathering_copper_trap_door", WeatheringCopperTrapDoorBlock.CODEC);
-        Registry.register(p_304881_, "weathering_lantern", WeatheringLanternBlock.CODEC);
-        Registry.register(p_304881_, "weathering_lightning_rod", WeatheringLightningRodBlock.CODEC);
         Registry.register(p_304881_, "web", WebBlock.CODEC);
         Registry.register(p_304881_, "weeping_vines_plant", WeepingVinesPlantBlock.CODEC);
         Registry.register(p_304881_, "weeping_vines", WeepingVinesBlock.CODEC);
