--- a/net/minecraft/world/level/block/DragonEggBlock.java
+++ b/net/minecraft/world/level/block/DragonEggBlock.java
@@ -54,8 +_,8 @@
                 p_52937_.random.nextInt(8) - p_52937_.random.nextInt(8),
                 p_52937_.random.nextInt(16) - p_52937_.random.nextInt(16)
             );
-            if (p_52937_.getBlockState(blockpos).isAir() && worldborder.isWithinBounds(blockpos) && !p_52937_.isOutsideBuildHeight(blockpos)) {
-                if (p_52937_.isClientSide()) {
+            if (p_52937_.getBlockState(blockpos).isAir() && worldborder.isWithinBounds(blockpos)) {
+                if (p_52937_.isClientSide) {
                     for (int j = 0; j < 128; j++) {
                         double d0 = p_52937_.random.nextDouble();
                         float f = (p_52937_.random.nextFloat() - 0.5F) * 0.2F;
