--- a/net/minecraft/world/level/block/LayeredCauldronBlock.java
+++ b/net/minecraft/world/level/block/LayeredCauldronBlock.java
@@ -5,7 +_,6 @@
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import net.minecraft.Util;
 import net.minecraft.core.BlockPos;
-import net.minecraft.core.Direction;
 import net.minecraft.core.cauldron.CauldronInteraction;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.entity.Entity;
@@ -27,12 +_,12 @@
 
 public class LayeredCauldronBlock extends AbstractCauldronBlock {
     public static final MapCodec<LayeredCauldronBlock> CODEC = RecordCodecBuilder.mapCodec(
-        p_432670_ -> p_432670_.group(
+        p_368428_ -> p_368428_.group(
                 Biome.Precipitation.CODEC.fieldOf("precipitation").forGetter(p_304371_ -> p_304371_.precipitationType),
                 CauldronInteraction.CODEC.fieldOf("interactions").forGetter(p_304369_ -> p_304369_.interactions),
                 propertiesCodec()
             )
-            .apply(p_432670_, LayeredCauldronBlock::new)
+            .apply(p_368428_, LayeredCauldronBlock::new)
     );
     public static final int MIN_FILL_LEVEL = 1;
     public static final int MAX_FILL_LEVEL = 3;
@@ -118,7 +_,7 @@
     }
 
     @Override
-    protected int getAnalogOutputSignal(BlockState p_153530_, Level p_153531_, BlockPos p_153532_, Direction p_432844_) {
+    protected int getAnalogOutputSignal(BlockState p_153530_, Level p_153531_, BlockPos p_153532_) {
         return p_153530_.getValue(LEVEL);
     }
 
