--- a/net/minecraft/world/level/block/PressurePlateBlock.java
+++ b/net/minecraft/world/level/block/PressurePlateBlock.java
@@ -16,8 +_,8 @@
 
 public class PressurePlateBlock extends BasePressurePlateBlock {
     public static final MapCodec<PressurePlateBlock> CODEC = RecordCodecBuilder.mapCodec(
-        p_432675_ -> p_432675_.group(BlockSetType.CODEC.fieldOf("block_set_type").forGetter(p_304917_ -> p_304917_.type), propertiesCodec())
-            .apply(p_432675_, PressurePlateBlock::new)
+        p_368433_ -> p_368433_.group(BlockSetType.CODEC.fieldOf("block_set_type").forGetter(p_304917_ -> p_304917_.type), propertiesCodec())
+            .apply(p_368433_, PressurePlateBlock::new)
     );
     public static final BooleanProperty POWERED = BlockStateProperties.POWERED;
 
