--- a/net/minecraft/world/level/block/WallSkullBlock.java
+++ b/net/minecraft/world/level/block/WallSkullBlock.java
@@ -18,8 +_,8 @@
 
 public class WallSkullBlock extends AbstractSkullBlock {
     public static final MapCodec<WallSkullBlock> CODEC = RecordCodecBuilder.mapCodec(
-        p_432693_ -> p_432693_.group(SkullBlock.Type.CODEC.fieldOf("kind").forGetter(AbstractSkullBlock::getType), propertiesCodec())
-            .apply(p_432693_, WallSkullBlock::new)
+        p_368448_ -> p_368448_.group(SkullBlock.Type.CODEC.fieldOf("kind").forGetter(AbstractSkullBlock::getType), propertiesCodec())
+            .apply(p_368448_, WallSkullBlock::new)
     );
     public static final EnumProperty<Direction> FACING = HorizontalDirectionalBlock.FACING;
     private static final Map<Direction, VoxelShape> SHAPES = Shapes.rotateHorizontal(Block.boxZ(8.0, 8.0, 16.0));
