--- a/net/minecraft/server/network/ConfigurationTask.java
+++ b/net/minecraft/server/network/ConfigurationTask.java
@@ -6,13 +_,13 @@
 public interface ConfigurationTask {
     void start(Consumer<Packet<?>> p_294184_);
 
-    default boolean tick() {
-        return false;
-    }
-
     ConfigurationTask.Type type();
 
     public record Type(String id) {
+        public Type(net.minecraft.resources.ResourceLocation location) {
+            this(location.toString());
+        }
+
         @Override
         public String toString() {
             return this.id;
