--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -22,7 +_,6 @@
 import java.util.Set;
 import java.util.concurrent.CancellationException;
 import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.TimeUnit;
 import java.util.function.BiFunction;
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
@@ -279,20 +_,6 @@
             this.ackBlockChangesUpTo = -1;
         }
 
-        if (this.server.isPaused() || !this.tickPlayer()) {
-            this.keepConnectionAlive();
-            this.chatSpamThrottler.tick();
-            this.dropSpamThrottler.tick();
-            if (this.player.getLastActionTime() > 0L
-                && this.server.getPlayerIdleTimeout() > 0
-                && Util.getMillis() - this.player.getLastActionTime() > TimeUnit.MINUTES.toMillis(this.server.getPlayerIdleTimeout())
-                && !this.player.wonGame) {
-                this.disconnect(Component.translatable("multiplayer.disconnect.idling"));
-            }
-        }
-    }
-
-    private boolean tickPlayer() {
         this.resetPosition();
         this.player.xo = this.player.getX();
         this.player.yo = this.player.getY();
@@ -305,7 +_,7 @@
             if (++this.aboveGroundTickCount > this.getMaximumFlyingTicks(this.player)) {
                 LOGGER.warn("{} was kicked for floating too long!", this.player.getName().getString());
                 this.disconnect(Component.translatable("multiplayer.disconnect.flying"));
-                return true;
+                return;
             }
         } else {
             this.clientIsFloating = false;
@@ -324,7 +_,7 @@
                 if (++this.aboveGroundVehicleTickCount > this.getMaximumFlyingTicks(this.lastVehicle)) {
                     LOGGER.warn("{} was kicked for floating a vehicle too long!", this.player.getName().getString());
                     this.disconnect(Component.translatable("multiplayer.disconnect.flying"));
-                    return true;
+                    return;
                 }
             } else {
                 this.clientVehicleIsFloating = false;
@@ -336,7 +_,14 @@
             this.aboveGroundVehicleTickCount = 0;
         }
 
-        return false;
+        this.keepConnectionAlive();
+        this.chatSpamThrottler.tick();
+        this.dropSpamThrottler.tick();
+        if (this.player.getLastActionTime() > 0L
+            && this.server.getPlayerIdleTimeout() > 0
+            && Util.getMillis() - this.player.getLastActionTime() > this.server.getPlayerIdleTimeout() * 1000L * 60L) {
+            this.disconnect(Component.translatable("multiplayer.disconnect.idling"));
+        }
     }
 
     private int getMaximumFlyingTicks(Entity p_326388_) {
@@ -471,18 +_,21 @@
 
                 if (flag1 && serverlevel.noCollision(entity, aabb) || this.isEntityCollidingWithAnythingNew(serverlevel, entity, aabb, d3, d4, d5)) {
                     entity.absSnapTo(d0, d1, d2, f, f1);
+                    resyncPlayerWithVehicle(entity); // Neo - Resync player position on vehicle moving
                     this.send(ClientboundMoveVehiclePacket.fromEntity(entity));
                     entity.removeLatestMovementRecording();
                     return;
                 }
 
                 entity.absSnapTo(d3, d4, d5, f, f1);
+                resyncPlayerWithVehicle(entity); // Neo - Resync player position on vehicle moving
                 this.player.level().getChunkSource().move(this.player);
                 Vec3 vec3 = new Vec3(entity.getX() - d0, entity.getY() - d1, entity.getZ() - d2);
                 this.handlePlayerKnownMovement(vec3);
                 entity.setOnGroundWithMovement(p_9876_.onGround(), vec3);
                 entity.doCheckFallDamage(vec3.x, vec3.y, vec3.z, p_9876_.onGround());
                 this.player.checkMovementStatistics(vec3.x, vec3.y, vec3.z);
+                this.player.checkRidingStatistics(vec3.x, vec3.y, vec3.z); // Neo: check riding stats too as vanilla checks them in rideTick based on the assumption that Entity#rideTick will move the entity, which we break
                 this.clientVehicleIsFloating = d11 >= -0.03125
                     && !flag
                     && !this.server.isFlightAllowed()
@@ -496,6 +_,23 @@
         }
     }
 
+    private void resyncPlayerWithVehicle(Entity vehicle) {
+        Vec3 oldPos = this.player.position();
+        float yRot = this.player.getYRot();
+        float xRot = this.player.getXRot();
+        float yHeadRot = this.player.getYHeadRot();
+
+        vehicle.positionRider(this.player);
+
+        // preserve old rotation and store old position in xo/yo/zo
+        this.player.setYRot(yRot);
+        this.player.setXRot(xRot);
+        this.player.setYHeadRot(yHeadRot);
+        this.player.xo = oldPos.x;
+        this.player.yo = oldPos.y;
+        this.player.zo = oldPos.z;
+    }
+
     private boolean noBlocksAround(Entity p_9794_) {
         return p_9794_.level()
             .getBlockStates(p_9794_.getBoundingBox().inflate(0.0625).expandTowards(0.0, -0.55, 0.0))
@@ -675,7 +_,7 @@
             if (serverlevel.isLoaded(blockpos)) {
                 BlockState blockstate = serverlevel.getBlockState(blockpos);
                 boolean flag = this.player.hasInfiniteMaterials() && p_382999_.includeData();
-                ItemStack itemstack = blockstate.getCloneItemStack(serverlevel, blockpos, flag);
+                ItemStack itemstack = blockstate.getCloneItemStack(blockpos, serverlevel, flag, player);
                 if (!itemstack.isEmpty()) {
                     if (flag) {
                         addBlockDataToItem(blockstate, serverlevel, blockpos, itemstack);
@@ -1094,7 +_,7 @@
                                         && !flag1
                                         && !this.player.isSpectator()
                                         && !this.server.isFlightAllowed()
-                                        && !this.player.getAbilities().mayfly
+                                        && !this.player.mayFly()
                                         && !this.player.hasEffect(MobEffects.LEVITATION)
                                         && !flag
                                         && !flag3
@@ -1209,9 +_,10 @@
             switch (serverboundplayeractionpacket$action) {
                 case SWAP_ITEM_WITH_OFFHAND:
                     if (!this.player.isSpectator()) {
-                        ItemStack itemstack = this.player.getItemInHand(InteractionHand.OFF_HAND);
-                        this.player.setItemInHand(InteractionHand.OFF_HAND, this.player.getItemInHand(InteractionHand.MAIN_HAND));
-                        this.player.setItemInHand(InteractionHand.MAIN_HAND, itemstack);
+                        var event = net.neoforged.neoforge.common.CommonHooks.onLivingSwapHandItems(this.player);
+                        if (event.isCanceled()) return;
+                        this.player.setItemInHand(InteractionHand.OFF_HAND, event.getItemSwappedToOffHand());
+                        this.player.setItemInHand(InteractionHand.MAIN_HAND, event.getItemSwappedToMainHand());
                         this.player.stopUsingItem();
                     }
 
@@ -1239,7 +_,7 @@
                         .handleBlockBreakAction(
                             blockpos, serverboundplayeractionpacket$action, p_9889_.getDirection(), this.player.level().getMaxY(), p_9889_.getSequence()
                         );
-                    this.ackBlockChangesUpTo(p_9889_.getSequence());
+                    this.player.connection.ackBlockChangesUpTo = p_9889_.getSequence();
                     return;
                 default:
                     throw new IllegalArgumentException("Invalid player action");
@@ -1260,7 +_,7 @@
     public void handleUseItemOn(ServerboundUseItemOnPacket p_9930_) {
         PacketUtils.ensureRunningOnSameThread(p_9930_, this, this.player.level());
         if (this.player.hasClientLoaded()) {
-            this.ackBlockChangesUpTo(p_9930_.getSequence());
+            this.player.connection.ackBlockChangesUpTo = p_9930_.getSequence();
             ServerLevel serverlevel = this.player.level();
             InteractionHand interactionhand = p_9930_.getHand();
             ItemStack itemstack = this.player.getItemInHand(interactionhand);
@@ -1406,7 +_,7 @@
     public void handleChat(ServerboundChatPacket p_9841_) {
         Optional<LastSeenMessages> optional = this.unpackAndApplyLastSeen(p_9841_.lastSeenMessages());
         if (!optional.isEmpty()) {
-            this.tryHandleChat(p_9841_.message(), false, () -> {
+            this.tryHandleChat(p_9841_.message(), () -> {
                 PlayerChatMessage playerchatmessage;
                 try {
                     playerchatmessage = this.getSignedMessage(p_9841_, optional.get());
@@ -1416,8 +_,9 @@
                 }
 
                 CompletableFuture<FilteredText> completablefuture = this.filterTextPacket(playerchatmessage.signedContent());
-                Component component = this.server.getChatDecorator().decorate(this.player, playerchatmessage.decoratedContent());
+                Component component = net.neoforged.neoforge.common.CommonHooks.getServerChatSubmittedDecorator().decorate(this.player, playerchatmessage.decoratedContent());
                 this.chatMessageChain.append(completablefuture, p_300785_ -> {
+                    if (component == null) return; // Forge: ServerChatEvent was canceled if this is null.
                     PlayerChatMessage playerchatmessage1 = playerchatmessage.withUnsignedContent(component).filter(p_300785_.mask());
                     this.broadcastChatMessage(playerchatmessage1);
                 });
@@ -1427,7 +_,7 @@
 
     @Override
     public void handleChatCommand(ServerboundChatCommandPacket p_215225_) {
-        this.tryHandleChat(p_215225_.command(), true, () -> {
+        this.tryHandleChat(p_215225_.command(), () -> {
             this.performUnsignedChatCommand(p_215225_.command());
             this.detectRateSpam();
         });
@@ -1449,7 +_,7 @@
     public void handleSignedChatCommand(ServerboundChatCommandSignedPacket p_338604_) {
         Optional<LastSeenMessages> optional = this.unpackAndApplyLastSeen(p_338604_.lastSeenMessages());
         if (!optional.isEmpty()) {
-            this.tryHandleChat(p_338604_.command(), true, () -> {
+            this.tryHandleChat(p_338604_.command(), () -> {
                 this.performSignedChatCommand(p_338604_, optional.get());
                 this.detectRateSpam();
             });
@@ -1533,10 +_,10 @@
         return commanddispatcher.parse(p_242938_, this.player.createCommandSourceStack());
     }
 
-    private void tryHandleChat(String p_338775_, boolean p_437229_, Runnable p_338235_) {
+    private void tryHandleChat(String p_338775_, Runnable p_338235_) {
         if (isChatMessageIllegal(p_338775_)) {
             this.disconnect(Component.translatable("multiplayer.disconnect.illegal_characters"));
-        } else if (!p_437229_ && this.player.getChatVisibility() == ChatVisiblity.HIDDEN) {
+        } else if (this.player.getChatVisibility() == ChatVisiblity.HIDDEN) {
             this.send(new ClientboundSystemChatPacket(Component.translatable("chat.disabled.options").withStyle(ChatFormatting.RED), false));
         } else {
             this.player.resetLastActionTime();
@@ -1587,8 +_,8 @@
     private void detectRateSpam() {
         this.chatSpamThrottler.increment();
         if (!this.chatSpamThrottler.isUnderThreshold()
-            && !this.server.getPlayerList().isOp(this.player.nameAndId())
-            && !this.server.isSingleplayerOwner(this.player.nameAndId())) {
+            && !this.server.getPlayerList().isOp(this.player.getGameProfile())
+            && !this.server.isSingleplayerOwner(this.player.getGameProfile())) {
             this.disconnect(Component.translatable("disconnect.spam"));
         }
     }
@@ -1751,7 +_,11 @@
 
                             @Override
                             public void onInteraction(InteractionHand p_143682_, Vec3 p_143683_) {
-                                this.performInteraction(p_143682_, (p_143686_, p_143687_, p_143688_) -> p_143687_.interactAt(p_143686_, p_143683_, p_143688_));
+                                this.performInteraction(p_143682_, (p_143686_, p_143687_, p_143688_) -> {
+                                    InteractionResult onInteractEntityAtResult = net.neoforged.neoforge.common.CommonHooks.onInteractEntityAt(player, entity, p_143683_, p_143682_);
+                                    if (onInteractEntityAtResult != null) return onInteractEntityAtResult;
+                                    return p_143687_.interactAt(p_143686_, p_143683_, p_143688_);
+                                });
                             }
 
                             @Override
@@ -1949,14 +_,16 @@
     @Override
     public void handlePlayerAbilities(ServerboundPlayerAbilitiesPacket p_9887_) {
         PacketUtils.ensureRunningOnSameThread(p_9887_, this, this.player.level());
-        this.player.getAbilities().flying = p_9887_.isFlying() && this.player.getAbilities().mayfly;
+        this.player.getAbilities().flying = p_9887_.isFlying() && this.player.mayFly();
     }
 
     @Override
     public void handleClientInformation(ServerboundClientInformationPacket p_301979_) {
         PacketUtils.ensureRunningOnSameThread(p_301979_, this, this.player.level());
         boolean flag = this.player.isModelPartShown(PlayerModelPart.HAT);
+        net.minecraft.server.level.ClientInformation oldInfo = this.player.clientInformation();
         this.player.updateOptions(p_301979_.information());
+        net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.event.entity.player.ClientInformationUpdatedEvent(this.player, oldInfo, p_301979_.information()));
         if (this.player.isModelPartShown(PlayerModelPart.HAT) != flag) {
             this.server.getPlayerList().broadcastAll(new ClientboundPlayerInfoUpdatePacket(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_HAT, this.player));
         }
@@ -2009,7 +_,7 @@
                 this.disconnect(ProfilePublicKey.EXPIRED_PROFILE_PUBLIC_KEY);
             } else {
                 try {
-                    SignatureValidator signaturevalidator = this.server.services().profileKeySignatureValidator();
+                    SignatureValidator signaturevalidator = this.server.getProfileKeySignatureValidator();
                     if (signaturevalidator == null) {
                         LOGGER.warn("Ignoring chat session from {} due to missing Services public key", this.player.getGameProfile().getName());
                         return;
@@ -2032,7 +_,7 @@
             this.connection
                 .setupInboundProtocol(
                     ConfigurationProtocols.SERVERBOUND,
-                    new ServerConfigurationPacketListenerImpl(this.server, this.connection, this.createCookie(this.player.clientInformation()))
+                    new ServerConfigurationPacketListenerImpl(this.server, this.connection, this.createCookie(this.player.clientInformation(), this.connectionType))
                 );
         }
     }
@@ -2067,6 +_,7 @@
 
     @Override
     public void handleCustomPayload(ServerboundCustomPayloadPacket p_333887_) {
+        super.handleCustomPayload(p_333887_); // Neo: Call super to invoke modded payload handling.
     }
 
     @Override
