--- a/net/minecraft/client/gui/screens/options/controls/KeyBindsList.java
+++ b/net/minecraft/client/gui/screens/options/controls/KeyBindsList.java
@@ -2,18 +_,23 @@
 
 import com.google.common.collect.ImmutableList;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.function.Supplier;
+import javax.annotation.Nullable;
 import net.minecraft.ChatFormatting;
 import net.minecraft.client.KeyMapping;
 import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.ComponentPath;
 import net.minecraft.client.gui.GuiGraphics;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.ContainerObjectSelectionList;
-import net.minecraft.client.gui.components.FocusableTextWidget;
 import net.minecraft.client.gui.components.Tooltip;
 import net.minecraft.client.gui.components.events.GuiEventListener;
 import net.minecraft.client.gui.narration.NarratableEntry;
+import net.minecraft.client.gui.narration.NarratedElementType;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.client.gui.navigation.FocusNavigationEvent;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.MutableComponent;
 import net.neoforged.api.distmarker.Dist;
@@ -40,7 +_,7 @@
                 this.addEntry(new KeyBindsList.CategoryEntry(Component.translatable(s1)));
             }
 
-            Component component = Component.translatable(keymapping.getName());
+            Component component = keymapping.getDisplayName();
             int i = p_346132_.font.width(component);
             if (i > this.maxNameWidth) {
                 this.maxNameWidth = i;
@@ -66,28 +_,54 @@
 
     @OnlyIn(Dist.CLIENT)
     public class CategoryEntry extends KeyBindsList.Entry {
-        private final FocusableTextWidget categoryName;
+        final Component name;
+        private final int width;
 
         public CategoryEntry(Component p_345224_) {
-            this.categoryName = new FocusableTextWidget(
-                KeyBindsList.this.getRowWidth(), p_345224_, KeyBindsList.this.minecraft.font, false, FocusableTextWidget.BackgroundFill.ON_FOCUS, 4
-            );
-        }
-
-        @Override
-        public void renderContent(GuiGraphics p_439431_, int p_440559_, int p_439613_, boolean p_440201_, float p_439040_) {
-            this.categoryName.setPosition(KeyBindsList.this.width / 2 - this.categoryName.getWidth() / 2, this.getContentBottom() - 9 - 1);
-            this.categoryName.render(p_439431_, p_440559_, p_439613_, p_439040_);
+            this.name = p_345224_;
+            this.width = KeyBindsList.this.minecraft.font.width(this.name);
+        }
+
+        @Override
+        public void render(
+            GuiGraphics p_345402_,
+            int p_345541_,
+            int p_345306_,
+            int p_346166_,
+            int p_346154_,
+            int p_345075_,
+            int p_346184_,
+            int p_346385_,
+            boolean p_346139_,
+            float p_345189_
+        ) {
+            p_345402_.drawString(KeyBindsList.this.minecraft.font, this.name, KeyBindsList.this.width / 2 - this.width / 2, p_345306_ + p_345075_ - 9 - 1, -1);
+        }
+
+        @Nullable
+        @Override
+        public ComponentPath nextFocusPath(FocusNavigationEvent p_344970_) {
+            return null;
         }
 
         @Override
         public List<? extends GuiEventListener> children() {
-            return List.of(this.categoryName);
+            return Collections.emptyList();
         }
 
         @Override
         public List<? extends NarratableEntry> narratables() {
-            return List.of(this.categoryName);
+            return ImmutableList.of(new NarratableEntry() {
+                @Override
+                public NarratableEntry.NarrationPriority narrationPriority() {
+                    return NarratableEntry.NarrationPriority.HOVERED;
+                }
+
+                @Override
+                public void updateNarration(NarrationElementOutput p_344973_) {
+                    p_344973_.add(NarratedElementType.TITLE, CategoryEntry.this.name);
+                }
+            });
         }
 
         @Override
@@ -125,6 +_,7 @@
                 )
                 .build();
             this.resetButton = Button.builder(RESET_BUTTON_TITLE, p_359096_ -> {
+                this.key.setToDefault();
                 p_345998_.setKey(p_345998_.getDefaultKey());
                 KeyBindsList.this.resetMappingAndUpdateButtons();
             }).bounds(0, 0, 50, 20).createNarration(p_344899_ -> Component.translatable("narrator.controls.reset", p_345196_)).build();
@@ -132,19 +_,30 @@
         }
 
         @Override
-        public void renderContent(GuiGraphics p_438877_, int p_439499_, int p_440337_, boolean p_440415_, float p_439878_) {
+        public void render(
+            GuiGraphics p_345065_,
+            int p_345504_,
+            int p_345678_,
+            int p_344740_,
+            int p_345885_,
+            int p_344888_,
+            int p_345213_,
+            int p_344829_,
+            boolean p_346415_,
+            float p_345934_
+        ) {
             int i = KeyBindsList.this.scrollBarX() - this.resetButton.getWidth() - 10;
-            int j = this.getContentY() - 2;
+            int j = p_345678_ - 2;
             this.resetButton.setPosition(i, j);
-            this.resetButton.render(p_438877_, p_439499_, p_440337_, p_439878_);
+            this.resetButton.render(p_345065_, p_345213_, p_344829_, p_345934_);
             int k = i - 5 - this.changeButton.getWidth();
             this.changeButton.setPosition(k, j);
-            this.changeButton.render(p_438877_, p_439499_, p_440337_, p_439878_);
-            p_438877_.drawString(KeyBindsList.this.minecraft.font, this.name, this.getContentX(), this.getContentYMiddle() - 9 / 2, -1);
+            this.changeButton.render(p_345065_, p_345213_, p_344829_, p_345934_);
+            p_345065_.drawString(KeyBindsList.this.minecraft.font, this.name, p_344740_, p_345678_ + p_344888_ / 2 - 9 / 2, -1);
             if (this.hasCollision) {
                 int l = 3;
                 int i1 = this.changeButton.getX() - 6;
-                p_438877_.fill(i1, this.getContentY() - 1, i1 + 3, this.getContentBottom(), -65536);
+                p_345065_.fill(i1, p_345678_ - 1, i1 + 3, p_345678_ + p_344888_, -65536);
             }
         }
 
@@ -166,13 +_,13 @@
             MutableComponent mutablecomponent = Component.empty();
             if (!this.key.isUnbound()) {
                 for (KeyMapping keymapping : KeyBindsList.this.minecraft.options.keyMappings) {
-                    if (keymapping != this.key && this.key.same(keymapping)) {
+                    if ((keymapping != this.key && this.key.same(keymapping)) || keymapping.hasKeyModifierConflict(this.key)) { // Neo: gracefully handle conflicts like SHIFT vs SHIFT+G
                         if (this.hasCollision) {
                             mutablecomponent.append(", ");
                         }
 
                         this.hasCollision = true;
-                        mutablecomponent.append(Component.translatable(keymapping.getName()));
+                        mutablecomponent.append(keymapping.getDisplayName());
                     }
                 }
             }
