--- a/net/minecraft/client/renderer/entity/state/PlayerRenderState.java
+++ b/net/minecraft/client/renderer/entity/state/PlayerRenderState.java
@@ -18,6 +_,7 @@
     public float capeLean2;
     public int arrowCount;
     public int stingerCount;
+    public int useItemRemainingTicks;
     public boolean isSpectator;
     public boolean showHat = true;
     public boolean showJacket = true;
@@ -29,6 +_,7 @@
     public float fallFlyingTimeInTicks;
     public boolean shouldApplyFlyingYRot;
     public float flyingYRot;
+    public boolean swinging;
     @Nullable
     public Component scoreText;
     @Nullable
