--- a/net/minecraft/client/renderer/special/ShulkerBoxSpecialRenderer.java
+++ b/net/minecraft/client/renderer/special/ShulkerBoxSpecialRenderer.java
@@ -6,8 +_,9 @@
 import com.mojang.serialization.codecs.RecordCodecBuilder;
 import com.mojang.serialization.codecs.RecordCodecBuilder.Instance;
 import java.util.Set;
+import net.minecraft.client.model.geom.EntityModelSet;
+import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.Sheets;
-import net.minecraft.client.renderer.SubmitNodeCollector;
 import net.minecraft.client.renderer.blockentity.ShulkerBoxRenderer;
 import net.minecraft.client.resources.model.Material;
 import net.minecraft.core.Direction;
@@ -33,8 +_,8 @@
     }
 
     @Override
-    public void submit(ItemDisplayContext p_439715_, PoseStack p_439819_, SubmitNodeCollector p_439007_, int p_439286_, int p_439124_, boolean p_438934_) {
-        this.shulkerBoxRenderer.submit(p_439819_, p_439007_, p_439286_, p_439124_, this.orientation, this.openness, null, this.material);
+    public void render(ItemDisplayContext p_387436_, PoseStack p_387209_, MultiBufferSource p_388521_, int p_388851_, int p_387596_, boolean p_388474_) {
+        this.shulkerBoxRenderer.render(p_387209_, p_388521_, p_388851_, p_387596_, this.orientation, this.openness, this.material);
     }
 
     @Override
@@ -67,8 +_,8 @@
         }
 
         @Override
-        public SpecialModelRenderer<?> bake(SpecialModelRenderer.BakingContext p_435748_) {
-            return new ShulkerBoxSpecialRenderer(new ShulkerBoxRenderer(p_435748_), this.openness, this.orientation, Sheets.SHULKER_MAPPER.apply(this.texture));
+        public SpecialModelRenderer<?> bake(EntityModelSet p_388396_) {
+            return new ShulkerBoxSpecialRenderer(new ShulkerBoxRenderer(p_388396_), this.openness, this.orientation, Sheets.SHULKER_MAPPER.apply(this.texture));
         }
     }
 }
