--- a/net/minecraft/client/renderer/entity/layers/ArrowLayer.java
+++ b/net/minecraft/client/renderer/entity/layers/ArrowLayer.java
@@ -6,18 +_,16 @@
 import net.minecraft.client.renderer.entity.EntityRendererProvider;
 import net.minecraft.client.renderer.entity.LivingEntityRenderer;
 import net.minecraft.client.renderer.entity.TippableArrowRenderer;
-import net.minecraft.client.renderer.entity.state.ArrowRenderState;
 import net.minecraft.client.renderer.entity.state.PlayerRenderState;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
 @OnlyIn(Dist.CLIENT)
-public class ArrowLayer<M extends PlayerModel> extends StuckInBodyLayer<M, ArrowRenderState> {
+public class ArrowLayer<M extends PlayerModel> extends StuckInBodyLayer<M> {
     public ArrowLayer(LivingEntityRenderer<?, PlayerRenderState, M> p_174466_, EntityRendererProvider.Context p_174465_) {
         super(
             p_174466_,
             new ArrowModel(p_174465_.bakeLayer(ModelLayers.ARROW)),
-            new ArrowRenderState(),
             TippableArrowRenderer.NORMAL_ARROW_LOCATION,
             StuckInBodyLayer.PlacementStyle.IN_CUBE
         );
