--- a/net/minecraft/client/renderer/entity/layers/BreezeEyesLayer.java
+++ b/net/minecraft/client/renderer/entity/layers/BreezeEyesLayer.java
@@ -1,11 +_,11 @@
 package net.minecraft.client.renderer.entity.layers;
 
 import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.blaze3d.vertex.VertexConsumer;
 import net.minecraft.client.model.BreezeModel;
-import net.minecraft.client.model.geom.EntityModelSet;
-import net.minecraft.client.model.geom.ModelLayers;
+import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.RenderType;
-import net.minecraft.client.renderer.SubmitNodeCollector;
+import net.minecraft.client.renderer.entity.BreezeRenderer;
 import net.minecraft.client.renderer.entity.RenderLayerParent;
 import net.minecraft.client.renderer.entity.state.BreezeRenderState;
 import net.minecraft.client.renderer.texture.OverlayTexture;
@@ -16,15 +_,15 @@
 @OnlyIn(Dist.CLIENT)
 public class BreezeEyesLayer extends RenderLayer<BreezeRenderState, BreezeModel> {
     private static final RenderType BREEZE_EYES = RenderType.breezeEyes(ResourceLocation.withDefaultNamespace("textures/entity/breeze/breeze_eyes.png"));
-    private final BreezeModel model;
 
-    public BreezeEyesLayer(RenderLayerParent<BreezeRenderState, BreezeModel> p_312409_, EntityModelSet p_432944_) {
+    public BreezeEyesLayer(RenderLayerParent<BreezeRenderState, BreezeModel> p_312409_) {
         super(p_312409_);
-        this.model = new BreezeModel(p_432944_.bakeLayer(ModelLayers.BREEZE_EYES));
     }
 
-    public void submit(PoseStack p_435727_, SubmitNodeCollector p_435319_, int p_433986_, BreezeRenderState p_434221_, float p_435361_, float p_435360_) {
-        p_435319_.order(1)
-            .submitModel(this.model, p_434221_, p_435727_, BREEZE_EYES, p_433986_, OverlayTexture.NO_OVERLAY, -1, null, p_434221_.outlineColor, null);
+    public void render(PoseStack p_312883_, MultiBufferSource p_312624_, int p_312874_, BreezeRenderState p_365352_, float p_312852_, float p_311980_) {
+        VertexConsumer vertexconsumer = p_312624_.getBuffer(BREEZE_EYES);
+        BreezeModel breezemodel = this.getParentModel();
+        BreezeRenderer.enable(breezemodel, breezemodel.head(), breezemodel.eyes())
+            .renderToBuffer(p_312883_, vertexconsumer, p_312874_, OverlayTexture.NO_OVERLAY);
     }
 }
