--- a/net/minecraft/client/renderer/entity/layers/CrossedArmsItemLayer.java
+++ b/net/minecraft/client/renderer/entity/layers/CrossedArmsItemLayer.java
@@ -4,7 +_,7 @@
 import com.mojang.math.Axis;
 import net.minecraft.client.model.EntityModel;
 import net.minecraft.client.model.VillagerLikeModel;
-import net.minecraft.client.renderer.SubmitNodeCollector;
+import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.entity.RenderLayerParent;
 import net.minecraft.client.renderer.entity.state.HoldingEntityRenderState;
 import net.minecraft.client.renderer.item.ItemStackRenderState;
@@ -18,18 +_,18 @@
         super(p_234818_);
     }
 
-    public void submit(PoseStack p_436000_, SubmitNodeCollector p_434361_, int p_432786_, S p_433547_, float p_435878_, float p_435123_) {
-        ItemStackRenderState itemstackrenderstate = p_433547_.heldItem;
+    public void render(PoseStack p_116699_, MultiBufferSource p_116700_, int p_116701_, S p_387384_, float p_116703_, float p_116704_) {
+        ItemStackRenderState itemstackrenderstate = p_387384_.heldItem;
         if (!itemstackrenderstate.isEmpty()) {
-            p_436000_.pushPose();
-            this.applyTranslation(p_433547_, p_436000_);
-            itemstackrenderstate.submit(p_436000_, p_434361_, p_432786_, OverlayTexture.NO_OVERLAY, p_433547_.outlineColor);
-            p_436000_.popPose();
+            p_116699_.pushPose();
+            this.applyTranslation(p_387384_, p_116699_);
+            itemstackrenderstate.render(p_116699_, p_116700_, p_116701_, OverlayTexture.NO_OVERLAY);
+            p_116699_.popPose();
         }
     }
 
     protected void applyTranslation(S p_386824_, PoseStack p_383057_) {
-        this.getParentModel().translateToArms(p_386824_, p_383057_);
+        this.getParentModel().translateToArms(p_383057_);
         p_383057_.mulPose(Axis.XP.rotation(0.75F));
         p_383057_.scale(1.07F, 1.07F, 1.07F);
         p_383057_.translate(0.0F, 0.13F, -0.34F);
