--- a/net/minecraft/client/renderer/entity/layers/IronGolemCrackinessLayer.java
+++ b/net/minecraft/client/renderer/entity/layers/IronGolemCrackinessLayer.java
@@ -4,7 +_,7 @@
 import com.mojang.blaze3d.vertex.PoseStack;
 import java.util.Map;
 import net.minecraft.client.model.IronGolemModel;
-import net.minecraft.client.renderer.SubmitNodeCollector;
+import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.entity.RenderLayerParent;
 import net.minecraft.client.renderer.entity.state.IronGolemRenderState;
 import net.minecraft.resources.ResourceLocation;
@@ -27,12 +_,12 @@
         super(p_117135_);
     }
 
-    public void submit(PoseStack p_433533_, SubmitNodeCollector p_434624_, int p_435788_, IronGolemRenderState p_434556_, float p_433354_, float p_433900_) {
-        if (!p_434556_.isInvisible) {
-            Crackiness.Level crackiness$level = p_434556_.crackiness;
+    public void render(PoseStack p_117137_, MultiBufferSource p_117138_, int p_117139_, IronGolemRenderState p_362922_, float p_117141_, float p_117142_) {
+        if (!p_362922_.isInvisible) {
+            Crackiness.Level crackiness$level = p_362922_.crackiness;
             if (crackiness$level != Crackiness.Level.NONE) {
                 ResourceLocation resourcelocation = resourceLocations.get(crackiness$level);
-                renderColoredCutoutModel(this.getParentModel(), resourcelocation, p_433533_, p_434624_, p_435788_, p_434556_, -1, 1);
+                renderColoredCutoutModel(this.getParentModel(), resourcelocation, p_117137_, p_117138_, p_117139_, p_362922_, -1);
             }
         }
     }
