--- a/net/minecraft/client/renderer/entity/layers/IronGolemFlowerLayer.java
+++ b/net/minecraft/client/renderer/entity/layers/IronGolemFlowerLayer.java
@@ -4,7 +_,7 @@
 import com.mojang.math.Axis;
 import net.minecraft.client.model.IronGolemModel;
 import net.minecraft.client.model.geom.ModelPart;
-import net.minecraft.client.renderer.SubmitNodeCollector;
+import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.block.BlockRenderDispatcher;
 import net.minecraft.client.renderer.entity.RenderLayerParent;
 import net.minecraft.client.renderer.entity.state.IronGolemRenderState;
@@ -22,19 +_,19 @@
         this.blockRenderer = p_234843_;
     }
 
-    public void submit(PoseStack p_432993_, SubmitNodeCollector p_435087_, int p_433962_, IronGolemRenderState p_434784_, float p_435316_, float p_435013_) {
-        if (p_434784_.offerFlowerTick != 0) {
-            p_432993_.pushPose();
+    public void render(PoseStack p_117172_, MultiBufferSource p_117173_, int p_117174_, IronGolemRenderState p_364431_, float p_117176_, float p_117177_) {
+        if (p_364431_.offerFlowerTick != 0) {
+            p_117172_.pushPose();
             ModelPart modelpart = this.getParentModel().getFlowerHoldingArm();
-            modelpart.translateAndRotate(p_432993_);
-            p_432993_.translate(-1.1875F, 1.0625F, -0.9375F);
-            p_432993_.translate(0.5F, 0.5F, 0.5F);
+            modelpart.translateAndRotate(p_117172_);
+            p_117172_.translate(-1.1875F, 1.0625F, -0.9375F);
+            p_117172_.translate(0.5F, 0.5F, 0.5F);
             float f = 0.5F;
-            p_432993_.scale(0.5F, 0.5F, 0.5F);
-            p_432993_.mulPose(Axis.XP.rotationDegrees(-90.0F));
-            p_432993_.translate(-0.5F, -0.5F, -0.5F);
-            p_435087_.submitBlock(p_432993_, Blocks.POPPY.defaultBlockState(), p_433962_, OverlayTexture.NO_OVERLAY, p_434784_.outlineColor);
-            p_432993_.popPose();
+            p_117172_.scale(0.5F, 0.5F, 0.5F);
+            p_117172_.mulPose(Axis.XP.rotationDegrees(-90.0F));
+            p_117172_.translate(-0.5F, -0.5F, -0.5F);
+            this.blockRenderer.renderSingleBlock(Blocks.POPPY.defaultBlockState(), p_117172_, p_117173_, p_117174_, OverlayTexture.NO_OVERLAY);
+            p_117172_.popPose();
         }
     }
 }
