--- a/net/minecraft/client/renderer/entity/layers/SimpleEquipmentLayer.java
+++ b/net/minecraft/client/renderer/entity/layers/SimpleEquipmentLayer.java
@@ -3,7 +_,7 @@
 import com.mojang.blaze3d.vertex.PoseStack;
 import java.util.function.Function;
 import net.minecraft.client.model.EntityModel;
-import net.minecraft.client.renderer.SubmitNodeCollector;
+import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.entity.RenderLayerParent;
 import net.minecraft.client.renderer.entity.state.LivingEntityRenderState;
 import net.minecraft.client.resources.model.EquipmentClientInfo;
@@ -21,25 +_,6 @@
     private final Function<S, ItemStack> itemGetter;
     private final EM adultModel;
     private final EM babyModel;
-    private final int order;
-
-    public SimpleEquipmentLayer(
-        RenderLayerParent<S, RM> p_397277_,
-        EquipmentLayerRenderer p_398036_,
-        EquipmentClientInfo.LayerType p_397312_,
-        Function<S, ItemStack> p_397526_,
-        EM p_397856_,
-        EM p_436778_,
-        int p_436781_
-    ) {
-        super(p_397277_);
-        this.equipmentRenderer = p_398036_;
-        this.layer = p_397312_;
-        this.itemGetter = p_397526_;
-        this.adultModel = p_397856_;
-        this.babyModel = p_436778_;
-        this.order = p_436781_;
-    }
 
     public SimpleEquipmentLayer(
         RenderLayerParent<S, RM> p_397685_,
@@ -49,18 +_,31 @@
         EM p_397462_,
         EM p_397451_
     ) {
-        this(p_397685_, p_397797_, p_397371_, p_397740_, p_397462_, p_397451_, 0);
-    }
-
-    public void submit(PoseStack p_434184_, SubmitNodeCollector p_433972_, int p_434492_, S p_434386_, float p_434090_, float p_432997_) {
-        ItemStack itemstack = this.itemGetter.apply(p_434386_);
+        super(p_397685_);
+        this.equipmentRenderer = p_397797_;
+        this.layer = p_397371_;
+        this.itemGetter = p_397740_;
+        this.adultModel = p_397462_;
+        this.babyModel = p_397451_;
+    }
+
+    public SimpleEquipmentLayer(
+        RenderLayerParent<S, RM> p_397277_,
+        EquipmentLayerRenderer p_398036_,
+        EM p_397856_,
+        EquipmentClientInfo.LayerType p_397312_,
+        Function<S, ItemStack> p_397526_
+    ) {
+        this(p_397277_, p_398036_, p_397312_, p_397526_, p_397856_, p_397856_);
+    }
+
+    public void render(PoseStack p_397396_, MultiBufferSource p_397882_, int p_397812_, S p_397405_, float p_397045_, float p_397886_) {
+        ItemStack itemstack = this.itemGetter.apply(p_397405_);
         Equippable equippable = itemstack.get(DataComponents.EQUIPPABLE);
         if (equippable != null && !equippable.assetId().isEmpty()) {
-            EM em = p_434386_.isBaby ? this.babyModel : this.adultModel;
-            this.equipmentRenderer
-                .renderLayers(
-                    this.layer, equippable.assetId().get(), em, p_434386_, itemstack, p_434184_, p_433972_, p_434492_, null, p_434386_.outlineColor, this.order
-                );
+            EM em = p_397405_.isBaby ? this.babyModel : this.adultModel;
+            em.setupAnim(p_397405_);
+            this.equipmentRenderer.renderLayers(this.layer, equippable.assetId().get(), em, itemstack, p_397396_, p_397882_, p_397812_);
         }
     }
 }
