--- a/net/minecraft/client/gui/screens/reporting/ReportReasonSelectionScreen.java
+++ b/net/minecraft/client/gui/screens/reporting/ReportReasonSelectionScreen.java
@@ -63,7 +_,7 @@
         linearlayout1.addChild(
             Button.builder(
                     CommonComponents.GUI_DONE,
-                    p_432234_ -> {
+                    p_386219_ -> {
                         ReportReasonSelectionScreen.ReasonSelectionList.Entry reportreasonselectionscreen$reasonselectionlist$entry1 = this.reasonSelectionList
                             .getSelected();
                         if (reportreasonselectionscreen$reasonselectionlist$entry1 != null) {
@@ -93,7 +_,7 @@
     public void render(GuiGraphics p_282815_, int p_283039_, int p_283620_, float p_281336_) {
         super.render(p_282815_, p_283039_, p_283620_, p_281336_);
         p_282815_.fill(this.descriptionLeft(), this.descriptionTop(), this.descriptionRight(), this.descriptionBottom(), -16777216);
-        p_282815_.submitOutline(this.descriptionLeft(), this.descriptionTop(), this.descriptionWidth(), this.descriptionHeight(), -1);
+        p_282815_.renderOutline(this.descriptionLeft(), this.descriptionTop(), this.descriptionWidth(), this.descriptionHeight(), -1);
         p_282815_.drawString(this.font, REASON_DESCRIPTION, this.descriptionLeft() + 4, this.descriptionTop() + 4, -1);
         ReportReasonSelectionScreen.ReasonSelectionList.Entry reportreasonselectionscreen$reasonselectionlist$entry = this.reasonSelectionList.getSelected();
         if (reportreasonselectionscreen$reasonselectionlist$entry != null) {
@@ -183,10 +_,21 @@
             }
 
             @Override
-            public void renderContent(GuiGraphics p_439346_, int p_440497_, int p_439746_, boolean p_439154_, float p_439726_) {
-                int i = this.getContentX() + 1;
-                int j = this.getContentY() + (this.getContentHeight() - 9) / 2 + 1;
-                p_439346_.drawString(ReportReasonSelectionScreen.this.font, this.reason.title(), i, j, -1);
+            public void render(
+                GuiGraphics p_281941_,
+                int p_281450_,
+                int p_281781_,
+                int p_283334_,
+                int p_283073_,
+                int p_282523_,
+                int p_282667_,
+                int p_281567_,
+                boolean p_282095_,
+                float p_283305_
+            ) {
+                int i = p_283334_ + 1;
+                int j = p_281781_ + (p_282523_ - 9) / 2 + 1;
+                p_281941_.drawString(ReportReasonSelectionScreen.this.font, this.reason.title(), i, j, -1);
             }
 
             @Override
@@ -195,9 +_,9 @@
             }
 
             @Override
-            public boolean mouseClicked(double p_240021_, double p_240022_, int p_240023_, boolean p_435027_) {
+            public boolean mouseClicked(double p_240021_, double p_240022_, int p_240023_) {
                 ReasonSelectionList.this.setSelected(this);
-                return super.mouseClicked(p_240021_, p_240022_, p_240023_, p_435027_);
+                return super.mouseClicked(p_240021_, p_240022_, p_240023_);
             }
 
             public ReportReason getReason() {
