--- a/net/minecraft/client/renderer/Sheets.java
+++ b/net/minecraft/client/renderer/Sheets.java
@@ -16,7 +_,6 @@
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.world.item.DyeColor;
-import net.minecraft.world.level.block.CopperChestBlock;
 import net.minecraft.world.level.block.entity.BannerPattern;
 import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.block.entity.DecoratedPotPattern;
@@ -37,9 +_,6 @@
     public static final ResourceLocation CHEST_SHEET = ResourceLocation.withDefaultNamespace("textures/atlas/chest.png");
     public static final ResourceLocation ARMOR_TRIMS_SHEET = ResourceLocation.withDefaultNamespace("textures/atlas/armor_trims.png");
     public static final ResourceLocation DECORATED_POT_SHEET = ResourceLocation.withDefaultNamespace("textures/atlas/decorated_pot.png");
-    public static final ResourceLocation GUI_SHEET = ResourceLocation.withDefaultNamespace("textures/atlas/gui.png");
-    public static final ResourceLocation MAP_DECORATIONS_SHEET = ResourceLocation.withDefaultNamespace("textures/atlas/map_decorations.png");
-    public static final ResourceLocation PAINTINGS_SHEET = ResourceLocation.withDefaultNamespace("textures/atlas/paintings.png");
     private static final RenderType SHULKER_BOX_SHEET_TYPE = RenderType.entityCutoutNoCull(SHULKER_SHEET);
     private static final RenderType BED_SHEET_TYPE = RenderType.entitySolid(BED_SHEET);
     private static final RenderType BANNER_SHEET_TYPE = RenderType.entityNoOutline(BANNER_SHEET);
@@ -53,7 +_,6 @@
     private static final RenderType TRANSLUCENT_ITEM_CULL_BLOCK_SHEET = RenderType.itemEntityTranslucentCull(TextureAtlas.LOCATION_BLOCKS);
     public static final MaterialMapper ITEMS_MAPPER = new MaterialMapper(TextureAtlas.LOCATION_BLOCKS, "item");
     public static final MaterialMapper BLOCKS_MAPPER = new MaterialMapper(TextureAtlas.LOCATION_BLOCKS, "block");
-    public static final MaterialMapper BLOCK_ENTITIES_MAPPER = new MaterialMapper(TextureAtlas.LOCATION_BLOCKS, "entity");
     public static final MaterialMapper BANNER_MAPPER = new MaterialMapper(BANNER_SHEET, "entity/banner");
     public static final MaterialMapper SHIELD_MAPPER = new MaterialMapper(SHIELD_SHEET, "entity/shield");
     public static final MaterialMapper CHEST_MAPPER = new MaterialMapper(CHEST_SHEET, "entity/chest");
@@ -93,18 +_,6 @@
     public static final Material CHEST_LOCATION_LEFT = CHEST_MAPPER.defaultNamespaceApply("normal_left");
     public static final Material CHEST_LOCATION_RIGHT = CHEST_MAPPER.defaultNamespaceApply("normal_right");
     public static final Material ENDER_CHEST_LOCATION = CHEST_MAPPER.defaultNamespaceApply("ender");
-    public static final Material COPPER_CHEST_LOCATION = CHEST_MAPPER.defaultNamespaceApply("copper");
-    public static final Material COPPER_CHEST_LOCATION_LEFT = CHEST_MAPPER.defaultNamespaceApply("copper_left");
-    public static final Material COPPER_CHEST_LOCATION_RIGHT = CHEST_MAPPER.defaultNamespaceApply("copper_right");
-    public static final Material EXPOSED_COPPER_CHEST_LOCATION = CHEST_MAPPER.defaultNamespaceApply("copper_exposed");
-    public static final Material EXPOSED_COPPER_CHEST_LOCATION_LEFT = CHEST_MAPPER.defaultNamespaceApply("copper_exposed_left");
-    public static final Material EXPOSED_COPPER_CHEST_LOCATION_RIGHT = CHEST_MAPPER.defaultNamespaceApply("copper_exposed_right");
-    public static final Material WEATHERED_COPPER_CHEST_LOCATION = CHEST_MAPPER.defaultNamespaceApply("copper_weathered");
-    public static final Material WEATHERED_COPPER_CHEST_LOCATION_LEFT = CHEST_MAPPER.defaultNamespaceApply("copper_weathered_left");
-    public static final Material WEATHERED_COPPER_CHEST_LOCATION_RIGHT = CHEST_MAPPER.defaultNamespaceApply("copper_weathered_right");
-    public static final Material OXIDIZED_COPPER_CHEST_LOCATION = CHEST_MAPPER.defaultNamespaceApply("copper_oxidized");
-    public static final Material OXIDIZED_COPPER_CHEST_LOCATION_LEFT = CHEST_MAPPER.defaultNamespaceApply("copper_oxidized_left");
-    public static final Material OXIDIZED_COPPER_CHEST_LOCATION_RIGHT = CHEST_MAPPER.defaultNamespaceApply("copper_oxidized_right");
 
     public static RenderType bannerSheet() {
         return BANNER_SHEET_TYPE;
@@ -175,11 +_,11 @@
     }
 
     private static Material createSignMaterial(WoodType p_173386_) {
-        return SIGN_MAPPER.defaultNamespaceApply(p_173386_.name());
+        return SIGN_MAPPER.apply(ResourceLocation.parse(p_173386_.name()));
     }
 
     private static Material createHangingSignMaterial(WoodType p_251735_) {
-        return HANGING_SIGN_MAPPER.defaultNamespaceApply(p_251735_.name());
+        return HANGING_SIGN_MAPPER.apply(ResourceLocation.parse(p_251735_.name()));
     }
 
     public static Material getSignMaterial(WoodType p_173382_) {
@@ -208,28 +_,13 @@
             return ENDER_CHEST_LOCATION;
         } else if (p_110770_) {
             return chooseMaterial(p_110769_, CHEST_XMAS_LOCATION, CHEST_XMAS_LOCATION_LEFT, CHEST_XMAS_LOCATION_RIGHT);
-        } else if (p_110768_ instanceof TrappedChestBlockEntity) {
-            return chooseMaterial(p_110769_, CHEST_TRAP_LOCATION, CHEST_TRAP_LOCATION_LEFT, CHEST_TRAP_LOCATION_RIGHT);
         } else {
-            return p_110768_.getBlockState().getBlock() instanceof CopperChestBlock copperchestblock
-                ? chooseCopperMaterial(p_110769_, copperchestblock)
+            return p_110768_ instanceof TrappedChestBlockEntity
+                ? chooseMaterial(p_110769_, CHEST_TRAP_LOCATION, CHEST_TRAP_LOCATION_LEFT, CHEST_TRAP_LOCATION_RIGHT)
                 : chooseMaterial(p_110769_, CHEST_LOCATION, CHEST_LOCATION_LEFT, CHEST_LOCATION_RIGHT);
         }
     }
 
-    private static Material chooseCopperMaterial(ChestType p_434456_, CopperChestBlock p_435551_) {
-        return switch (p_435551_.getState()) {
-            case UNAFFECTED -> chooseMaterial(p_434456_, COPPER_CHEST_LOCATION, COPPER_CHEST_LOCATION_LEFT, COPPER_CHEST_LOCATION_RIGHT);
-            case EXPOSED -> chooseMaterial(p_434456_, EXPOSED_COPPER_CHEST_LOCATION, EXPOSED_COPPER_CHEST_LOCATION_LEFT, EXPOSED_COPPER_CHEST_LOCATION_RIGHT);
-            case WEATHERED -> chooseMaterial(
-                p_434456_, WEATHERED_COPPER_CHEST_LOCATION, WEATHERED_COPPER_CHEST_LOCATION_LEFT, WEATHERED_COPPER_CHEST_LOCATION_RIGHT
-            );
-            case OXIDIZED -> chooseMaterial(
-                p_434456_, OXIDIZED_COPPER_CHEST_LOCATION, OXIDIZED_COPPER_CHEST_LOCATION_LEFT, OXIDIZED_COPPER_CHEST_LOCATION_RIGHT
-            );
-        };
-    }
-
     private static Material chooseMaterial(ChestType p_110772_, Material p_110773_, Material p_110774_, Material p_110775_) {
         switch (p_110772_) {
             case LEFT:
@@ -239,6 +_,23 @@
             case SINGLE:
             default:
                 return p_110773_;
+        }
+    }
+
+    /**
+     * Not threadsafe. Enqueue it in client setup.
+     */
+    public static void addWoodType(WoodType woodType) {
+        SIGN_MATERIALS.put(woodType, createSignMaterial(woodType));
+        HANGING_SIGN_MATERIALS.put(woodType, createHangingSignMaterial(woodType));
+    }
+
+    static {
+        if (!net.neoforged.fml.ModLoader.hasErrors() && !net.neoforged.neoforge.internal.CommonModLoader.areRegistriesLoaded()) {
+            com.mojang.logging.LogUtils.getLogger().error(
+                      "net.minecraft.client.renderer.Sheets loaded too early, modded registry-based materials may not work correctly",
+                      new IllegalStateException("net.minecraft.client.renderer.Sheets loaded too early")
+            );
         }
     }
 }
