--- a/net/minecraft/client/renderer/SubmitNodeCollector.java
+++ /dev/null
@@ -1,16 +_,0 @@
-package net.minecraft.client.renderer;
-
-import com.mojang.blaze3d.vertex.PoseStack;
-import com.mojang.blaze3d.vertex.VertexConsumer;
-import net.neoforged.api.distmarker.Dist;
-import net.neoforged.api.distmarker.OnlyIn;
-
-@OnlyIn(Dist.CLIENT)
-public interface SubmitNodeCollector extends OrderedSubmitNodeCollector {
-    OrderedSubmitNodeCollector order(int p_438904_);
-
-    @OnlyIn(Dist.CLIENT)
-    public interface CustomGeometryRenderer {
-        void render(PoseStack.Pose p_434723_, VertexConsumer p_434248_);
-    }
-}
