--- a/net/minecraft/tags/BlockTags.java
+++ b/net/minecraft/tags/BlockTags.java
@@ -18,7 +_,6 @@
     public static final TagKey<Block> WOODEN_FENCES = create("wooden_fences");
     public static final TagKey<Block> FENCE_GATES = create("fence_gates");
     public static final TagKey<Block> WOODEN_PRESSURE_PLATES = create("wooden_pressure_plates");
-    public static final TagKey<Block> WOODEN_SHELVES = create("wooden_shelves");
     public static final TagKey<Block> DOORS = create("doors");
     public static final TagKey<Block> SAPLINGS = create("saplings");
     public static final TagKey<Block> BAMBOO_BLOCKS = create("bamboo_blocks");
@@ -65,13 +_,6 @@
     public static final TagKey<Block> TERRACOTTA = create("terracotta");
     public static final TagKey<Block> COMPLETES_FIND_TREE_TUTORIAL = create("completes_find_tree_tutorial");
     public static final TagKey<Block> SHULKER_BOXES = create("shulker_boxes");
-    public static final TagKey<Block> COPPER_CHESTS = create("copper_chests");
-    public static final TagKey<Block> LIGHTNING_RODS = create("lightning_rods");
-    public static final TagKey<Block> COPPER = create("copper");
-    public static final TagKey<Block> CHAINS = create("chains");
-    public static final TagKey<Block> COPPER_GOLEM_STATUES = create("copper_golem_statues");
-    public static final TagKey<Block> LANTERNS = create("lanterns");
-    public static final TagKey<Block> BARS = create("bars");
     public static final TagKey<Block> CEILING_HANGING_SIGNS = create("ceiling_hanging_signs");
     public static final TagKey<Block> STANDING_SIGNS = create("standing_signs");
     public static final TagKey<Block> BEE_ATTRACTIVE = create("bee_attractive");
@@ -159,7 +_,6 @@
     public static final TagKey<Block> INCORRECT_FOR_NETHERITE_TOOL = create("incorrect_for_netherite_tool");
     public static final TagKey<Block> INCORRECT_FOR_DIAMOND_TOOL = create("incorrect_for_diamond_tool");
     public static final TagKey<Block> INCORRECT_FOR_IRON_TOOL = create("incorrect_for_iron_tool");
-    public static final TagKey<Block> INCORRECT_FOR_COPPER_TOOL = create("incorrect_for_copper_tool");
     public static final TagKey<Block> INCORRECT_FOR_STONE_TOOL = create("incorrect_for_stone_tool");
     public static final TagKey<Block> INCORRECT_FOR_GOLD_TOOL = create("incorrect_for_gold_tool");
     public static final TagKey<Block> INCORRECT_FOR_WOODEN_TOOL = create("incorrect_for_wooden_tool");
@@ -214,5 +_,9 @@
 
     private static TagKey<Block> create(String p_203847_) {
         return TagKey.create(Registries.BLOCK, ResourceLocation.withDefaultNamespace(p_203847_));
+    }
+
+    public static TagKey<Block> create(ResourceLocation name) {
+        return TagKey.create(Registries.BLOCK, name);
     }
 }
