--- a/net/minecraft/network/protocol/game/ClientboundPlayerRotationPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundPlayerRotationPacket.java
@@ -6,16 +_,12 @@
 import net.minecraft.network.protocol.Packet;
 import net.minecraft.network.protocol.PacketType;
 
-public record ClientboundPlayerRotationPacket(float yRot, boolean relativeY, float xRot, boolean relativeX) implements Packet<ClientGamePacketListener> {
+public record ClientboundPlayerRotationPacket(float yRot, float xRot) implements Packet<ClientGamePacketListener> {
     public static final StreamCodec<FriendlyByteBuf, ClientboundPlayerRotationPacket> STREAM_CODEC = StreamCodec.composite(
         ByteBufCodecs.FLOAT,
         ClientboundPlayerRotationPacket::yRot,
-        ByteBufCodecs.BOOL,
-        ClientboundPlayerRotationPacket::relativeY,
         ByteBufCodecs.FLOAT,
         ClientboundPlayerRotationPacket::xRot,
-        ByteBufCodecs.BOOL,
-        ClientboundPlayerRotationPacket::relativeX,
         ClientboundPlayerRotationPacket::new
     );
 
