--- a/net/minecraft/client/model/geom/ModelPart.java
+++ b/net/minecraft/client/model/geom/ModelPart.java
@@ -71,6 +_,18 @@
         this.zScale = p_171323_.zScale();
     }
 
+    public void copyFrom(ModelPart p_104316_) {
+        this.xScale = p_104316_.xScale;
+        this.yScale = p_104316_.yScale;
+        this.zScale = p_104316_.zScale;
+        this.xRot = p_104316_.xRot;
+        this.yRot = p_104316_.yRot;
+        this.zRot = p_104316_.zRot;
+        this.x = p_104316_.x;
+        this.y = p_104316_.y;
+        this.z = p_104316_.z;
+    }
+
     public boolean hasChild(String p_233563_) {
         return this.children.containsKey(p_233563_);
     }
