--- a/net/minecraft/world/waypoints/TrackedWaypoint.java
+++ b/net/minecraft/world/waypoints/TrackedWaypoint.java
@@ -19,7 +_,7 @@
 
 public abstract class TrackedWaypoint implements Waypoint {
     static final Logger LOGGER = LogUtils.getLogger();
-    public static final StreamCodec<ByteBuf, TrackedWaypoint> STREAM_CODEC = StreamCodec.ofMember(TrackedWaypoint::write, TrackedWaypoint::read);
+    public static StreamCodec<ByteBuf, TrackedWaypoint> STREAM_CODEC = StreamCodec.ofMember(TrackedWaypoint::write, TrackedWaypoint::read);
     protected final Either<UUID, String> identifier;
     private final Waypoint.Icon icon;
     private final TrackedWaypoint.Type type;
@@ -70,9 +_,9 @@
         return new TrackedWaypoint.EmptyWaypoint(p_416251_);
     }
 
-    public abstract double yawAngleToCamera(Level p_419922_, TrackedWaypoint.Camera p_419734_, PartialTickSupplier p_437296_);
+    public abstract double yawAngleToCamera(Level p_419922_, TrackedWaypoint.Camera p_419734_);
 
-    public abstract TrackedWaypoint.PitchDirection pitchDirectionToCamera(Level p_420064_, TrackedWaypoint.Projector p_419795_, PartialTickSupplier p_437349_);
+    public abstract TrackedWaypoint.PitchDirection pitchDirectionToCamera(Level p_420064_, TrackedWaypoint.Projector p_419795_);
 
     public abstract double distanceSquared(Entity p_416521_);
 
@@ -108,12 +_,12 @@
         }
 
         @Override
-        public double yawAngleToCamera(Level p_419593_, TrackedWaypoint.Camera p_419977_, PartialTickSupplier p_437329_) {
+        public double yawAngleToCamera(Level p_419593_, TrackedWaypoint.Camera p_419977_) {
             return Mth.degreesDifference(p_419977_.yaw(), this.angle * (180.0F / (float)Math.PI));
         }
 
         @Override
-        public TrackedWaypoint.PitchDirection pitchDirectionToCamera(Level p_419920_, TrackedWaypoint.Projector p_416265_, PartialTickSupplier p_437258_) {
+        public TrackedWaypoint.PitchDirection pitchDirectionToCamera(Level p_419920_, TrackedWaypoint.Projector p_416265_) {
             double d0 = p_416265_.projectHorizonToScreen();
             if (d0 < -1.0) {
                 return TrackedWaypoint.PitchDirection.DOWN;
@@ -167,7 +_,7 @@
         }
 
         @Override
-        public double yawAngleToCamera(Level p_420006_, TrackedWaypoint.Camera p_419530_, PartialTickSupplier p_437285_) {
+        public double yawAngleToCamera(Level p_420006_, TrackedWaypoint.Camera p_419530_) {
             Vec3 vec3 = p_419530_.position();
             Vec3 vec31 = vec3.subtract(this.position(vec3.y())).rotateClockwise90();
             float f = (float)Mth.atan2(vec31.z(), vec31.x()) * (180.0F / (float)Math.PI);
@@ -175,7 +_,7 @@
         }
 
         @Override
-        public TrackedWaypoint.PitchDirection pitchDirectionToCamera(Level p_419513_, TrackedWaypoint.Projector p_416711_, PartialTickSupplier p_437246_) {
+        public TrackedWaypoint.PitchDirection pitchDirectionToCamera(Level p_419513_, TrackedWaypoint.Projector p_416711_) {
             double d0 = p_416711_.projectHorizonToScreen();
             if (d0 < -1.0) {
                 return TrackedWaypoint.PitchDirection.DOWN;
@@ -208,12 +_,12 @@
         }
 
         @Override
-        public double yawAngleToCamera(Level p_419897_, TrackedWaypoint.Camera p_419457_, PartialTickSupplier p_437411_) {
+        public double yawAngleToCamera(Level p_419897_, TrackedWaypoint.Camera p_419457_) {
             return Double.NaN;
         }
 
         @Override
-        public TrackedWaypoint.PitchDirection pitchDirectionToCamera(Level p_419455_, TrackedWaypoint.Projector p_416651_, PartialTickSupplier p_437225_) {
+        public TrackedWaypoint.PitchDirection pitchDirectionToCamera(Level p_419455_, TrackedWaypoint.Projector p_416651_) {
             return TrackedWaypoint.PitchDirection.NONE;
         }
 
@@ -277,24 +_,24 @@
             VarInt.write(p_416442_, this.vector.getZ());
         }
 
-        private Vec3 position(Level p_416641_, PartialTickSupplier p_437189_) {
+        private Vec3 position(Level p_416641_) {
             return this.identifier
                 .left()
                 .map(p_416641_::getEntity)
-                .map(p_437185_ -> p_437185_.blockPosition().distManhattan(this.vector) > 3 ? null : p_437185_.getEyePosition(p_437189_.apply(p_437185_)))
+                .map(p_415868_ -> p_415868_.blockPosition().distManhattan(this.vector) > 3 ? null : p_415868_.getEyePosition())
                 .orElseGet(() -> Vec3.atCenterOf(this.vector));
         }
 
         @Override
-        public double yawAngleToCamera(Level p_419799_, TrackedWaypoint.Camera p_419893_, PartialTickSupplier p_437238_) {
-            Vec3 vec3 = p_419893_.position().subtract(this.position(p_419799_, p_437238_)).rotateClockwise90();
+        public double yawAngleToCamera(Level p_419799_, TrackedWaypoint.Camera p_419893_) {
+            Vec3 vec3 = p_419893_.position().subtract(this.position(p_419799_)).rotateClockwise90();
             float f = (float)Mth.atan2(vec3.z(), vec3.x()) * (180.0F / (float)Math.PI);
             return Mth.degreesDifference(p_419893_.yaw(), f);
         }
 
         @Override
-        public TrackedWaypoint.PitchDirection pitchDirectionToCamera(Level p_419459_, TrackedWaypoint.Projector p_419783_, PartialTickSupplier p_437301_) {
-            Vec3 vec3 = p_419783_.projectPointToScreen(this.position(p_419459_, p_437301_));
+        public TrackedWaypoint.PitchDirection pitchDirectionToCamera(Level p_419459_, TrackedWaypoint.Projector p_419783_) {
+            Vec3 vec3 = p_419783_.projectPointToScreen(this.position(p_419459_));
             boolean flag = vec3.z > 1.0;
             double d0 = flag ? -vec3.y : vec3.y;
             if (d0 < -1.0) {
