--- a/net/minecraft/world/level/levelgen/DensityFunction.java
+++ b/net/minecraft/world/level/levelgen/DensityFunction.java
@@ -55,10 +_,6 @@
         return DensityFunctions.map(this, DensityFunctions.Mapped.Type.QUARTER_NEGATIVE);
     }
 
-    default DensityFunction invert() {
-        return DensityFunctions.map(this, DensityFunctions.Mapped.Type.INVERT);
-    }
-
     default DensityFunction squeeze() {
         return DensityFunctions.map(this, DensityFunctions.Mapped.Type.SQUEEZE);
     }
