--- a/net/minecraft/client/data/models/model/TextureMapping.java
+++ b/net/minecraft/client/data/models/model/TextureMapping.java
@@ -385,10 +_,6 @@
             .put(TextureSlot.BOTTOM, getBlockTexture(Blocks.SCULK_SHRIEKER, "_bottom"));
     }
 
-    public static TextureMapping bars(Block p_436719_) {
-        return new TextureMapping().put(TextureSlot.BARS, getBlockTexture(p_436719_)).put(TextureSlot.EDGE, getBlockTexture(p_436719_));
-    }
-
     public static TextureMapping layer0(Item p_386958_) {
         return new TextureMapping().put(TextureSlot.LAYER0, getItemTexture(p_386958_));
     }
@@ -427,5 +_,13 @@
     public static ResourceLocation getItemTexture(Item p_386842_, String p_386898_) {
         ResourceLocation resourcelocation = BuiltInRegistries.ITEM.getKey(p_386842_);
         return resourcelocation.withPath(p_387396_ -> "item/" + p_387396_ + p_386898_);
+    }
+
+    // Neo: Added to allow easier texture map copying
+    public TextureMapping copy() {
+        TextureMapping texturemapping = new TextureMapping();
+        texturemapping.slots.putAll(this.slots);
+        texturemapping.forcedSlots.addAll(this.forcedSlots);
+        return texturemapping;
     }
 }
