--- a/net/minecraft/server/players/ServerOpList.java
+++ b/net/minecraft/server/players/ServerOpList.java
@@ -1,30 +_,31 @@
 package net.minecraft.server.players;
 
 import com.google.gson.JsonObject;
+import com.mojang.authlib.GameProfile;
 import java.io.File;
 import java.util.Objects;
 
-public class ServerOpList extends StoredUserList<NameAndId, ServerOpListEntry> {
+public class ServerOpList extends StoredUserList<GameProfile, ServerOpListEntry> {
     public ServerOpList(File p_11345_) {
         super(p_11345_);
     }
 
     @Override
-    protected StoredUserEntry<NameAndId> createEntry(JsonObject p_11348_) {
+    protected StoredUserEntry<GameProfile> createEntry(JsonObject p_11348_) {
         return new ServerOpListEntry(p_11348_);
     }
 
     @Override
     public String[] getUserList() {
-        return this.getEntries().stream().map(StoredUserEntry::getUser).filter(Objects::nonNull).map(NameAndId::name).toArray(String[]::new);
+        return this.getEntries().stream().map(StoredUserEntry::getUser).filter(Objects::nonNull).map(GameProfile::getName).toArray(String[]::new);
     }
 
-    public boolean canBypassPlayerLimit(NameAndId p_435874_) {
-        ServerOpListEntry serveroplistentry = this.get(p_435874_);
+    public boolean canBypassPlayerLimit(GameProfile p_11352_) {
+        ServerOpListEntry serveroplistentry = this.get(p_11352_);
         return serveroplistentry != null ? serveroplistentry.getBypassesPlayerLimit() : false;
     }
 
-    protected String getKeyForUser(NameAndId p_433823_) {
-        return p_433823_.id().toString();
+    protected String getKeyForUser(GameProfile p_11354_) {
+        return p_11354_.getId().toString();
     }
 }
