--- a/net/minecraft/client/gui/screens/inventory/TestInstanceBlockEditScreen.java
+++ b/net/minecraft/client/gui/screens/inventory/TestInstanceBlockEditScreen.java
@@ -33,6 +_,7 @@
     private static final Component ROTATION_LABEL = Component.translatable("test_instance_block.rotation");
     private static final int BUTTON_PADDING = 8;
     private static final int WIDTH = 316;
+    private static final int COLOR_SILVER = -4144960;
     private final TestInstanceBlockEntity blockEntity;
     @Nullable
     private EditBox idEdit;
@@ -170,10 +_,10 @@
         super.render(p_397620_, p_397635_, p_397823_, p_397214_);
         int i = this.width / 2 - 158;
         p_397620_.drawCenteredString(this.font, this.title, this.width / 2, 10, -1);
-        p_397620_.drawString(this.font, ID_LABEL, i, 30, -6250336);
-        p_397620_.drawString(this.font, SIZE_LABEL, i, 150, -6250336);
-        p_397620_.drawString(this.font, ROTATION_LABEL, this.rotationButton.getX(), 150, -6250336);
-        p_397620_.drawString(this.font, INCLUDE_ENTITIES_LABEL, this.includeEntitiesButton.getX(), 150, -6250336);
+        p_397620_.drawString(this.font, ID_LABEL, i, 30, -4144960);
+        p_397620_.drawString(this.font, SIZE_LABEL, i, 150, -4144960);
+        p_397620_.drawString(this.font, ROTATION_LABEL, this.rotationButton.getX(), 150, -4144960);
+        p_397620_.drawString(this.font, INCLUDE_ENTITIES_LABEL, this.includeEntitiesButton.getX(), 150, -4144960);
     }
 
     private void updateTestInfo(boolean p_397030_) {
@@ -233,7 +_,7 @@
     }
 
     @Override
-    public boolean isInGameUi() {
-        return true;
+    public void renderBackground(GuiGraphics p_399858_, int p_400056_, int p_399919_, float p_400070_) {
+        this.renderTransparentBackground(p_399858_);
     }
 }
