--- a/net/minecraft/world/entity/animal/allay/Allay.java
+++ b/net/minecraft/world/entity/animal/allay/Allay.java
@@ -232,7 +_,7 @@
     @Override
     public void aiStep() {
         super.aiStep();
-        if (!this.level().isClientSide() && this.isAlive() && this.tickCount % 10 == 0) {
+        if (!this.level().isClientSide && this.isAlive() && this.tickCount % 10 == 0) {
             this.heal(1.0F);
         }
 
@@ -247,7 +_,7 @@
     @Override
     public void tick() {
         super.tick();
-        if (this.level().isClientSide()) {
+        if (this.level().isClientSide) {
             this.holdingItemAnimationTicks0 = this.holdingItemAnimationTicks;
             if (this.hasItemInHand()) {
                 this.holdingItemAnimationTicks = Mth.clamp(this.holdingItemAnimationTicks + 1.0F, 0.0F, 5.0F);
@@ -356,9 +_,9 @@
     public boolean wantsToPickUp(ServerLevel p_376111_, ItemStack p_218387_) {
         ItemStack itemstack = this.getItemInHand(InteractionHand.MAIN_HAND);
         return !itemstack.isEmpty()
-            && p_376111_.getGameRules().getBoolean(GameRules.RULE_MOBGRIEFING)
             && this.inventory.canAddItem(p_218387_)
-            && this.allayConsidersItemEqual(itemstack, p_218387_);
+            && this.allayConsidersItemEqual(itemstack, p_218387_)
+            && net.neoforged.neoforge.event.EventHooks.canEntityGrief(p_376111_, this);
     }
 
     private boolean allayConsidersItemEqual(ItemStack p_252278_, ItemStack p_250405_) {
@@ -400,7 +_,7 @@
     }
 
     public void setDancing(boolean p_240178_) {
-        if (!this.level().isClientSide() && this.isEffectiveAi() && (!p_240178_ || !this.isPanicking())) {
+        if (!this.level().isClientSide && this.isEffectiveAi() && (!p_240178_ || !this.isPanicking())) {
             this.entityData.set(DATA_DANCING, p_240178_);
         }
     }
