--- a/net/minecraft/world/item/BoatItem.java
+++ b/net/minecraft/world/item/BoatItem.java
@@ -59,7 +_,7 @@
                     if (!p_40622_.noCollision(abstractboat, abstractboat.getBoundingBox())) {
                         return InteractionResult.FAIL;
                     } else {
-                        if (!p_40622_.isClientSide()) {
+                        if (!p_40622_.isClientSide) {
                             p_40622_.addFreshEntity(abstractboat);
                             p_40622_.gameEvent(p_40623_, GameEvent.ENTITY_PLACE, hitresult.getLocation());
                             itemstack.consume(1, p_40623_);
