--- a/net/minecraft/world/item/DebugStickItem.java
+++ b/net/minecraft/world/item/DebugStickItem.java
@@ -27,7 +_,7 @@
 
     @Override
     public boolean canDestroyBlock(ItemStack p_393705_, BlockState p_393781_, Level p_393500_, BlockPos p_394185_, LivingEntity p_394515_) {
-        if (!p_393500_.isClientSide() && p_394515_ instanceof Player player) {
+        if (!p_393500_.isClientSide && p_394515_ instanceof Player player) {
             this.handleInteraction(player, p_393781_, p_393500_, p_394185_, false, p_393705_);
         }
 
@@ -38,7 +_,7 @@
     public InteractionResult useOn(UseOnContext p_40960_) {
         Player player = p_40960_.getPlayer();
         Level level = p_40960_.getLevel();
-        if (!level.isClientSide() && player != null) {
+        if (!level.isClientSide && player != null) {
             BlockPos blockpos = p_40960_.getClickedPos();
             if (!this.handleInteraction(player, level.getBlockState(blockpos), level, blockpos, true, p_40960_.getItemInHand())) {
                 return InteractionResult.FAIL;
