--- a/net/minecraft/world/item/FoodOnAStickItem.java
+++ b/net/minecraft/world/item/FoodOnAStickItem.java
@@ -7,6 +_,7 @@
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.entity.EquipmentSlot;
 import net.minecraft.world.entity.ItemSteerable;
+import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.level.Level;
 
@@ -23,12 +_,12 @@
     @Override
     public InteractionResult use(Level p_41314_, Player p_41315_, InteractionHand p_41316_) {
         ItemStack itemstack = p_41315_.getItemInHand(p_41316_);
-        if (p_41314_.isClientSide()) {
+        if (p_41314_.isClientSide) {
             return InteractionResult.PASS;
         } else {
             Entity entity = p_41315_.getControlledVehicle();
             if (p_41315_.isPassenger() && entity instanceof ItemSteerable itemsteerable && entity.getType() == this.canInteractWith && itemsteerable.boost()) {
-                EquipmentSlot equipmentslot = p_41316_.asEquipmentSlot();
+                EquipmentSlot equipmentslot = LivingEntity.getSlotForHand(p_41316_);
                 ItemStack itemstack1 = itemstack.hurtAndConvertOnBreak(this.consumeItemDamage, Items.FISHING_ROD, p_41315_, equipmentslot);
                 return InteractionResult.SUCCESS_SERVER.heldItemTransformedTo(itemstack1);
             } else {
