--- a/net/minecraft/world/item/HangingEntityItem.java
+++ b/net/minecraft/world/item/HangingEntityItem.java
@@ -61,7 +_,7 @@
 
             EntityType.<HangingEntity>createDefaultStackConfig(level, itemstack, player).accept(hangingentity);
             if (hangingentity.survives()) {
-                if (!level.isClientSide()) {
+                if (!level.isClientSide) {
                     hangingentity.playPlacementSound();
                     level.gameEvent(player, GameEvent.ENTITY_PLACE, hangingentity.position());
                     level.addFreshEntity(hangingentity);
