--- a/net/minecraft/world/item/LeadItem.java
+++ b/net/minecraft/world/item/LeadItem.java
@@ -25,7 +_,7 @@
         BlockState blockstate = level.getBlockState(blockpos);
         if (blockstate.is(BlockTags.FENCES)) {
             Player player = p_42834_.getPlayer();
-            if (!level.isClientSide() && player != null) {
+            if (!level.isClientSide && player != null) {
                 return bindPlayerMobs(player, level, blockpos);
             }
         }
