--- a/net/minecraft/world/item/MapItem.java
+++ b/net/minecraft/world/item/MapItem.java
@@ -54,6 +_,16 @@
 
     @Nullable
     public static MapItemSavedData getSavedData(ItemStack p_42854_, Level p_42855_) {
+        // Neo: Add instance method so that mods can override
+        Item map = p_42854_.getItem();
+        if(map instanceof MapItem) {
+            return ((MapItem)map).getCustomMapData(p_42854_, p_42855_);
+        }
+        return null;
+    }
+
+    @Nullable
+    protected MapItemSavedData getCustomMapData(ItemStack p_42854_, Level p_42855_) {
         MapId mapid = p_42854_.get(DataComponents.MAP_ID);
         return getSavedData(mapid, p_42855_);
     }
@@ -322,7 +_,7 @@
     public InteractionResult useOn(UseOnContext p_42885_) {
         BlockState blockstate = p_42885_.getLevel().getBlockState(p_42885_.getClickedPos());
         if (blockstate.is(BlockTags.BANNERS)) {
-            if (!p_42885_.getLevel().isClientSide()) {
+            if (!p_42885_.getLevel().isClientSide) {
                 MapItemSavedData mapitemsaveddata = getSavedData(p_42885_.getItemInHand(), p_42885_.getLevel());
                 if (mapitemsaveddata != null && !mapitemsaveddata.toggleBanner(p_42885_.getLevel(), p_42885_.getClickedPos())) {
                     return InteractionResult.FAIL;
