--- a/net/minecraft/world/item/SolidBucketItem.java
+++ b/net/minecraft/world/item/SolidBucketItem.java
@@ -41,7 +_,7 @@
     @Override
     public boolean emptyContents(@Nullable LivingEntity p_393886_, Level p_151193_, BlockPos p_151194_, @Nullable BlockHitResult p_151195_) {
         if (p_151193_.isInWorldBounds(p_151194_) && p_151193_.isEmptyBlock(p_151194_)) {
-            if (!p_151193_.isClientSide()) {
+            if (!p_151193_.isClientSide) {
                 p_151193_.setBlock(p_151194_, this.getBlock().defaultBlockState(), 3);
             }
 
