--- a/net/minecraft/client/gui/font/glyphs/EffectGlyph.java
+++ /dev/null
@@ -1,12 +_,0 @@
-package net.minecraft.client.gui.font.glyphs;
-
-import net.minecraft.client.gui.font.TextRenderable;
-import net.neoforged.api.distmarker.Dist;
-import net.neoforged.api.distmarker.OnlyIn;
-
-@OnlyIn(Dist.CLIENT)
-public interface EffectGlyph {
-    TextRenderable createEffect(
-        float p_439377_, float p_439337_, float p_440013_, float p_439810_, float p_439827_, int p_438939_, int p_439390_, float p_439024_
-    );
-}
