--- a/net/minecraft/client/gui/font/glyphs/SpecialGlyphs.java
+++ b/net/minecraft/client/gui/font/glyphs/SpecialGlyphs.java
@@ -1,13 +_,12 @@
 package net.minecraft.client.gui.font.glyphs;
 
-import com.mojang.blaze3d.font.GlyphBitmap;
 import com.mojang.blaze3d.font.GlyphInfo;
+import com.mojang.blaze3d.font.SheetGlyphInfo;
 import com.mojang.blaze3d.platform.NativeImage;
 import com.mojang.blaze3d.systems.RenderSystem;
 import com.mojang.blaze3d.textures.GpuTexture;
+import java.util.function.Function;
 import java.util.function.Supplier;
-import javax.annotation.Nullable;
-import net.minecraft.client.gui.font.GlyphStitcher;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
@@ -47,11 +_,10 @@
         return this.image.getWidth() + 1;
     }
 
-    @Nullable
-    public BakedSheetGlyph bake(GlyphStitcher p_433022_) {
-        return p_433022_.stitch(
-            this,
-            new GlyphBitmap() {
+    @Override
+    public BakedGlyph bake(Function<SheetGlyphInfo, BakedGlyph> p_232616_) {
+        return p_232616_.apply(
+            new SheetGlyphInfo() {
                 @Override
                 public int getPixelWidth() {
                     return SpecialGlyphs.this.image.getWidth();
