--- a/net/minecraft/client/resources/sounds/SoundEventRegistrationSerializer.java
+++ b/net/minecraft/client/resources/sounds/SoundEventRegistrationSerializer.java
@@ -9,7 +_,6 @@
 import com.google.gson.JsonParseException;
 import java.lang.reflect.Type;
 import java.util.List;
-import java.util.Objects;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.util.GsonHelper;
 import net.minecraft.util.valueproviders.ConstantFloat;
@@ -68,7 +_,7 @@
         Sound.Type sound$type = p_119834_;
         if (p_119833_.has("type")) {
             sound$type = Sound.Type.getByName(GsonHelper.getAsString(p_119833_, "type"));
-            Objects.requireNonNull(sound$type, "Invalid type");
+            Validate.notNull(sound$type, "Invalid type");
         }
 
         return sound$type;
