--- a/net/minecraft/client/renderer/texture/TextureAtlas.java
+++ b/net/minecraft/client/renderer/texture/TextureAtlas.java
@@ -13,7 +_,6 @@
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
-import java.util.Objects;
 import java.util.Map.Entry;
 import javax.annotation.Nullable;
 import net.minecraft.CrashReport;
@@ -92,6 +_,8 @@
             this.sprites = List.copyOf(list);
             this.animatedTextures = List.copyOf(list1);
         }
+
+        net.neoforged.neoforge.client.ClientHooks.onTextureAtlasStitched(this);
     }
 
     @Override
@@ -146,10 +_,6 @@
         }
     }
 
-    public TextureAtlasSprite missingSprite() {
-        return Objects.requireNonNull(this.missingSprite, "Atlas not initialized");
-    }
-
     public void clearTextureData() {
         this.sprites.forEach(SpriteContents::close);
         this.animatedTextures.forEach(TextureAtlasSprite.Ticker::close);
@@ -173,5 +_,9 @@
 
     int getHeight() {
         return this.height;
+    }
+
+    public Map<ResourceLocation, TextureAtlasSprite> getTextures() {
+        return java.util.Collections.unmodifiableMap(texturesByName);
     }
 }
