--- a/net/minecraft/client/renderer/entity/FishingHookRenderer.java
+++ b/net/minecraft/client/renderer/entity/FishingHookRenderer.java
@@ -3,8 +_,8 @@
 import com.mojang.blaze3d.vertex.PoseStack;
 import com.mojang.blaze3d.vertex.VertexConsumer;
 import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.RenderType;
-import net.minecraft.client.renderer.SubmitNodeCollector;
 import net.minecraft.client.renderer.culling.Frustum;
 import net.minecraft.client.renderer.entity.state.FishingHookRenderState;
 import net.minecraft.client.renderer.texture.OverlayTexture;
@@ -32,37 +_,35 @@
         return super.shouldRender(p_363069_, p_362635_, p_361840_, p_361502_, p_360380_) && p_363069_.getPlayerOwner() != null;
     }
 
-    public void submit(FishingHookRenderState p_435975_, PoseStack p_434669_, SubmitNodeCollector p_434047_) {
-        p_434669_.pushPose();
-        p_434669_.pushPose();
-        p_434669_.scale(0.5F, 0.5F, 0.5F);
-        p_434669_.mulPose(this.entityRenderDispatcher.cameraOrientation());
-        p_434047_.submitCustomGeometry(p_434669_, RENDER_TYPE, (p_434943_, p_432878_) -> {
-            vertex(p_432878_, p_434943_, p_435975_.lightCoords, 0.0F, 0, 0, 1);
-            vertex(p_432878_, p_434943_, p_435975_.lightCoords, 1.0F, 0, 1, 1);
-            vertex(p_432878_, p_434943_, p_435975_.lightCoords, 1.0F, 1, 1, 0);
-            vertex(p_432878_, p_434943_, p_435975_.lightCoords, 0.0F, 1, 0, 0);
-        });
-        p_434669_.popPose();
-        float f = (float)p_435975_.lineOriginOffset.x;
-        float f1 = (float)p_435975_.lineOriginOffset.y;
-        float f2 = (float)p_435975_.lineOriginOffset.z;
-        p_434047_.submitCustomGeometry(p_434669_, RenderType.lines(), (p_436505_, p_436506_) -> {
-            int i = 16;
-
-            for (int j = 0; j < 16; j++) {
-                float f3 = fraction(j, 16);
-                float f4 = fraction(j + 1, 16);
-                stringVertex(f, f1, f2, p_436506_, p_436505_, f3, f4);
-                stringVertex(f, f1, f2, p_436506_, p_436505_, f4, f3);
-            }
-        });
-        p_434669_.popPose();
-        super.submit(p_435975_, p_434669_, p_434047_);
+    public void render(FishingHookRenderState p_362456_, PoseStack p_114699_, MultiBufferSource p_114700_, int p_114701_) {
+        p_114699_.pushPose();
+        p_114699_.pushPose();
+        p_114699_.scale(0.5F, 0.5F, 0.5F);
+        p_114699_.mulPose(this.entityRenderDispatcher.cameraOrientation());
+        PoseStack.Pose posestack$pose = p_114699_.last();
+        VertexConsumer vertexconsumer = p_114700_.getBuffer(RENDER_TYPE);
+        vertex(vertexconsumer, posestack$pose, p_114701_, 0.0F, 0, 0, 1);
+        vertex(vertexconsumer, posestack$pose, p_114701_, 1.0F, 0, 1, 1);
+        vertex(vertexconsumer, posestack$pose, p_114701_, 1.0F, 1, 1, 0);
+        vertex(vertexconsumer, posestack$pose, p_114701_, 0.0F, 1, 0, 0);
+        p_114699_.popPose();
+        float f = (float)p_362456_.lineOriginOffset.x;
+        float f1 = (float)p_362456_.lineOriginOffset.y;
+        float f2 = (float)p_362456_.lineOriginOffset.z;
+        VertexConsumer vertexconsumer1 = p_114700_.getBuffer(RenderType.lineStrip());
+        PoseStack.Pose posestack$pose1 = p_114699_.last();
+        int i = 16;
+
+        for (int j = 0; j <= 16; j++) {
+            stringVertex(f, f1, f2, vertexconsumer1, posestack$pose1, fraction(j, 16), fraction(j + 1, 16));
+        }
+
+        p_114699_.popPose();
+        super.render(p_362456_, p_114699_, p_114700_, p_114701_);
     }
 
     public static HumanoidArm getHoldingArm(Player p_386900_) {
-        return p_386900_.getMainHandItem().getItem() instanceof FishingRodItem ? p_386900_.getMainArm() : p_386900_.getMainArm().getOpposite();
+        return p_386900_.getMainHandItem().canPerformAction(net.neoforged.neoforge.common.ItemAbilities.FISHING_ROD_CAST) ? p_386900_.getMainArm() : p_386900_.getMainArm().getOpposite();
     }
 
     private Vec3 getPlayerHandPos(Player p_340935_, float p_340872_, float p_341261_) {
