--- a/net/minecraft/client/renderer/entity/GiantMobRenderer.java
+++ b/net/minecraft/client/renderer/entity/GiantMobRenderer.java
@@ -20,7 +_,10 @@
         this.addLayer(new ItemInHandLayer<>(this));
         this.addLayer(
             new HumanoidArmorLayer<>(
-                this, ArmorModelSet.bake(ModelLayers.GIANT_ARMOR, p_174131_.getModelSet(), GiantZombieModel::new), p_174131_.getEquipmentRenderer()
+                this,
+                new GiantZombieModel(p_174131_.bakeLayer(ModelLayers.GIANT_INNER_ARMOR)),
+                new GiantZombieModel(p_174131_.bakeLayer(ModelLayers.GIANT_OUTER_ARMOR)),
+                p_174131_.getEquipmentRenderer()
             )
         );
     }
