--- a/net/minecraft/client/renderer/entity/HumanoidMobRenderer.java
+++ b/net/minecraft/client/renderer/entity/HumanoidMobRenderer.java
@@ -30,7 +_,7 @@
 
     public HumanoidMobRenderer(EntityRendererProvider.Context p_361718_, M p_361903_, M p_360842_, float p_364257_, CustomHeadLayer.Transforms p_364599_) {
         super(p_361718_, p_361903_, p_360842_, p_364257_);
-        this.addLayer(new CustomHeadLayer<>(this, p_361718_.getModelSet(), p_361718_.getPlayerSkinRenderCache(), p_364599_));
+        this.addLayer(new CustomHeadLayer<>(this, p_361718_.getModelSet(), p_364599_));
         this.addLayer(new WingsLayer<>(this, p_361718_.getModelSet(), p_361718_.getEquipmentRenderer()));
         this.addLayer(new ItemInHandLayer<>(this));
     }
@@ -51,7 +_,7 @@
         p_362998_.isCrouching = p_365104_.isCrouching();
         p_362998_.isFallFlying = p_365104_.isFallFlying();
         p_362998_.isVisuallySwimming = p_365104_.isVisuallySwimming();
-        p_362998_.isPassenger = p_365104_.isPassenger();
+        p_362998_.isPassenger = p_365104_.isPassenger() && (p_365104_.getVehicle() != null && p_365104_.getVehicle().shouldRiderSit());
         p_362998_.speedValue = 1.0F;
         if (p_362998_.isFallFlying) {
             p_362998_.speedValue = (float)p_365104_.getDeltaMovement().lengthSqr();
