--- a/net/minecraft/client/renderer/entity/ItemFrameRenderer.java
+++ b/net/minecraft/client/renderer/entity/ItemFrameRenderer.java
@@ -4,9 +_,10 @@
 import com.mojang.math.Axis;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.MapRenderer;
+import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.RenderType;
-import net.minecraft.client.renderer.SubmitNodeCollector;
 import net.minecraft.client.renderer.block.BlockRenderDispatcher;
+import net.minecraft.client.renderer.block.ModelBlockRenderer;
 import net.minecraft.client.renderer.block.model.BlockStateModel;
 import net.minecraft.client.renderer.entity.state.ItemFrameRenderState;
 import net.minecraft.client.renderer.item.ItemModelResolver;
@@ -48,14 +_,14 @@
             : super.getBlockLightLevel(p_174216_, p_174217_);
     }
 
-    public void submit(ItemFrameRenderState p_433594_, PoseStack p_434437_, SubmitNodeCollector p_434593_) {
-        super.submit(p_433594_, p_434437_, p_434593_);
-        p_434437_.pushPose();
-        Direction direction = p_433594_.direction;
-        Vec3 vec3 = this.getRenderOffset(p_433594_);
-        p_434437_.translate(-vec3.x(), -vec3.y(), -vec3.z());
+    public void render(ItemFrameRenderState p_364723_, PoseStack p_115079_, MultiBufferSource p_115080_, int p_115081_) {
+        super.render(p_364723_, p_115079_, p_115080_, p_115081_);
+        p_115079_.pushPose();
+        Direction direction = p_364723_.direction;
+        Vec3 vec3 = this.getRenderOffset(p_364723_);
+        p_115079_.translate(-vec3.x(), -vec3.y(), -vec3.z());
         double d0 = 0.46875;
-        p_434437_.translate(direction.getStepX() * 0.46875, direction.getStepY() * 0.46875, direction.getStepZ() * 0.46875);
+        p_115079_.translate(direction.getStepX() * 0.46875, direction.getStepY() * 0.46875, direction.getStepZ() * 0.46875);
         float f;
         float f1;
         if (direction.getAxis().isHorizontal()) {
@@ -66,51 +_,52 @@
             f1 = 180.0F;
         }
 
-        p_434437_.mulPose(Axis.XP.rotationDegrees(f));
-        p_434437_.mulPose(Axis.YP.rotationDegrees(f1));
-        if (!p_433594_.isInvisible) {
-            BlockState blockstate = BlockStateDefinitions.getItemFrameFakeState(p_433594_.isGlowFrame, p_433594_.mapId != null);
+        p_115079_.mulPose(Axis.XP.rotationDegrees(f));
+        p_115079_.mulPose(Axis.YP.rotationDegrees(f1));
+        if (!p_364723_.isInvisible) {
+            BlockState blockstate = BlockStateDefinitions.getItemFrameFakeState(p_364723_.isGlowFrame, p_364723_.mapId != null);
             BlockStateModel blockstatemodel = this.blockRenderer.getBlockModel(blockstate);
-            p_434437_.pushPose();
-            p_434437_.translate(-0.5F, -0.5F, -0.5F);
-            p_434593_.submitBlockModel(
-                p_434437_,
-                RenderType.entitySolidZOffsetForward(TextureAtlas.LOCATION_BLOCKS),
+            p_115079_.pushPose();
+            p_115079_.translate(-0.5F, -0.5F, -0.5F);
+            ModelBlockRenderer.renderModel(
+                p_115079_.last(),
+                p_115080_.getBuffer(RenderType.entitySolidZOffsetForward(TextureAtlas.LOCATION_BLOCKS)),
                 blockstatemodel,
                 1.0F,
                 1.0F,
                 1.0F,
-                p_433594_.lightCoords,
-                OverlayTexture.NO_OVERLAY,
-                p_433594_.outlineColor
+                p_115081_,
+                OverlayTexture.NO_OVERLAY
             );
-            p_434437_.popPose();
+            p_115079_.popPose();
         }
 
-        if (p_433594_.isInvisible) {
-            p_434437_.translate(0.0F, 0.0F, 0.5F);
+        if (p_364723_.isInvisible) {
+            p_115079_.translate(0.0F, 0.0F, 0.5F);
         } else {
-            p_434437_.translate(0.0F, 0.0F, 0.4375F);
+            p_115079_.translate(0.0F, 0.0F, 0.4375F);
         }
 
-        if (p_433594_.mapId != null) {
-            int j = p_433594_.rotation % 4 * 2;
-            p_434437_.mulPose(Axis.ZP.rotationDegrees(j * 360.0F / 8.0F));
-            p_434437_.mulPose(Axis.ZP.rotationDegrees(180.0F));
+        if (!net.neoforged.neoforge.common.NeoForge.EVENT_BUS.post(new net.neoforged.neoforge.client.event.RenderItemInFrameEvent(p_364723_, this, p_115079_, p_115080_, p_115081_)).isCanceled()) {
+        if (p_364723_.mapId != null) {
+            int j = p_364723_.rotation % 4 * 2;
+            p_115079_.mulPose(Axis.ZP.rotationDegrees(j * 360.0F / 8.0F));
+            p_115079_.mulPose(Axis.ZP.rotationDegrees(180.0F));
             float f2 = 0.0078125F;
-            p_434437_.scale(0.0078125F, 0.0078125F, 0.0078125F);
-            p_434437_.translate(-64.0F, -64.0F, 0.0F);
-            p_434437_.translate(0.0F, 0.0F, -1.0F);
-            int i = this.getLightCoords(p_433594_.isGlowFrame, 15728850, p_433594_.lightCoords);
-            this.mapRenderer.render(p_433594_.mapRenderState, p_434437_, p_434593_, true, i);
-        } else if (!p_433594_.item.isEmpty()) {
-            p_434437_.mulPose(Axis.ZP.rotationDegrees(p_433594_.rotation * 360.0F / 8.0F));
-            int k = this.getLightCoords(p_433594_.isGlowFrame, 15728880, p_433594_.lightCoords);
-            p_434437_.scale(0.5F, 0.5F, 0.5F);
-            p_433594_.item.submit(p_434437_, p_434593_, k, OverlayTexture.NO_OVERLAY, p_433594_.outlineColor);
+            p_115079_.scale(0.0078125F, 0.0078125F, 0.0078125F);
+            p_115079_.translate(-64.0F, -64.0F, 0.0F);
+            p_115079_.translate(0.0F, 0.0F, -1.0F);
+            int i = this.getLightCoords(p_364723_.isGlowFrame, 15728850, p_115081_);
+            this.mapRenderer.render(p_364723_.mapRenderState, p_115079_, p_115080_, true, i);
+        } else if (!p_364723_.item.isEmpty()) {
+            p_115079_.mulPose(Axis.ZP.rotationDegrees(p_364723_.rotation * 360.0F / 8.0F));
+            int k = this.getLightCoords(p_364723_.isGlowFrame, 15728880, p_115081_);
+            p_115079_.scale(0.5F, 0.5F, 0.5F);
+            p_364723_.item.render(p_115079_, p_115080_, k, OverlayTexture.NO_OVERLAY);
+        }
         }
 
-        p_434437_.popPose();
+        p_115079_.popPose();
     }
 
     private int getLightCoords(boolean p_364675_, int p_174210_, int p_174211_) {
@@ -144,7 +_,7 @@
         if (!itemstack.isEmpty()) {
             MapId mapid = p_363125_.getFramedMapId(itemstack);
             if (mapid != null) {
-                MapItemSavedData mapitemsaveddata = p_363125_.level().getMapData(mapid);
+                MapItemSavedData mapitemsaveddata = net.minecraft.world.item.MapItem.getSavedData(itemstack, p_363125_.level());
                 if (mapitemsaveddata != null) {
                     this.mapRenderer.extractRenderState(mapid, mapitemsaveddata, p_362907_.mapRenderState);
                     p_362907_.mapId = mapid;
