--- a/net/minecraft/client/renderer/entity/ShulkerBulletRenderer.java
+++ b/net/minecraft/client/renderer/entity/ShulkerBulletRenderer.java
@@ -1,11 +_,12 @@
 package net.minecraft.client.renderer.entity;
 
 import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.blaze3d.vertex.VertexConsumer;
 import com.mojang.math.Axis;
 import net.minecraft.client.model.ShulkerBulletModel;
 import net.minecraft.client.model.geom.ModelLayers;
+import net.minecraft.client.renderer.MultiBufferSource;
 import net.minecraft.client.renderer.RenderType;
-import net.minecraft.client.renderer.SubmitNodeCollector;
 import net.minecraft.client.renderer.entity.state.ShulkerBulletRenderState;
 import net.minecraft.client.renderer.texture.OverlayTexture;
 import net.minecraft.core.BlockPos;
@@ -30,31 +_,22 @@
         return 15;
     }
 
-    public void submit(ShulkerBulletRenderState p_433448_, PoseStack p_435021_, SubmitNodeCollector p_435459_) {
-        p_435021_.pushPose();
-        float f = p_433448_.ageInTicks;
-        p_435021_.translate(0.0F, 0.15F, 0.0F);
-        p_435021_.mulPose(Axis.YP.rotationDegrees(Mth.sin(f * 0.1F) * 180.0F));
-        p_435021_.mulPose(Axis.XP.rotationDegrees(Mth.cos(f * 0.1F) * 180.0F));
-        p_435021_.mulPose(Axis.ZP.rotationDegrees(Mth.sin(f * 0.15F) * 360.0F));
-        p_435021_.scale(-0.5F, -0.5F, 0.5F);
-        p_435459_.submitModel(
-            this.model,
-            p_433448_,
-            p_435021_,
-            this.model.renderType(TEXTURE_LOCATION),
-            p_433448_.lightCoords,
-            OverlayTexture.NO_OVERLAY,
-            p_433448_.outlineColor,
-            null
-        );
-        p_435021_.scale(1.5F, 1.5F, 1.5F);
-        p_435459_.order(1)
-            .submitModel(
-                this.model, p_433448_, p_435021_, RENDER_TYPE, p_433448_.lightCoords, OverlayTexture.NO_OVERLAY, 654311423, null, p_433448_.outlineColor, null
-            );
-        p_435021_.popPose();
-        super.submit(p_433448_, p_435021_, p_435459_);
+    public void render(ShulkerBulletRenderState p_364049_, PoseStack p_115853_, MultiBufferSource p_115854_, int p_115855_) {
+        p_115853_.pushPose();
+        float f = p_364049_.ageInTicks;
+        p_115853_.translate(0.0F, 0.15F, 0.0F);
+        p_115853_.mulPose(Axis.YP.rotationDegrees(Mth.sin(f * 0.1F) * 180.0F));
+        p_115853_.mulPose(Axis.XP.rotationDegrees(Mth.cos(f * 0.1F) * 180.0F));
+        p_115853_.mulPose(Axis.ZP.rotationDegrees(Mth.sin(f * 0.15F) * 360.0F));
+        p_115853_.scale(-0.5F, -0.5F, 0.5F);
+        this.model.setupAnim(p_364049_);
+        VertexConsumer vertexconsumer = p_115854_.getBuffer(this.model.renderType(TEXTURE_LOCATION));
+        this.model.renderToBuffer(p_115853_, vertexconsumer, p_115855_, OverlayTexture.NO_OVERLAY);
+        p_115853_.scale(1.5F, 1.5F, 1.5F);
+        VertexConsumer vertexconsumer1 = p_115854_.getBuffer(RENDER_TYPE);
+        this.model.renderToBuffer(p_115853_, vertexconsumer1, p_115855_, OverlayTexture.NO_OVERLAY, 654311423);
+        p_115853_.popPose();
+        super.render(p_364049_, p_115853_, p_115854_, p_115855_);
     }
 
     public ShulkerBulletRenderState createRenderState() {
