--- a/net/minecraft/client/renderer/entity/VillagerRenderer.java
+++ b/net/minecraft/client/renderer/entity/VillagerRenderer.java
@@ -19,16 +_,8 @@
 
     public VillagerRenderer(EntityRendererProvider.Context p_174437_) {
         super(p_174437_, new VillagerModel(p_174437_.bakeLayer(ModelLayers.VILLAGER)), new VillagerModel(p_174437_.bakeLayer(ModelLayers.VILLAGER_BABY)), 0.5F);
-        this.addLayer(new CustomHeadLayer<>(this, p_174437_.getModelSet(), p_174437_.getPlayerSkinRenderCache(), CUSTOM_HEAD_TRANSFORMS));
-        this.addLayer(
-            new VillagerProfessionLayer<>(
-                this,
-                p_174437_.getResourceManager(),
-                "villager",
-                new VillagerModel(p_174437_.bakeLayer(ModelLayers.VILLAGER_NO_HAT)),
-                new VillagerModel(p_174437_.bakeLayer(ModelLayers.VILLAGER_BABY_NO_HAT))
-            )
-        );
+        this.addLayer(new CustomHeadLayer<>(this, p_174437_.getModelSet(), CUSTOM_HEAD_TRANSFORMS));
+        this.addLayer(new VillagerProfessionLayer<>(this, p_174437_.getResourceManager(), "villager"));
         this.addLayer(new CrossedArmsItemLayer<>(this));
     }
 
