--- a/net/minecraft/client/renderer/entity/WanderingTraderRenderer.java
+++ b/net/minecraft/client/renderer/entity/WanderingTraderRenderer.java
@@ -17,7 +_,7 @@
 
     public WanderingTraderRenderer(EntityRendererProvider.Context p_174441_) {
         super(p_174441_, new VillagerModel(p_174441_.bakeLayer(ModelLayers.WANDERING_TRADER)), 0.5F);
-        this.addLayer(new CustomHeadLayer<>(this, p_174441_.getModelSet(), p_174441_.getPlayerSkinRenderCache()));
+        this.addLayer(new CustomHeadLayer<>(this, p_174441_.getModelSet()));
         this.addLayer(new CrossedArmsItemLayer<>(this));
     }
 
