--- a/net/minecraft/client/renderer/entity/ZombieRenderer.java
+++ b/net/minecraft/client/renderer/entity/ZombieRenderer.java
@@ -11,7 +_,15 @@
 @OnlyIn(Dist.CLIENT)
 public class ZombieRenderer extends AbstractZombieRenderer<Zombie, ZombieRenderState, ZombieModel<ZombieRenderState>> {
     public ZombieRenderer(EntityRendererProvider.Context p_174456_) {
-        this(p_174456_, ModelLayers.ZOMBIE, ModelLayers.ZOMBIE_BABY, ModelLayers.ZOMBIE_ARMOR, ModelLayers.ZOMBIE_BABY_ARMOR);
+        this(
+            p_174456_,
+            ModelLayers.ZOMBIE,
+            ModelLayers.ZOMBIE_BABY,
+            ModelLayers.ZOMBIE_INNER_ARMOR,
+            ModelLayers.ZOMBIE_OUTER_ARMOR,
+            ModelLayers.ZOMBIE_BABY_INNER_ARMOR,
+            ModelLayers.ZOMBIE_BABY_OUTER_ARMOR
+        );
     }
 
     public ZombieRenderState createRenderState() {
@@ -22,15 +_,19 @@
         EntityRendererProvider.Context p_174458_,
         ModelLayerLocation p_174459_,
         ModelLayerLocation p_174460_,
-        ArmorModelSet<ModelLayerLocation> p_435945_,
-        ArmorModelSet<ModelLayerLocation> p_434195_
+        ModelLayerLocation p_174461_,
+        ModelLayerLocation p_362432_,
+        ModelLayerLocation p_361708_,
+        ModelLayerLocation p_365510_
     ) {
         super(
             p_174458_,
             new ZombieModel<>(p_174458_.bakeLayer(p_174459_)),
             new ZombieModel<>(p_174458_.bakeLayer(p_174460_)),
-            ArmorModelSet.bake(p_435945_, p_174458_.getModelSet(), ZombieModel::new),
-            ArmorModelSet.bake(p_434195_, p_174458_.getModelSet(), ZombieModel::new)
+            new ZombieModel<>(p_174458_.bakeLayer(p_174461_)),
+            new ZombieModel<>(p_174458_.bakeLayer(p_362432_)),
+            new ZombieModel<>(p_174458_.bakeLayer(p_361708_)),
+            new ZombieModel<>(p_174458_.bakeLayer(p_365510_))
         );
     }
 }
