--- a/net/minecraft/world/entity/ai/behavior/warden/TryToSniff.java
+++ b/net/minecraft/world/entity/ai/behavior/warden/TryToSniff.java
@@ -23,11 +_,11 @@
                     p_259979_.present(MemoryModuleType.NEAREST_ATTACKABLE),
                     p_259979_.absent(MemoryModuleType.DISTURBANCE_LOCATION)
                 )
-                .apply(p_259979_, (p_260219_, p_260252_, p_260090_, p_259577_, p_260020_) -> (p_432539_, p_432540_, p_432541_) -> {
+                .apply(p_259979_, (p_260219_, p_260252_, p_260090_, p_259577_, p_260020_) -> (p_425340_, p_425341_, p_425342_) -> {
                     p_260219_.set(Unit.INSTANCE);
-                    p_260090_.setWithExpiry(Unit.INSTANCE, SNIFF_COOLDOWN.sample(p_432539_.getRandom()));
+                    p_260090_.setWithExpiry(Unit.INSTANCE, SNIFF_COOLDOWN.sample(p_425340_.getRandom()));
                     p_260252_.erase();
-                    p_432540_.setPose(Pose.SNIFFING);
+                    p_425341_.setPose(Pose.SNIFFING);
                     return true;
                 })
         );
