--- a/net/minecraft/world/level/levelgen/feature/FeatureCountTracker.java
+++ b/net/minecraft/world/level/levelgen/feature/FeatureCountTracker.java
@@ -13,6 +_,7 @@
 import net.minecraft.core.Registry;
 import net.minecraft.core.registries.Registries;
 import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.level.levelgen.placement.PlacedFeature;
 import org.apache.commons.lang3.mutable.MutableInt;
@@ -56,23 +_,25 @@
         LOGGER.debug("Logging feature counts:");
         data.asMap()
             .forEach(
-                (p_432720_, p_432721_) -> {
-                    String s = p_432720_.dimension().location().toString();
-                    boolean flag = p_432720_.getServer().isRunning();
-                    Registry<PlacedFeature> registry = p_432720_.registryAccess().lookupOrThrow(Registries.PLACED_FEATURE);
+                (p_190888_, p_190889_) -> {
+                    String s = p_190888_.dimension().location().toString();
+                    boolean flag = p_190888_.getServer().isRunning();
+                    Registry<PlacedFeature> registry = p_190888_.registryAccess().lookupOrThrow(Registries.PLACED_FEATURE);
                     String s1 = (flag ? "running" : "dead") + " " + s;
-                    Integer integer = p_432721_.chunksWithFeatures().getValue();
-                    LOGGER.debug("{} total_chunks: {}", s1, integer);
-                    p_432721_.featureData()
+                    Integer integer = p_190889_.chunksWithFeatures().getValue();
+                    LOGGER.debug(s1 + " total_chunks: " + integer);
+                    p_190889_.featureData()
                         .forEach(
-                            (p_432725_, p_432726_) -> LOGGER.debug(
-                                "{} {} {} {} {} {}",
-                                s1,
-                                String.format(Locale.ROOT, "%10d", p_432726_),
-                                String.format(Locale.ROOT, "%10f", (double)p_432726_.intValue() / integer.intValue()),
-                                p_432725_.topFeature().flatMap(registry::getResourceKey).map(ResourceKey::location),
-                                p_432725_.feature().feature(),
-                                p_432725_.feature()
+                            (p_339550_, p_339551_) -> LOGGER.debug(
+                                s1
+                                    + " "
+                                    + String.format(Locale.ROOT, "%10d ", p_339551_)
+                                    + String.format(Locale.ROOT, "%10f ", (double)p_339551_.intValue() / integer.intValue())
+                                    + p_339550_.topFeature().flatMap(registry::getResourceKey).<ResourceLocation>map(ResourceKey::location)
+                                    + " "
+                                    + p_339550_.feature().feature()
+                                    + " "
+                                    + p_339550_.feature()
                             )
                         );
                 }
