--- a/net/minecraft/server/level/progress/ChunkLoadStatusView.java
+++ /dev/null
@@ -1,16 +_,0 @@
-package net.minecraft.server.level.progress;
-
-import javax.annotation.Nullable;
-import net.minecraft.resources.ResourceKey;
-import net.minecraft.world.level.ChunkPos;
-import net.minecraft.world.level.Level;
-import net.minecraft.world.level.chunk.status.ChunkStatus;
-
-public interface ChunkLoadStatusView {
-    void moveTo(ResourceKey<Level> p_433845_, ChunkPos p_435642_);
-
-    @Nullable
-    ChunkStatus get(int p_434162_, int p_435494_);
-
-    int radius();
-}
