--- a/net/minecraft/data/tags/VanillaBlockTagsProvider.java
+++ b/net/minecraft/data/tags/VanillaBlockTagsProvider.java
@@ -179,8 +_,7 @@
                 Blocks.GREEN_STAINED_GLASS,
                 Blocks.RED_STAINED_GLASS,
                 Blocks.BLACK_STAINED_GLASS,
-                Blocks.TINTED_GLASS,
-                Blocks.BARRIER
+                Blocks.TINTED_GLASS
             );
         this.tag(BlockTags.UNDERWATER_BONEMEALS).add(Blocks.SEAGRASS).addTag(BlockTags.CORALS).addTag(BlockTags.WALL_CORALS);
         this.tag(BlockTags.BAMBOO_PLANTABLE_ON)
@@ -281,7 +_,7 @@
         this.tag(BlockTags.BEACON_BASE_BLOCKS).add(Blocks.NETHERITE_BLOCK, Blocks.EMERALD_BLOCK, Blocks.DIAMOND_BLOCK, Blocks.GOLD_BLOCK, Blocks.IRON_BLOCK);
         this.tag(BlockTags.SOUL_SPEED_BLOCKS).add(Blocks.SOUL_SAND, Blocks.SOUL_SOIL);
         this.tag(BlockTags.WALL_POST_OVERRIDE)
-            .add(Blocks.TORCH, Blocks.SOUL_TORCH, Blocks.REDSTONE_TORCH, Blocks.COPPER_TORCH, Blocks.TRIPWIRE)
+            .add(Blocks.TORCH, Blocks.SOUL_TORCH, Blocks.REDSTONE_TORCH, Blocks.TRIPWIRE)
             .addTag(BlockTags.SIGNS)
             .addTag(BlockTags.BANNERS)
             .addTag(BlockTags.PRESSURE_PLATES)
@@ -309,7 +_,6 @@
         this.tag(BlockTags.STRIDER_WARM_BLOCKS).add(Blocks.LAVA);
         this.tag(BlockTags.CAMPFIRES).add(Blocks.CAMPFIRE, Blocks.SOUL_CAMPFIRE);
         this.tag(BlockTags.GUARDED_BY_PIGLINS)
-            .addTag(BlockTags.COPPER_CHESTS)
             .add(Blocks.GOLD_BLOCK, Blocks.BARREL, Blocks.CHEST, Blocks.ENDER_CHEST, Blocks.GILDED_BLACKSTONE, Blocks.TRAPPED_CHEST, Blocks.RAW_GOLD_BLOCK)
             .addTag(BlockTags.SHULKER_BOXES)
             .addTag(BlockTags.GOLD_ORES);
@@ -477,7 +_,6 @@
             .add(Blocks.BAMBOO_MOSAIC, Blocks.BAMBOO_MOSAIC_SLAB, Blocks.BAMBOO_MOSAIC_STAIRS)
             .addTag(BlockTags.BAMBOO_BLOCKS)
             .add(Blocks.CHISELED_BOOKSHELF)
-            .addTag(BlockTags.WOODEN_SHELVES)
             .add(Blocks.CREAKING_HEART);
         this.tag(BlockTags.MINEABLE_WITH_HOE)
             .addTag(BlockTags.LEAVES)
@@ -547,6 +_,8 @@
                 Blocks.MOSSY_STONE_BRICKS,
                 Blocks.CRACKED_STONE_BRICKS,
                 Blocks.CHISELED_STONE_BRICKS,
+                Blocks.IRON_BARS,
+                Blocks.CHAIN,
                 Blocks.BRICK_STAIRS,
                 Blocks.STONE_BRICK_STAIRS,
                 Blocks.NETHER_BRICKS,
@@ -716,6 +_,8 @@
                 Blocks.GRINDSTONE,
                 Blocks.STONECUTTER,
                 Blocks.BELL,
+                Blocks.LANTERN,
+                Blocks.SOUL_LANTERN,
                 Blocks.WARPED_NYLIUM,
                 Blocks.CRIMSON_NYLIUM,
                 Blocks.NETHERITE_BLOCK,
@@ -775,6 +_,7 @@
                 Blocks.WAXED_WEATHERED_CUT_COPPER_SLAB,
                 Blocks.WAXED_EXPOSED_CUT_COPPER_SLAB,
                 Blocks.WAXED_CUT_COPPER_SLAB,
+                Blocks.LIGHTNING_ROD,
                 Blocks.POINTED_DRIPSTONE,
                 Blocks.DRIPSTONE_BLOCK,
                 Blocks.DEEPSLATE,
@@ -889,13 +_,7 @@
                 Blocks.RESIN_BRICK_WALL,
                 Blocks.RESIN_BRICK_STAIRS,
                 Blocks.CHISELED_RESIN_BRICKS
-            )
-            .addTag(BlockTags.COPPER_CHESTS)
-            .addTag(BlockTags.COPPER_GOLEM_STATUES)
-            .addTag(BlockTags.LIGHTNING_RODS)
-            .addTag(BlockTags.LANTERNS)
-            .addTag(BlockTags.CHAINS)
-            .addTag(BlockTags.BARS);
+            );
         this.tag(BlockTags.MINEABLE_WITH_SHOVEL)
             .add(
                 Blocks.CLAY,
@@ -987,6 +_,7 @@
                 Blocks.WAXED_OXIDIZED_CUT_COPPER_SLAB,
                 Blocks.WAXED_OXIDIZED_CUT_COPPER_STAIRS,
                 Blocks.WAXED_OXIDIZED_CUT_COPPER,
+                Blocks.LIGHTNING_ROD,
                 Blocks.CRAFTER,
                 Blocks.CHISELED_COPPER,
                 Blocks.EXPOSED_CHISELED_COPPER,
@@ -1020,14 +_,11 @@
                 Blocks.WAXED_EXPOSED_COPPER_TRAPDOOR,
                 Blocks.WAXED_WEATHERED_COPPER_TRAPDOOR,
                 Blocks.WAXED_OXIDIZED_COPPER_TRAPDOOR
-            )
-            .addTag(BlockTags.COPPER_CHESTS)
-            .addTag(BlockTags.LIGHTNING_RODS);
+            );
         this.tag(BlockTags.INCORRECT_FOR_NETHERITE_TOOL);
         this.tag(BlockTags.INCORRECT_FOR_DIAMOND_TOOL);
         this.tag(BlockTags.INCORRECT_FOR_IRON_TOOL).addTag(BlockTags.NEEDS_DIAMOND_TOOL);
         this.tag(BlockTags.INCORRECT_FOR_STONE_TOOL).addTag(BlockTags.NEEDS_DIAMOND_TOOL).addTag(BlockTags.NEEDS_IRON_TOOL);
-        this.tag(BlockTags.INCORRECT_FOR_COPPER_TOOL).addTag(BlockTags.NEEDS_DIAMOND_TOOL).addTag(BlockTags.NEEDS_IRON_TOOL);
         this.tag(BlockTags.INCORRECT_FOR_GOLD_TOOL).addTag(BlockTags.NEEDS_DIAMOND_TOOL).addTag(BlockTags.NEEDS_IRON_TOOL).addTag(BlockTags.NEEDS_STONE_TOOL);
         this.tag(BlockTags.INCORRECT_FOR_WOODEN_TOOL).addTag(BlockTags.NEEDS_DIAMOND_TOOL).addTag(BlockTags.NEEDS_IRON_TOOL).addTag(BlockTags.NEEDS_STONE_TOOL);
         this.tag(BlockTags.FEATURES_CANNOT_REPLACE)
