--- a/net/minecraft/server/packs/resources/PreparableReloadListener.java
+++ b/net/minecraft/server/packs/resources/PreparableReloadListener.java
@@ -1,19 +_,11 @@
 package net.minecraft.server.packs.resources;
 
-import java.util.IdentityHashMap;
-import java.util.Map;
-import java.util.Objects;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 
 @FunctionalInterface
 public interface PreparableReloadListener {
-    CompletableFuture<Void> reload(
-        PreparableReloadListener.SharedState p_433552_, Executor p_10642_, PreparableReloadListener.PreparationBarrier p_10638_, Executor p_10643_
-    );
-
-    default void prepareSharedState(PreparableReloadListener.SharedState p_436024_) {
-    }
+    CompletableFuture<Void> reload(PreparableReloadListener.PreparationBarrier p_10638_, ResourceManager p_10639_, Executor p_10642_, Executor p_10643_);
 
     default String getName() {
         return this.getClass().getSimpleName();
@@ -22,29 +_,5 @@
     @FunctionalInterface
     public interface PreparationBarrier {
         <T> CompletableFuture<T> wait(T p_10644_);
-    }
-
-    public static final class SharedState {
-        private final ResourceManager manager;
-        private final Map<PreparableReloadListener.StateKey<?>, Object> state = new IdentityHashMap<>();
-
-        public SharedState(ResourceManager p_435353_) {
-            this.manager = p_435353_;
-        }
-
-        public ResourceManager resourceManager() {
-            return this.manager;
-        }
-
-        public <T> void set(PreparableReloadListener.StateKey<T> p_433408_, T p_435497_) {
-            this.state.put(p_433408_, p_435497_);
-        }
-
-        public <T> T get(PreparableReloadListener.StateKey<T> p_432816_) {
-            return Objects.requireNonNull((T)this.state.get(p_432816_));
-        }
-    }
-
-    public static final class StateKey<T> {
     }
 }
