--- a/net/minecraft/network/FriendlyByteBuf.java
+++ b/net/minecraft/network/FriendlyByteBuf.java
@@ -490,14 +_,6 @@
         writeVec3(this, p_296062_);
     }
 
-    public Vec3 readLpVec3() {
-        return LpVec3.read(this);
-    }
-
-    public void writeLpVec3(Vec3 p_436604_) {
-        LpVec3.write(this, p_436604_);
-    }
-
     public <T extends Enum<T>> T readEnum(Class<T> p_130067_) {
         return p_130067_.getEnumConstants()[this.readVarInt()];
     }
@@ -1656,5 +_,10 @@
     @Override
     public boolean release(int p_130347_) {
         return this.source.release(p_130347_);
+    }
+
+    @org.jetbrains.annotations.ApiStatus.Internal
+    public ByteBuf getSource() {
+        return this.source;
     }
 }
