--- a/net/minecraft/world/level/block/entity/ContainerOpenersCounter.java
+++ b/net/minecraft/world/level/block/entity/ContainerOpenersCounter.java
@@ -1,14 +_,11 @@
 package net.minecraft.world.level.block.entity;
 
 import java.util.List;
-import java.util.stream.Collectors;
 import net.minecraft.core.BlockPos;
-import net.minecraft.world.entity.ContainerUser;
-import net.minecraft.world.entity.Entity;
-import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.entity.EntityTypeTest;
 import net.minecraft.world.level.gameevent.GameEvent;
 import net.minecraft.world.phys.AABB;
 
@@ -23,52 +_,43 @@
 
     protected abstract void openerCountChanged(Level p_155463_, BlockPos p_155464_, BlockState p_155465_, int p_155466_, int p_155467_);
 
-    public abstract boolean isOwnContainer(Player p_155451_);
+    protected abstract boolean isOwnContainer(Player p_155451_);
 
-    public void incrementOpeners(LivingEntity p_432931_, Level p_155454_, BlockPos p_155455_, BlockState p_155456_, double p_434817_) {
+    public void incrementOpeners(Player p_155453_, Level p_155454_, BlockPos p_155455_, BlockState p_155456_) {
         int i = this.openCount++;
         if (i == 0) {
             this.onOpen(p_155454_, p_155455_, p_155456_);
-            p_155454_.gameEvent(p_432931_, GameEvent.CONTAINER_OPEN, p_155455_);
+            p_155454_.gameEvent(p_155453_, GameEvent.CONTAINER_OPEN, p_155455_);
             scheduleRecheck(p_155454_, p_155455_, p_155456_);
         }
 
         this.openerCountChanged(p_155454_, p_155455_, p_155456_, i, this.openCount);
-        this.maxInteractionRange = Math.max(p_434817_, this.maxInteractionRange);
+        this.maxInteractionRange = Math.max(p_155453_.blockInteractionRange(), this.maxInteractionRange);
     }
 
-    public void decrementOpeners(LivingEntity p_434198_, Level p_155470_, BlockPos p_155471_, BlockState p_155472_) {
+    public void decrementOpeners(Player p_155469_, Level p_155470_, BlockPos p_155471_, BlockState p_155472_) {
         int i = this.openCount--;
         if (this.openCount == 0) {
             this.onClose(p_155470_, p_155471_, p_155472_);
-            p_155470_.gameEvent(p_434198_, GameEvent.CONTAINER_CLOSE, p_155471_);
+            p_155470_.gameEvent(p_155469_, GameEvent.CONTAINER_CLOSE, p_155471_);
             this.maxInteractionRange = 0.0;
         }
 
         this.openerCountChanged(p_155470_, p_155471_, p_155472_, i, this.openCount);
     }
 
-    public List<ContainerUser> getEntitiesWithContainerOpen(Level p_435399_, BlockPos p_435474_) {
+    private List<Player> getPlayersWithContainerOpen(Level p_320335_, BlockPos p_320624_) {
         double d0 = this.maxInteractionRange + 4.0;
-        AABB aabb = new AABB(p_435474_).inflate(d0);
-        return p_435399_.getEntities((Entity)null, aabb, p_435398_ -> this.hasContainerOpen(p_435398_, p_435474_))
-            .stream()
-            .map(p_433993_ -> (ContainerUser)p_433993_)
-            .collect(Collectors.toList());
-    }
-
-    private boolean hasContainerOpen(Entity p_434620_, BlockPos p_435060_) {
-        return p_434620_ instanceof ContainerUser containeruser && !containeruser.getLivingEntity().isSpectator()
-            ? containeruser.hasContainerOpen(this, p_435060_)
-            : false;
+        AABB aabb = new AABB(p_320624_).inflate(d0);
+        return p_320335_.getEntities(EntityTypeTest.forClass(Player.class), aabb, this::isOwnContainer);
     }
 
     public void recheckOpeners(Level p_155477_, BlockPos p_155478_, BlockState p_155479_) {
-        List<ContainerUser> list = this.getEntitiesWithContainerOpen(p_155477_, p_155478_);
+        List<Player> list = this.getPlayersWithContainerOpen(p_155477_, p_155478_);
         this.maxInteractionRange = 0.0;
 
-        for (ContainerUser containeruser : list) {
-            this.maxInteractionRange = Math.max(containeruser.getContainerInteractionRange(), this.maxInteractionRange);
+        for (Player player : list) {
+            this.maxInteractionRange = Math.max(player.blockInteractionRange(), this.maxInteractionRange);
         }
 
         int i = list.size();
