--- a/net/minecraft/world/level/block/entity/TheEndGatewayBlockEntity.java
+++ b/net/minecraft/world/level/block/entity/TheEndGatewayBlockEntity.java
@@ -113,7 +_,7 @@
     }
 
     public static void triggerCooldown(Level p_155850_, BlockPos p_155851_, BlockState p_155852_, TheEndGatewayBlockEntity p_155853_) {
-        if (!p_155850_.isClientSide()) {
+        if (!p_155850_.isClientSide) {
             p_155853_.teleportCooldown = 40;
             p_155850_.blockEvent(p_155851_, p_155852_.getBlock(), 1, 0);
             setChanged(p_155850_, p_155851_, p_155852_);
@@ -255,7 +_,7 @@
 
     @Override
     public boolean shouldRenderFace(Direction p_59959_) {
-        return Block.shouldRenderFace(this.getBlockState(), this.level.getBlockState(this.getBlockPos().relative(p_59959_)), p_59959_);
+        return Block.shouldRenderFace(this.level, this.worldPosition, this.getBlockState(), this.level.getBlockState(this.getBlockPos().relative(p_59959_)), p_59959_);
     }
 
     public int getParticleAmount() {
