--- a/net/minecraft/world/level/chunk/storage/SectionStorage.java
+++ b/net/minecraft/world/level/chunk/storage/SectionStorage.java
@@ -291,6 +_,10 @@
         }
     }
 
+    static int getVersion(Dynamic<?> p_63806_) {
+        return p_63806_.get("DataVersion").asInt(1945);
+    }
+
     public void flush(ChunkPos p_63797_) {
         if (this.dirtyChunks.remove(p_63797_.toLong())) {
             this.writeChunk(p_63797_);
@@ -302,22 +_,37 @@
         this.simpleRegionStorage.close();
     }
 
+    /**
+     * Neo: Removes the data for the given chunk position.
+     * See PR #937
+     */
+    public void remove(ChunkPos chunkPos) {
+        synchronized (this.loadLock) {
+            for (int y = this.levelHeightAccessor.getMinSectionY(); y <= this.levelHeightAccessor.getMaxSectionY(); y++) {
+                this.storage.remove(getKey(chunkPos, y));
+            }
+            this.loadedChunks.remove(chunkPos.toLong());
+        }
+    }
+
     record PackedChunk<T>(Int2ObjectMap<T> sectionsByY, boolean versionChanged) {
         public static <T> SectionStorage.PackedChunk<T> parse(
             Codec<T> p_365233_, DynamicOps<Tag> p_363840_, Tag p_364375_, SimpleRegionStorage p_362076_, LevelHeightAccessor p_362314_
         ) {
             Dynamic<Tag> dynamic = new Dynamic<>(p_363840_, p_364375_);
-            Dynamic<Tag> dynamic1 = p_362076_.upgradeChunkTag(dynamic, 1945);
-            boolean flag = dynamic != dynamic1;
+            int i = SectionStorage.getVersion(dynamic);
+            int j = SharedConstants.getCurrentVersion().dataVersion().version();
+            boolean flag = i != j;
+            Dynamic<Tag> dynamic1 = p_362076_.upgradeChunkTag(dynamic, i);
             OptionalDynamic<Tag> optionaldynamic = dynamic1.get("Sections");
             Int2ObjectMap<T> int2objectmap = new Int2ObjectOpenHashMap<>();
 
-            for (int i = p_362314_.getMinSectionY(); i <= p_362314_.getMaxSectionY(); i++) {
-                Optional<T> optional = optionaldynamic.get(Integer.toString(i))
+            for (int k = p_362314_.getMinSectionY(); k <= p_362314_.getMaxSectionY(); k++) {
+                Optional<T> optional = optionaldynamic.get(Integer.toString(k))
                     .result()
                     .flatMap(p_361362_ -> p_365233_.parse((Dynamic<Tag>)p_361362_).resultOrPartial(SectionStorage.LOGGER::error));
                 if (optional.isPresent()) {
-                    int2objectmap.put(i, optional.get());
+                    int2objectmap.put(k, optional.get());
                 }
             }
 
