--- a/net/minecraft/client/model/ArmedModel.java
+++ b/net/minecraft/client/model/ArmedModel.java
@@ -1,12 +_,11 @@
 package net.minecraft.client.model;
 
 import com.mojang.blaze3d.vertex.PoseStack;
-import net.minecraft.client.renderer.entity.state.EntityRenderState;
 import net.minecraft.world.entity.HumanoidArm;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
 @OnlyIn(Dist.CLIENT)
-public interface ArmedModel<T extends EntityRenderState> {
-    void translateToHand(T p_435159_, HumanoidArm p_102108_, PoseStack p_102109_);
+public interface ArmedModel {
+    void translateToHand(HumanoidArm p_102108_, PoseStack p_102109_);
 }
