--- a/net/minecraft/client/model/BannerModel.java
+++ b/net/minecraft/client/model/BannerModel.java
@@ -7,12 +_,11 @@
 import net.minecraft.client.model.geom.builders.MeshDefinition;
 import net.minecraft.client.model.geom.builders.PartDefinition;
 import net.minecraft.client.renderer.RenderType;
-import net.minecraft.util.Unit;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
 @OnlyIn(Dist.CLIENT)
-public class BannerModel extends Model<Unit> {
+public class BannerModel extends Model {
     public static final int BANNER_WIDTH = 20;
     public static final int BANNER_HEIGHT = 40;
     public static final String FLAG = "flag";
