--- a/net/minecraft/client/model/BellModel.java
+++ b/net/minecraft/client/model/BellModel.java
@@ -1,6 +_,5 @@
 package net.minecraft.client.model;
 
-import javax.annotation.Nullable;
 import net.minecraft.client.model.geom.ModelPart;
 import net.minecraft.client.model.geom.PartPose;
 import net.minecraft.client.model.geom.builders.CubeListBuilder;
@@ -10,11 +_,12 @@
 import net.minecraft.client.renderer.RenderType;
 import net.minecraft.core.Direction;
 import net.minecraft.util.Mth;
+import net.minecraft.world.level.block.entity.BellBlockEntity;
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
 @OnlyIn(Dist.CLIENT)
-public class BellModel extends Model<BellModel.State> {
+public class BellModel extends Model {
     private static final String BELL_BODY = "bell_body";
     private final ModelPart bellBody;
 
@@ -35,35 +_,24 @@
         return LayerDefinition.create(meshdefinition, 32, 32);
     }
 
-    public void setupAnim(BellModel.State p_432780_) {
-        super.setupAnim(p_432780_);
-        float f = 0.0F;
+    public void setupAnim(BellBlockEntity p_365361_, float p_364638_) {
+        float f = p_365361_.ticks + p_364638_;
         float f1 = 0.0F;
-        if (p_432780_.shakeDirection != null) {
-            float f2 = Mth.sin(p_432780_.ticks / (float) Math.PI) / (4.0F + p_432780_.ticks / 3.0F);
-            switch (p_432780_.shakeDirection) {
-                case NORTH:
-                    f = -f2;
-                    break;
-                case SOUTH:
-                    f = f2;
-                    break;
-                case EAST:
-                    f1 = -f2;
-                    break;
-                case WEST:
-                    f1 = f2;
+        float f2 = 0.0F;
+        if (p_365361_.shaking) {
+            float f3 = Mth.sin(f / (float) Math.PI) / (4.0F + f / 3.0F);
+            if (p_365361_.clickDirection == Direction.NORTH) {
+                f1 = -f3;
+            } else if (p_365361_.clickDirection == Direction.SOUTH) {
+                f1 = f3;
+            } else if (p_365361_.clickDirection == Direction.EAST) {
+                f2 = -f3;
+            } else if (p_365361_.clickDirection == Direction.WEST) {
+                f2 = f3;
             }
         }
 
-        this.bellBody.xRot = f;
-        this.bellBody.zRot = f1;
-    }
-
-    @OnlyIn(Dist.CLIENT)
-    public static class State {
-        public float ticks;
-        @Nullable
-        public Direction shakeDirection;
+        this.bellBody.xRot = f1;
+        this.bellBody.zRot = f2;
     }
 }
