--- a/net/minecraft/world/entity/monster/Guardian.java
+++ b/net/minecraft/world/entity/monster/Guardian.java
@@ -124,7 +_,7 @@
     public LivingEntity getActiveAttackTarget() {
         if (!this.hasActiveAttackTarget()) {
             return null;
-        } else if (this.level().isClientSide()) {
+        } else if (this.level().isClientSide) {
             if (this.clientSideCachedAttackTarget != null) {
                 return this.clientSideCachedAttackTarget;
             } else {
@@ -185,7 +_,7 @@
     @Override
     public void aiStep() {
         if (this.isAlive()) {
-            if (this.level().isClientSide()) {
+            if (this.level().isClientSide) {
                 this.clientSideTailAnimationO = this.clientSideTailAnimation;
                 if (!this.isInWater()) {
                     this.clientSideTailAnimationSpeed = 2.0F;
