--- a/net/minecraft/world/entity/monster/SpellcasterIllager.java
+++ b/net/minecraft/world/entity/monster/SpellcasterIllager.java
@@ -57,7 +_,7 @@
     }
 
     public boolean isCastingSpell() {
-        return this.level().isClientSide() ? this.entityData.get(DATA_SPELL_CASTING_ID) > 0 : this.spellCastingTickCount > 0;
+        return this.level().isClientSide ? this.entityData.get(DATA_SPELL_CASTING_ID) > 0 : this.spellCastingTickCount > 0;
     }
 
     public void setIsCastingSpell(SpellcasterIllager.IllagerSpell p_33728_) {
@@ -66,7 +_,7 @@
     }
 
     protected SpellcasterIllager.IllagerSpell getCurrentSpell() {
-        return !this.level().isClientSide() ? this.currentSpell : SpellcasterIllager.IllagerSpell.byId(this.entityData.get(DATA_SPELL_CASTING_ID));
+        return !this.level().isClientSide ? this.currentSpell : SpellcasterIllager.IllagerSpell.byId(this.entityData.get(DATA_SPELL_CASTING_ID));
     }
 
     @Override
@@ -80,7 +_,7 @@
     @Override
     public void tick() {
         super.tick();
-        if (this.level().isClientSide() && this.isCastingSpell()) {
+        if (this.level().isClientSide && this.isCastingSpell()) {
             SpellcasterIllager.IllagerSpell spellcasterillager$illagerspell = this.getCurrentSpell();
             float f = (float)spellcasterillager$illagerspell.spellColor[0];
             float f1 = (float)spellcasterillager$illagerspell.spellColor[1];
