--- a/net/minecraft/world/entity/monster/ZombifiedPiglin.java
+++ b/net/minecraft/world/entity/monster/ZombifiedPiglin.java
@@ -221,6 +_,11 @@
     }
 
     @Override
+    protected ItemStack getSkull() {
+        return ItemStack.EMPTY;
+    }
+
+    @Override
     protected void randomizeReinforcementsChance() {
         this.getAttribute(Attributes.SPAWN_REINFORCEMENTS_CHANCE).setBaseValue(0.0);
     }
