--- a/net/minecraft/ReportType.java
+++ b/net/minecraft/ReportType.java
@@ -131,6 +_,7 @@
         p_350666_.append("---- ");
         p_350666_.append(this.header());
         p_350666_.append(" ----\n");
+        // TODO 1.21 PORTING - do we still need to extend the header here?
         p_350666_.append("// ");
         p_350666_.append(this.getErrorComment());
         p_350666_.append('\n');
