--- a/net/minecraft/server/level/GenerationChunkHolder.java
+++ b/net/minecraft/server/level/GenerationChunkHolder.java
@@ -34,6 +_,7 @@
     private final AtomicReference<ChunkGenerationTask> task = new AtomicReference<>();
     private final AtomicInteger generationRefCount = new AtomicInteger();
     private volatile CompletableFuture<Void> generationSaveSyncFuture = CompletableFuture.completedFuture(null);
+    public net.minecraft.world.level.chunk.LevelChunk currentlyLoading; // Forge: Used to bypass future chain when loading chunks.
 
     public GenerationChunkHolder(ChunkPos p_347689_) {
         this.pos = p_347689_;
@@ -308,12 +_,14 @@
     @Nullable
     @VisibleForDebug
     public ChunkStatus getLatestStatus() {
-        ChunkStatus chunkstatus = this.startedWork.get();
-        if (chunkstatus == null) {
-            return null;
-        } else {
+        for (int i = CHUNK_STATUSES.size() - 1; i >= 0; i--) {
+            ChunkStatus chunkstatus = CHUNK_STATUSES.get(i);
             ChunkAccess chunkaccess = this.getChunkIfPresentUnchecked(chunkstatus);
-            return chunkaccess != null ? chunkstatus : chunkstatus.getParent();
+            if (chunkaccess != null) {
+                return chunkstatus;
+            }
         }
+
+        return null;
     }
 }
