--- a/net/minecraft/world/item/equipment/EquipmentAssets.java
+++ b/net/minecraft/world/item/equipment/EquipmentAssets.java
@@ -10,7 +_,6 @@
 public interface EquipmentAssets {
     ResourceKey<? extends Registry<EquipmentAsset>> ROOT_ID = ResourceKey.createRegistryKey(ResourceLocation.withDefaultNamespace("equipment_asset"));
     ResourceKey<EquipmentAsset> LEATHER = createId("leather");
-    ResourceKey<EquipmentAsset> COPPER = createId("copper");
     ResourceKey<EquipmentAsset> CHAINMAIL = createId("chainmail");
     ResourceKey<EquipmentAsset> IRON = createId("iron");
     ResourceKey<EquipmentAsset> GOLD = createId("gold");
