--- a/net/minecraft/client/renderer/chunk/ChunkSectionLayer.java
+++ b/net/minecraft/client/renderer/chunk/ChunkSectionLayer.java
@@ -1,6 +_,7 @@
 package net.minecraft.client.renderer.chunk;
 
 import com.mojang.blaze3d.pipeline.RenderPipeline;
+import com.mojang.blaze3d.pipeline.RenderTarget;
 import com.mojang.blaze3d.textures.GpuTextureView;
 import java.util.Locale;
 import net.minecraft.client.Minecraft;
@@ -54,5 +_,19 @@
         AbstractTexture abstracttexture = texturemanager.getTexture(TextureAtlas.LOCATION_BLOCKS);
         abstracttexture.setUseMipmaps(this.useMipmaps);
         return abstracttexture.getTextureView();
+    }
+
+    public RenderTarget outputTarget() {
+        Minecraft minecraft = Minecraft.getInstance();
+        switch (this) {
+            case TRANSLUCENT:
+                RenderTarget rendertarget1 = minecraft.levelRenderer.getTranslucentTarget();
+                return rendertarget1 != null ? rendertarget1 : minecraft.getMainRenderTarget();
+            case TRIPWIRE:
+                RenderTarget rendertarget = minecraft.levelRenderer.getWeatherTarget();
+                return rendertarget != null ? rendertarget : minecraft.getMainRenderTarget();
+            default:
+                return minecraft.getMainRenderTarget();
+        }
     }
 }
