--- a/net/minecraft/client/resources/model/ModelBakery.java
+++ b/net/minecraft/client/resources/model/ModelBakery.java
@@ -11,7 +_,6 @@
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 import net.minecraft.client.model.geom.EntityModelSet;
-import net.minecraft.client.renderer.PlayerSkinRenderCache;
 import net.minecraft.client.renderer.RenderType;
 import net.minecraft.client.renderer.Sheets;
 import net.minecraft.client.renderer.block.model.BlockStateModel;
@@ -23,6 +_,7 @@
 import net.minecraft.client.renderer.item.ItemModel;
 import net.minecraft.client.renderer.item.MissingItemModel;
 import net.minecraft.client.renderer.item.ModelRenderProperties;
+import net.minecraft.client.renderer.texture.TextureAtlas;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
 import net.minecraft.resources.ResourceLocation;
 import net.minecraft.util.thread.ParallelMapTransform;
@@ -33,11 +_,11 @@
 
 @OnlyIn(Dist.CLIENT)
 public class ModelBakery {
-    public static final Material FIRE_0 = Sheets.BLOCKS_MAPPER.defaultNamespaceApply("fire_0");
-    public static final Material FIRE_1 = Sheets.BLOCKS_MAPPER.defaultNamespaceApply("fire_1");
-    public static final Material LAVA_FLOW = Sheets.BLOCKS_MAPPER.defaultNamespaceApply("lava_flow");
-    public static final Material WATER_FLOW = Sheets.BLOCKS_MAPPER.defaultNamespaceApply("water_flow");
-    public static final Material WATER_OVERLAY = Sheets.BLOCKS_MAPPER.defaultNamespaceApply("water_overlay");
+    public static final Material FIRE_0 = new Material(TextureAtlas.LOCATION_BLOCKS, ResourceLocation.withDefaultNamespace("block/fire_0"));
+    public static final Material FIRE_1 = new Material(TextureAtlas.LOCATION_BLOCKS, ResourceLocation.withDefaultNamespace("block/fire_1"));
+    public static final Material LAVA_FLOW = new Material(TextureAtlas.LOCATION_BLOCKS, ResourceLocation.withDefaultNamespace("block/lava_flow"));
+    public static final Material WATER_FLOW = new Material(TextureAtlas.LOCATION_BLOCKS, ResourceLocation.withDefaultNamespace("block/water_flow"));
+    public static final Material WATER_OVERLAY = new Material(TextureAtlas.LOCATION_BLOCKS, ResourceLocation.withDefaultNamespace("block/water_overlay"));
     public static final Material BANNER_BASE = new Material(Sheets.BANNER_SHEET, ResourceLocation.withDefaultNamespace("entity/banner_base"));
     public static final Material SHIELD_BASE = new Material(Sheets.SHIELD_SHEET, ResourceLocation.withDefaultNamespace("entity/shield_base"));
     public static final Material NO_PATTERN_SHIELD = new Material(Sheets.SHIELD_SHEET, ResourceLocation.withDefaultNamespace("entity/shield_base_nopattern"));
@@ -51,29 +_,40 @@
     public static final List<RenderType> DESTROY_TYPES = BREAKING_LOCATIONS.stream().map(RenderType::crumbling).collect(Collectors.toList());
     static final Logger LOGGER = LogUtils.getLogger();
     private final EntityModelSet entityModelSet;
-    private final MaterialSet materials;
-    private final PlayerSkinRenderCache playerSkinRenderCache;
     private final Map<BlockState, BlockStateModel.UnbakedRoot> unbakedBlockStateModels;
     private final Map<ResourceLocation, ClientItem> clientInfos;
     final Map<ResourceLocation, ResolvedModel> resolvedModels;
     final ResolvedModel missingModel;
+    private final net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.LoadedModels standaloneModels;
 
+    /**
+     * @deprecated Neo: use {@link #ModelBakery(EntityModelSet, Map, Map, Map, ResolvedModel, net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.LoadedModels)} instead
+     */
+    @Deprecated
     public ModelBakery(
         EntityModelSet p_388903_,
-        MaterialSet p_434703_,
-        PlayerSkinRenderCache p_440085_,
         Map<BlockState, BlockStateModel.UnbakedRoot> p_251087_,
         Map<ResourceLocation, ClientItem> p_250416_,
         Map<ResourceLocation, ResolvedModel> p_388404_,
         ResolvedModel p_404940_
     ) {
+        this(p_388903_, p_251087_, p_250416_, p_388404_, p_404940_, net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.LoadedModels.EMPTY);
+    }
+
+    public ModelBakery(
+            EntityModelSet p_388903_,
+            Map<BlockState, BlockStateModel.UnbakedRoot> p_251087_,
+            Map<ResourceLocation, ClientItem> p_250416_,
+            Map<ResourceLocation, ResolvedModel> p_388404_,
+            ResolvedModel p_404940_,
+            net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.LoadedModels standaloneModels
+    ) {
         this.entityModelSet = p_388903_;
-        this.materials = p_434703_;
-        this.playerSkinRenderCache = p_440085_;
         this.unbakedBlockStateModels = p_251087_;
         this.clientInfos = p_250416_;
         this.resolvedModels = p_388404_;
         this.missingModel = p_404940_;
+        this.standaloneModels = standaloneModels;
     }
 
     public CompletableFuture<ModelBakery.BakingResult> bakeModels(SpriteGetter p_404922_, Executor p_405407_) {
@@ -91,26 +_,23 @@
         );
         CompletableFuture<Map<ResourceLocation, ItemModel>> completablefuture1 = ParallelMapTransform.schedule(
             this.clientInfos,
-            (p_438821_, p_438822_) -> {
+            (p_404123_, p_404124_) -> {
                 try {
-                    return p_438822_.model()
+                    return p_404124_.model()
                         .bake(
                             new ItemModel.BakingContext(
-                                modelbakery$modelbakerimpl,
-                                this.entityModelSet,
-                                this.materials,
-                                this.playerSkinRenderCache,
-                                modelbakery$missingmodels.item,
-                                p_438822_.registrySwapper()
+                                modelbakery$modelbakerimpl, this.entityModelSet, modelbakery$missingmodels.item, p_404124_.registrySwapper()
                             )
                         );
                 } catch (Exception exception) {
-                    LOGGER.warn("Unable to bake item model: '{}'", p_438821_, exception);
+                    LOGGER.warn("Unable to bake item model: '{}'", p_404123_, exception);
                     return null;
                 }
             },
             p_405407_
         );
+        CompletableFuture<net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.BakedModels> standaloneModelsFuture =
+                net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.bake(this.standaloneModels, modelbakery$modelbakerimpl, p_405407_);
         Map<ResourceLocation, ClientItem.Properties> map = new HashMap<>(this.clientInfos.size());
         this.clientInfos.forEach((p_404116_, p_404117_) -> {
             ClientItem.Properties clientitem$properties = p_404117_.properties();
@@ -118,10 +_,11 @@
                 map.put(p_404116_, clientitem$properties);
             }
         });
+        completablefuture = completablefuture.thenCombine(standaloneModelsFuture, (stateModels, standaloneModels) -> stateModels);
         return completablefuture.thenCombine(
             completablefuture1,
             (p_404127_, p_404128_) -> new ModelBakery.BakingResult(
-                modelbakery$missingmodels, (Map<BlockState, BlockStateModel>)p_404127_, (Map<ResourceLocation, ItemModel>)p_404128_, map
+                modelbakery$missingmodels, (Map<BlockState, BlockStateModel>)p_404127_, (Map<ResourceLocation, ItemModel>)p_404128_, map, standaloneModelsFuture.join()
             )
         );
     }
@@ -132,7 +_,20 @@
         Map<BlockState, BlockStateModel> blockStateModels,
         Map<ResourceLocation, ItemModel> itemStackModels,
         Map<ResourceLocation, ClientItem.Properties> itemProperties
+        , net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.BakedModels standaloneModels
     ) {
+        /**
+         * @deprecated Neo: use {@link #BakingResult(ModelBakery.MissingModels, Map, Map, Map, net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.BakedModels)} instead
+         */
+        @Deprecated
+        public BakingResult(
+                ModelBakery.MissingModels missingModels,
+                Map<BlockState, BlockStateModel> blockStateModels,
+                Map<ResourceLocation, ItemModel> itemStackModels,
+                Map<ResourceLocation, ClientItem.Properties> itemProperties
+        ) {
+            this(missingModels, blockStateModels, itemStackModels, itemProperties, net.neoforged.neoforge.client.model.standalone.StandaloneModelLoader.BakedModels.EMPTY);
+        }
     }
 
     @OnlyIn(Dist.CLIENT)
