--- a/net/minecraft/server/commands/WhitelistCommand.java
+++ b/net/minecraft/server/commands/WhitelistCommand.java
@@ -1,5 +_,6 @@
 package net.minecraft.server.commands;
 
+import com.mojang.authlib.GameProfile;
 import com.mojang.brigadier.CommandDispatcher;
 import com.mojang.brigadier.context.CommandContext;
 import com.mojang.brigadier.exceptions.CommandSyntaxException;
@@ -11,11 +_,10 @@
 import net.minecraft.commands.SharedSuggestionProvider;
 import net.minecraft.commands.arguments.GameProfileArgument;
 import net.minecraft.network.chat.Component;
-import net.minecraft.server.players.NameAndId;
+import net.minecraft.server.level.ServerPlayer;
 import net.minecraft.server.players.PlayerList;
 import net.minecraft.server.players.UserWhiteList;
 import net.minecraft.server.players.UserWhiteListEntry;
-import net.minecraft.world.entity.player.Player;
 
 public class WhitelistCommand {
     private static final SimpleCommandExceptionType ERROR_ALREADY_ENABLED = new SimpleCommandExceptionType(
@@ -43,15 +_,14 @@
                         .then(
                             Commands.argument("targets", GameProfileArgument.gameProfile())
                                 .suggests(
-                                    (p_432438_, p_432439_) -> {
-                                        PlayerList playerlist = p_432438_.getSource().getServer().getPlayerList();
+                                    (p_139216_, p_139217_) -> {
+                                        PlayerList playerlist = p_139216_.getSource().getServer().getPlayerList();
                                         return SharedSuggestionProvider.suggest(
                                             playerlist.getPlayers()
                                                 .stream()
-                                                .map(Player::nameAndId)
-                                                .filter(p_432443_ -> !playerlist.getWhiteList().isWhiteListed(p_432443_))
-                                                .map(NameAndId::name),
-                                            p_432439_
+                                                .filter(p_426961_ -> !playerlist.getWhiteList().isWhiteListed(p_426961_.getGameProfile()))
+                                                .map(p_426959_ -> p_426959_.getGameProfile().getName()),
+                                            p_139217_
                                         );
                                     }
                                 )
@@ -81,15 +_,15 @@
         return 1;
     }
 
-    private static int addPlayers(CommandSourceStack p_139211_, Collection<NameAndId> p_139212_) throws CommandSyntaxException {
+    private static int addPlayers(CommandSourceStack p_139211_, Collection<GameProfile> p_139212_) throws CommandSyntaxException {
         UserWhiteList userwhitelist = p_139211_.getServer().getPlayerList().getWhiteList();
         int i = 0;
 
-        for (NameAndId nameandid : p_139212_) {
-            if (!userwhitelist.isWhiteListed(nameandid)) {
-                UserWhiteListEntry userwhitelistentry = new UserWhiteListEntry(nameandid);
+        for (GameProfile gameprofile : p_139212_) {
+            if (!userwhitelist.isWhiteListed(gameprofile)) {
+                UserWhiteListEntry userwhitelistentry = new UserWhiteListEntry(gameprofile);
                 userwhitelist.add(userwhitelistentry);
-                p_139211_.sendSuccess(() -> Component.translatable("commands.whitelist.add.success", Component.literal(nameandid.name())), true);
+                p_139211_.sendSuccess(() -> Component.translatable("commands.whitelist.add.success", Component.literal(gameprofile.getName())), true);
                 i++;
             }
         }
@@ -101,15 +_,15 @@
         }
     }
 
-    private static int removePlayers(CommandSourceStack p_139221_, Collection<NameAndId> p_139222_) throws CommandSyntaxException {
+    private static int removePlayers(CommandSourceStack p_139221_, Collection<GameProfile> p_139222_) throws CommandSyntaxException {
         UserWhiteList userwhitelist = p_139221_.getServer().getPlayerList().getWhiteList();
         int i = 0;
 
-        for (NameAndId nameandid : p_139222_) {
-            if (userwhitelist.isWhiteListed(nameandid)) {
-                UserWhiteListEntry userwhitelistentry = new UserWhiteListEntry(nameandid);
+        for (GameProfile gameprofile : p_139222_) {
+            if (userwhitelist.isWhiteListed(gameprofile)) {
+                UserWhiteListEntry userwhitelistentry = new UserWhiteListEntry(gameprofile);
                 userwhitelist.remove(userwhitelistentry);
-                p_139221_.sendSuccess(() -> Component.translatable("commands.whitelist.remove.success", Component.literal(nameandid.name())), true);
+                p_139221_.sendSuccess(() -> Component.translatable("commands.whitelist.remove.success", Component.literal(gameprofile.getName())), true);
                 i++;
             }
         }
