--- a/net/minecraft/server/commands/WorldBorderCommand.java
+++ b/net/minecraft/server/commands/WorldBorderCommand.java
@@ -54,21 +_,21 @@
                         .then(
                             Commands.argument("distance", DoubleArgumentType.doubleArg(-5.999997E7F, 5.999997E7F))
                                 .executes(
-                                    p_432445_ -> setSize(
-                                        p_432445_.getSource(),
-                                        p_432445_.getSource().getLevel().getWorldBorder().getSize() + DoubleArgumentType.getDouble(p_432445_, "distance"),
+                                    p_425258_ -> setSize(
+                                        p_425258_.getSource(),
+                                        p_425258_.getSource().getLevel().getWorldBorder().getSize() + DoubleArgumentType.getDouble(p_425258_, "distance"),
                                         0L
                                     )
                                 )
                                 .then(
                                     Commands.argument("time", IntegerArgumentType.integer(0))
                                         .executes(
-                                            p_432444_ -> setSize(
-                                                p_432444_.getSource(),
-                                                p_432444_.getSource().getLevel().getWorldBorder().getSize()
-                                                    + DoubleArgumentType.getDouble(p_432444_, "distance"),
-                                                p_432444_.getSource().getLevel().getWorldBorder().getLerpRemainingTime()
-                                                    + IntegerArgumentType.getInteger(p_432444_, "time") * 1000L
+                                            p_425259_ -> setSize(
+                                                p_425259_.getSource(),
+                                                p_425259_.getSource().getLevel().getWorldBorder().getSize()
+                                                    + DoubleArgumentType.getDouble(p_425259_, "distance"),
+                                                p_425259_.getSource().getLevel().getWorldBorder().getLerpRemainingTime()
+                                                    + IntegerArgumentType.getInteger(p_425259_, "time") * 1000L
                                             )
                                         )
                                 )
