--- a/net/minecraft/core/particles/ParticleTypes.java
+++ b/net/minecraft/core/particles/ParticleTypes.java
@@ -18,7 +_,6 @@
     );
     public static final SimpleParticleType BUBBLE = register("bubble", false);
     public static final SimpleParticleType CLOUD = register("cloud", false);
-    public static final SimpleParticleType COPPER_FIRE_FLAME = register("copper_fire_flame", false);
     public static final SimpleParticleType CRIT = register("crit", false);
     public static final SimpleParticleType DAMAGE_INDICATOR = register("damage_indicator", true);
     public static final SimpleParticleType DRAGON_BREATH = register("dragon_breath", false);
